<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134228602-6"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134228602-6');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>üìà üêâ ü§≥üèø D√©veloppement de prot√©ines dans le cloud en utilisant Python et Transcriptic ou Comment cr√©er une prot√©ine pour 360 $ üë®üèø‚Äçüåæ üë®‚Äç‚öñÔ∏è üö∂üèø</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Et si vous avez une id√©e d'une prot√©ine fra√Æche et saine, et que vous voulez l'obtenir en r√©alit√©? Par exemple, aimeriez-vous cr√©er un vaccin contre H...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <link href="https://fonts.googleapis.com/css?family=Quicksand&display=swap" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script>document.write('<script src="//pagea' + 'd2.googles' + 'yndication.com/pagea' + 'd/js/a' + 'dsby' + 'google.js"><\/script>')</script>
  <script>
        var superSpecialObject = {};
        superSpecialObject['google_a' + 'd_client'] = 'ca-p' + 'ub-6974184241884155';
        superSpecialObject['enable_page_level_a' + 'ds'] = true;
       (window['a' + 'dsbygoogle'] = window['a' + 'dsbygoogle'] || []).push(superSpecialObject);
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly-es.github.io/index.html"></a>
    <div class="page-header-text">Geekly articles weekly</div>
  </header>
  <section class="page js-page"><h1>D√©veloppement de prot√©ines dans le cloud en utilisant Python et Transcriptic ou Comment cr√©er une prot√©ine pour 360 $</h1><div class="post__body post__body_full"><div class="post__text post__text-html" id="post-content-body" data-io-article-url="https://habr.com/ru/post/451124/">  Et si vous avez une id√©e d'une prot√©ine fra√Æche et saine, et que vous voulez l'obtenir en r√©alit√©?  Par exemple, aimeriez-vous cr√©er un vaccin contre <i>H. pylori</i> (comme l' <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">√©quipe slov√®ne √† iGEM 2008</a> ) en cr√©ant une prot√©ine hybride qui combine des fragments de flagelline d' <i>E. Coli</i> qui stimulent la r√©ponse immunitaire avec la flagelline <i>H. pylori</i> habituelle? <br><br><div style="text-align:center;"><img src="https://habrastorage.org/getpro/habr/post_images/e9c/dd5/691/e9cdd5691f1bab368bb400a8818b6507.png"></div>  <i><font color="gray">H. pylori Hybrid Flagellin Design Pr√©sent√© par l'√©quipe slov√®ne √† iGEM 2008</font></i> <br><br>  √âtonnamment, nous sommes tr√®s pr√®s de cr√©er toutes les prot√©ines que nous voulons sans quitter le cahier de Jupyter, gr√¢ce aux derniers d√©veloppements en g√©nomique, en biologie synth√©tique et plus r√©cemment dans les laboratoires de cloud computing. <br><br>  Dans cet article, je vais montrer le code Python de l'id√©e d'une prot√©ine √† son expression dans une cellule bact√©rienne, sans toucher √† une pipette ni parler √† personne.  Le co√ªt total ne sera que de quelques centaines de dollars!  En utilisant la <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">terminologie de Vijaya Pande de A16Z</a> , il s'agit de Biologie 2.0. <br><a name="habracut"></a><br>  Plus pr√©cis√©ment, dans l'article, le code Python du laboratoire cloud effectue les op√©rations suivantes: <br><br><ul><li>  <b>Synth√®se d'</b> une s√©quence d'ADN qui code pour n'importe quelle prot√©ine que je veux. <br></li><li>  <b>Clonage de</b> cet ADN synth√©tique dans un <b>vecteur</b> pouvant l'exprimer. <br></li><li>  <b>Transformation des</b> bact√©ries avec ce <b>vecteur</b> et confirmation de l'expression. </li></ul><br><h1>  Configuration de Python </h1><br>  Tout d'abord, les param√®tres g√©n√©raux de Python qui sont n√©cessaires pour tout bloc-notes Jupyter.  Nous importons des modules Python utiles et cr√©ons des fonctions utilitaires, principalement pour la visualisation des donn√©es. <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre><code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">import</span></span> re <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> json <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> logging <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> requests <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> itertools <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> numpy <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> np <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> seaborn <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> sns <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> pandas <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> pd <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> matplotlib <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> mpl <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> matplotlib.pyplot <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> plt <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> io <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> StringIO <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> pprint <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> pprint <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> Bio.Seq <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Seq <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> Bio.Alphabet <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> generic_dna <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> IPython.display <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> display, Image, HTML, SVG <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">uprint</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr)</span></span></span><span class="hljs-function">:</span></span> print(astr + <span class="hljs-string"><span class="hljs-string">"\n"</span></span> + <span class="hljs-string"><span class="hljs-string">"-"</span></span>*len(astr)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">show_html</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> display(HTML(<span class="hljs-string"><span class="hljs-string">'{}'</span></span>.format(astr))) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">show_svg</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr, w=</span></span><span class="hljs-number"><span class="hljs-function"><span class="hljs-params"><span class="hljs-number">1000</span></span></span></span><span class="hljs-function"><span class="hljs-params">, h=</span></span><span class="hljs-number"><span class="hljs-function"><span class="hljs-params"><span class="hljs-number">1000</span></span></span></span><span class="hljs-function"><span class="hljs-params">)</span></span></span><span class="hljs-function">:</span></span> SVG_HEAD = <span class="hljs-string"><span class="hljs-string">'''&lt;?xml version="1.0" standalone="no"?&gt;&lt;!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"&gt;'''</span></span> SVG_START = <span class="hljs-string"><span class="hljs-string">'''&lt;svg viewBox="0 0 {w:} {h:}" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink= "http://www.w3.org/1999/xlink"&gt;'''</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> display(SVG(SVG_HEAD + SVG_START.format(w=w, h=h) + astr + <span class="hljs-string"><span class="hljs-string">'&lt;/svg&gt;'</span></span>)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">table_print</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(rows, header=True)</span></span></span><span class="hljs-function">:</span></span> html = [<span class="hljs-string"><span class="hljs-string">"&lt;table&gt;"</span></span>] html_row = <span class="hljs-string"><span class="hljs-string">"&lt;/td&gt;&lt;td&gt;"</span></span>.join(k <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> k <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> rows[<span class="hljs-number"><span class="hljs-number">0</span></span>]) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;tr style='font-weight:{}'&gt;&lt;td&gt;{}&lt;/td&gt;&lt;/tr&gt;"</span></span>.format(<span class="hljs-string"><span class="hljs-string">'bold'</span></span> <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> header <span class="hljs-keyword"><span class="hljs-keyword">is</span></span> <span class="hljs-keyword"><span class="hljs-keyword">True</span></span> <span class="hljs-keyword"><span class="hljs-keyword">else</span></span> <span class="hljs-string"><span class="hljs-string">'normal'</span></span>, html_row)) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> row <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> rows[<span class="hljs-number"><span class="hljs-number">1</span></span>:]: html_row = <span class="hljs-string"><span class="hljs-string">"&lt;/td&gt;&lt;td&gt;"</span></span>.join(row) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;tr style='font-family:monospace;'&gt;&lt;td&gt;{:}&lt;/td&gt;&lt;/tr&gt;"</span></span>.format(html_row)) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;/table&gt;"</span></span>) show_html(<span class="hljs-string"><span class="hljs-string">''</span></span>.join(html)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">clean_seq</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(dna)</span></span></span><span class="hljs-function">:</span></span> dna = re.sub(<span class="hljs-string"><span class="hljs-string">"\s"</span></span>,<span class="hljs-string"><span class="hljs-string">""</span></span>,dna) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACGTN"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> dna) <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> Seq(dna, generic_dna) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">clean_aas</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(aas)</span></span></span><span class="hljs-function">:</span></span> aas = re.sub(<span class="hljs-string"><span class="hljs-string">"\s"</span></span>,<span class="hljs-string"><span class="hljs-string">""</span></span>,aas) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(aa <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACDEFGHIKLMNPQRSTVWY*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> aa <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> aas) <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> aas <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">Images</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(images, header=None, width=</span></span><span class="hljs-string"><span class="hljs-function"><span class="hljs-params"><span class="hljs-string">"100%"</span></span></span></span><span class="hljs-function"><span class="hljs-params">)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-comment"><span class="hljs-comment"># to match Image syntax if type(width)==type(1): width = "{}px".format(width) html = ["&lt;table style='width:{}'&gt;&lt;tr&gt;".format(width)] if header is not None: html += ["&lt;th&gt;{}&lt;/th&gt;".format(h) for h in header] + ["&lt;/tr&gt;&lt;tr&gt;"] for image in images: html.append("&lt;td&gt;&lt;img src='{}' /&gt;&lt;/td&gt;".format(image)) html.append("&lt;/tr&gt;&lt;/table&gt;") show_html(''.join(html)) def new_section(title, color="#66aa33", padding="120px"): style = "text-align:center;background:{};padding:{} 10px {} 10px;".format(color,padding,padding) style += "color:#ffffff;font-size:2.55em;line-height:1.2em;" return HTML('&lt;div style="{}"&gt;{}&lt;/div&gt;'.format(style, title)) # Show or hide text HTML(""" &lt;style&gt; .section { display:flex;align-items:center;justify-content:center;width:100%; height:400px; background:#6a3;color:#eee;font-size:275%; } .showhide_label { display:block; cursor:pointer; } .showhide { position: absolute; left: -999em; } .showhide + div { display: none; } .showhide:checked + div { display: block; } .shown_or_hidden { font-size:85%; } &lt;/style&gt; """) # Plotting style plt.rc("axes", titlesize=20, labelsize=15, linewidth=.25, edgecolor='#444444') sns.set_context("notebook", font_scale=1.2, rc={}) %matplotlib inline %config InlineBackend.figure_format = 'retina' # or 'svg'</span></span></code> </pre> </div></div><br><h1>  Laboratoires cloud </h1><br>  Comme AWS ou tout autre cloud informatique, le laboratoire de cloud poss√®de des √©quipements de biologie mol√©culaire, ainsi que des robots qu'il loue sur Internet.  Vous pouvez envoyer des instructions √† vos robots en cliquant sur quelques boutons de l'interface ou en √©crivant du code qui les programme vous-m√™me.  Il n'est pas n√©cessaire d'√©crire vos propres protocoles, comme je le ferai ici, une partie importante de la biologie mol√©culaire est des t√¢ches de routine standard, il est donc g√©n√©ralement pr√©f√©rable de s'appuyer sur un protocole √©tranger fiable qui a montr√© une bonne interaction avec les robots. <br><br>  R√©cemment, un certain nombre d'entreprises dot√©es de laboratoires cloud sont apparues: <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Transcriptic</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Autodesk Wet Lab Accelerator</a> (b√™ta et construit sur la base de Transcriptic), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Arcturus BioCloud</a> (b√™ta), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Emerald Cloud Lab</a> (b√™ta), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Synthego</a> (n'a pas encore commenc√©).  Il existe m√™me des entreprises construites au-dessus de laboratoires cloud tels que <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Desktop Genetics</a> , sp√©cialis√© dans CRISPR.  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Des articles scientifiques</a> sur l'utilisation des laboratoires cloud dans la vraie science commencent √† para√Ætre. <br><br>  Au moment d'√©crire ces lignes, seul Transcriptic est dans le domaine public, nous allons donc l'utiliser.  Si je comprends bien, la plupart des activit√©s de transcription reposent sur l'automatisation de protocoles courants, et l'√©criture de vos propres protocoles en Python (comme je le ferai dans cet article) est moins courante. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/94e/38f/0a6/94e38f0a6f9ea9af3509a0278242b9b4.jpg"><br>  <i><font color="gray">¬´Cellule de travail¬ª transcriptique avec r√©frig√©rateurs en bas et divers √©quipements de laboratoire sur le stand</font></i> <br><br>  Je donnerai aux robots Transcriptic des instructions sur l' <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">auto-protocole</a> .  Autoprotocol est un langage bas√© sur JSON pour √©crire des protocoles pour les robots de laboratoire (et les humains, pour ainsi dire).  Le protocole automatique est principalement r√©alis√© sur <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">cette biblioth√®que Python</a> .  Le langage a √©t√© cr√©√© √† l'origine et est toujours pris en charge par Transcriptic, mais, si je comprends bien, il est compl√®tement ouvert.  Il y a une bonne <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">documentation</a> . <br><br>  Une id√©e int√©ressante est que vous pouvez √©crire des instructions pour les personnes dans des laboratoires distants, par exemple en Chine ou en Inde, sur l'auto-protocole et potentiellement obtenir des avantages en utilisant √† la fois des personnes (leur jugement) et des robots (manque de jugement).  Nous devons mentionner <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">ici protocoles.io</a> , il s'agit d'une tentative de normalisation des protocoles pour am√©liorer la reproductibilit√©, mais pour les humains, pas pour les robots. <br><br><pre> <code class="json hljs"><span class="hljs-string"><span class="hljs-string">"instructions"</span></span>: [ { <span class="hljs-attr"><span class="hljs-attr">"to"</span></span>: [ { <span class="hljs-attr"><span class="hljs-attr">"well"</span></span>: <span class="hljs-string"><span class="hljs-string">"water/0"</span></span>, <span class="hljs-attr"><span class="hljs-attr">"volume"</span></span>: <span class="hljs-string"><span class="hljs-string">"500.0:microliter"</span></span> } ], <span class="hljs-attr"><span class="hljs-attr">"op"</span></span>: <span class="hljs-string"><span class="hljs-string">"provision"</span></span>, <span class="hljs-attr"><span class="hljs-attr">"resource_id"</span></span>: <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span> }, ... ]</code> </pre> <br>  <i><font color="gray">Exemple de fragment autoprotocole</font></i> <br><br><h1>  Param√®tres Python pour la biologie mol√©culaire </h1><br>  En plus d'importer des biblioth√®ques standard, j'aurai besoin de certains utilitaires biologiques mol√©culaires sp√©cifiques.  Ce code est principalement destin√© aux protocoles automatiques et transcriptiques. <br><br>  Le concept de ¬´volume mort¬ª se retrouve souvent dans le code.  Cela signifie la derni√®re goutte de liquide que les robots Transcriptic ne peuvent pas prendre avec une pipette des tubes (car ils ne peuvent pas la voir!).  Vous devez passer beaucoup de temps pour vous assurer que les flacons contiennent suffisamment de mati√®re. <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">import</span></span> autoprotocol <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Unit <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.container <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Container <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.protocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Protocol <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.protocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Ref <span class="hljs-comment"><span class="hljs-comment"># "Link a ref name (string) to a Container instance." import requests import logging # Transcriptic authorization org_name = 'hgbrian' tsc_headers = {k:v for k,v in json.load(open("auth.json")).items() if k in ["X_User_Email","X_User_Token"]} # Transcriptic-specific dead volumes _dead_volume = [("96-pcr",3), ("96-flat",25), ("96-flat-uv",25), ("96-deep",15), ("384-pcr",2), ("384-flat",5), ("384-echo",15), ("micro-1.5",15), ("micro-2.0",15)] dead_volume = {k:Unit(v,"microliter") for k,v in _dead_volume} def init_inventory_well(well, headers=tsc_headers, org_name=org_name): """Initialize well (set volume etc) for Transcriptic""" def _container_url(container_id): return 'https://secure.transcriptic.com/{}/samples/{}.json'.format(org_name, container_id) response = requests.get(_container_url(well.container.id), headers=headers) response.raise_for_status() container = response.json() well_data = container['aliquots'][well.index] well.name = "{}/{}".format(container["label"], well_data['name']) if well_data['name'] is not None else container["label"] well.properties = well_data['properties'] well.volume = Unit(well_data['volume_ul'], 'microliter') if 'ERROR' in well.properties: raise ValueError("Well {} has ERROR property: {}".format(well, well.properties["ERROR"])) if well.volume &lt; Unit(20, "microliter"): logging.warn("Low volume for well {} : {}".format(well.name, well.volume)) return True def touchdown(fromC, toC, durations, stepsize=2, meltC=98, extC=72): """Touchdown PCR protocol generator""" assert 0 &lt; stepsize &lt; toC &lt; fromC def td(temp, dur): return {"temperature":"{:2g}:celsius".format(temp), "duration":"{:d}:second".format(dur)} return [{"cycles": 1, "steps": [td(meltC, durations[0]), td(C, durations[1]), td(extC, durations[2])]} for C in np.arange(fromC, toC-stepsize, -stepsize)] def convert_ug_to_pmol(ug_dsDNA, num_nts): """Convert ug dsDNA to pmol""" return float(ug_dsDNA)/num_nts * (1e6 / 660.0) def expid(val): """Generate a unique ID per experiment""" return "{}_{}".format(experiment_name, val) def ¬µl(microliters): """Unicode function name for creating microliter volumes""" return Unit(microliters,"microliter")</span></span></code> </pre> </div></div><br><h1>  Synth√®se d'ADN et biologie synth√©tique </h1><br>  Malgr√© sa connexion avec la biologie synth√©tique moderne, la synth√®se d'ADN est une technologie assez ancienne.  Pendant des d√©cennies, nous avons pu fabriquer des oligonucl√©otides (c'est-√†-dire des s√©quences d'ADN jusqu'√† 200 bases).  Cependant, c'√©tait toujours cher, et la chimie n'a jamais permis de longues s√©quences d'ADN.  R√©cemment, il est devenu possible √† un prix raisonnable de synth√©tiser des g√®nes entiers (jusqu'√† des milliers de bases).  Cette r√©alisation ouvre v√©ritablement l'√®re de la ¬´biologie synth√©tique¬ª. <br><br>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">La g√©nomique synth√©tique de</a> Craig Venter <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">a</a> pouss√© la biologie synth√©tique le plus loin en <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">synth√©tisant un organisme entier</a> - plus d'un million de bases de long.  √Ä mesure que la longueur de l'ADN augmente, le probl√®me n'est plus la synth√®se, mais l'assemblage (c'est-√†-dire l'assemblage de s√©quences d'ADN synth√©tis√©es).  Avec chaque assemblage, vous pouvez doubler la longueur d'ADN (ou plus), donc apr√®s une dizaine d'it√©rations, vous obtenez une <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">mol√©cule assez longue</a> !  La distinction entre synth√®se et assemblage devrait bient√¥t devenir claire pour l'utilisateur final. <br><br><h1>  La loi de Moore? </h1><br>  Le prix de la synth√®se d'ADN baisse assez rapidement, passant de plus de 0,30 $ il y a deux ans √† environ 0,10 $ aujourd'hui, mais elle se d√©veloppe plus comme des bact√©ries que comme des processeurs.  En revanche, les prix du s√©quen√ßage de l'ADN baissent plus rapidement que la loi de Moore.  Un objectif de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">0,02 $ par base est d√©fini comme un point d'inflexion</a> o√π vous pouvez remplacer de nombreuses manipulations d'ADN chronophages par une simple synth√®se.  Par exemple, √† ce prix, vous pouvez synth√©tiser un plasmide entier de 3 ko pour <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">60 $</a> et sauter un tas de biologie mol√©culaire.  J'esp√®re que nous y parviendrons dans quelques ann√©es. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/373/b42/050/373b42050b870646857266bb3e07c7ed.png"><br>  <i><font color="gray">Prix ‚Äã‚Äãde synth√®se d'ADN compar√©s aux prix de s√©quen√ßage d'ADN, prix pour 1 base (Carlson, 2014)</font></i> <br><br><h1>  Entreprises de synth√®se d'ADN </h1><br>  Il existe plusieurs grandes entreprises dans le domaine de la synth√®se d'ADN: IDT est le plus grand producteur d'oligonucl√©otides et peut √©galement produire des ¬´fragments de g√®nes¬ª ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">gBlocks</a> ) plus longs (jusqu'√† 2 <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">kb</a> ).  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Gen9</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Twist</a> et <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">DNA 2.0</a> se sp√©cialisent g√©n√©ralement dans les s√©quences d'ADN plus longues - ce sont des soci√©t√©s de synth√®se de g√®nes.  Il existe √©galement de nouvelles soci√©t√©s int√©ressantes, telles que <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Cambrian Genomics</a> et <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Genesis DNA</a> , qui travaillent sur des m√©thodes de synth√®se de nouvelle g√©n√©ration. <br><br>  D'autres soci√©t√©s, comme <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Amyris</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Zymergen</a> et <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Ginkgo Bioworks</a> , utilisent l'ADN synth√©tis√© par ces soci√©t√©s pour travailler au niveau du corps.  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">La g√©nomique synth√©tique le</a> fait aussi, mais elle synth√©tise l'ADN lui-m√™me. <br><br>  Ginkgo a r√©cemment <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">conclu un accord avec Twist</a> pour faire 100 millions de bases: le plus gros accord que j'ai vu.  Cela prouve que nous vivons dans le futur, Twist a m√™me annonc√© un code promotionnel sur Twitter: lorsque vous achetez 10 millions de bases d'ADN (presque tout le g√©nome de la levure!), Vous obtenez 10 millions suppl√©mentaires gratuitement. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/e07/bfb/ac9/e07bfbac9c53ddb79634b63efc3fc0e7.png"><br>  <i><font color="gray"><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Offre</a> Twitter <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Twist</a> Niche</font></i> <br><br><h1>  Premi√®re partie: Conception de l'exp√©rience </h1><br><h3>  Prot√©ine fluorescente verte </h3><br>  Dans cette exp√©rience, nous synth√©tisons une s√©quence d'ADN pour une simple <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">prot√©ine fluorescente verte</a> (GFP).  La prot√©ine GFP a √©t√© d√©couverte pour la premi√®re fois dans une <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">m√©duse</a> fluorescente sous lumi√®re ultraviolette.  Il s'agit d'une prot√©ine extr√™mement utile car il est facile de d√©tecter son expression simplement en mesurant la fluorescence.  Il existe des options GFP qui produisent du jaune, du rouge, de l'orange et d'autres couleurs. <br><br>  Il est int√©ressant de voir comment diverses mutations affectent la couleur d'une prot√©ine, et c'est un probl√®me d'apprentissage automatique potentiellement int√©ressant.  Plus r√©cemment, il faudrait passer beaucoup de temps en laboratoire pour cela, mais maintenant je vais vous montrer que c'est (presque) aussi simple que d'√©diter un fichier texte! <br><br>  Techniquement, mon GFP est une option Super Folder (sfGFP) avec quelques mutations pour am√©liorer la qualit√©. <br><br><img src="https://habrastorage.org/webt/tb/tm/_v/tbtm_vcly3elxlwhbcoknhsjpuo.jpeg"><br>  <i><font color="gray">Dans <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">superfolder-GFP</a> (sfGFP), certaines mutations lui conf√®rent certaines propri√©t√©s utiles.</font></i> <br><br><img src="https://habrastorage.org/getpro/habr/post_images/645/232/c67/645232c67631f484ed6e8b30f6fa91eb.jpg"><br>  <i><font color="gray">Structure GFP (visualis√©e √† l'aide de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">PV</a> )</font></i> <br><br><h3>  Synth√®se de GFP dans Twist </h3><br>  J'ai eu la chance de participer au programme de test alpha de Twist, j'ai donc utilis√© leur service de synth√®se d'ADN (ils ont gentiment pass√© ma petite commande - merci, Twist!).  Il s'agit d'une nouvelle entreprise dans notre domaine, avec un nouveau processus de synth√®se simplifi√©.  Leurs prix sont d'environ <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">0,10 $ par base ou moins</a> , mais ils sont <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">toujours en version b√™ta</a> , et le programme alpha auquel j'ai particip√© a √©t√© ferm√©.  Twist a recueilli environ 150 millions de dollars, de sorte que leur technologie est vivante. <br><br>  J'ai envoy√© ma s√©quence d'ADN √† Twist sous forme de feuille de calcul Excel (il n'y a pas encore d'API, mais je suppose que ce sera bient√¥t), et ils ont envoy√© l'ADN synth√©tis√© directement dans ma bo√Æte dans le laboratoire de transcription (j'ai √©galement utilis√© IDT pour la synth√®se, mais ils n'ont pas envoy√© ADN en transcriptic, ce qui g√¢che un peu le plaisir). <br><br>  De toute √©vidence, ce processus n'est pas encore devenu un cas d'utilisation typique et n√©cessite un certain support, mais il a fonctionn√©, de sorte que l'ensemble du pipeline reste virtuel.  Sans cela, j'aurais probablement besoin d'avoir acc√®s au laboratoire - de nombreuses entreprises n'enverront pas d'ADN ou de r√©actifs √† leur domicile. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/734/378/b0b/734378b0b151d2ed64bbd3ebb6edf0f3.jpg"><br>  <i><font color="gray">GFP est inoffensif, donc <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">tout type est mis en √©vidence</a></font></i> <br><br><h3>  Vecteur plasmidique </h3><br>  Pour exprimer cette prot√©ine dans des bact√©ries, le g√®ne doit vivre quelque part, sinon l'ADN synth√©tique codant pour le g√®ne se d√©grade simplement instantan√©ment.  En r√®gle g√©n√©rale, en biologie mol√©culaire, nous utilisons un plasmide, un morceau d'ADN rond qui vit en dehors du g√©nome bact√©rien et exprime des prot√©ines.  Les plasmides sont un moyen pratique pour les bact√©ries de partager des modules fonctionnels utiles et autonomes, tels que la r√©sistance aux antibiotiques.  Il peut y avoir des centaines de plasmides dans une cellule. <br><br>  La terminologie largement utilis√©e est qu'un plasmide est un <b>vecteur</b> et que l'ADN synth√©tique est une insertion (insertion).  Donc, ici, nous essayons de cloner l'insertion dans un vecteur, puis de transformer les bact√©ries √† l'aide du vecteur. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/66c/9ae/d03/66c9aed037ae408c6f642433d9a2a130.jpg"><br>  <i><font color="gray">G√©nome bact√©rien et plasmide (pas √† l'√©chelle!) ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Wikipedia</a> )</font></i> <br><br><h3>  pUC19 </h3><br>  J'ai choisi un plasmide assez standard dans la s√©rie <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">pUC19</a> .  Ce plasmide est tr√®s souvent utilis√©, et puisqu'il est disponible dans le cadre de l'inventaire transcriptique standard, nous n'avons pas besoin de leur envoyer quoi que ce soit. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/4f3/576/8f1/4f35768f17d5ff5506f7d1aca0b064a0.png"><br>  <i><font color="gray">Structure de pUC19: les principaux composants sont le g√®ne de r√©sistance √† l'ampicilline, lacZŒ±, MCS / polylinker et l'origine de la r√©plication (Wikipedia)</font></i> <br><br>  Le PUC19 a une fonction int√©ressante: puisqu'il contient le g√®ne lacZŒ±, vous pouvez utiliser la m√©thode de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">s√©lection bleu-blanc</a> dessus et voir dans quelles colonies l'insertion a r√©ussi.  Deux produits chimiques sont n√©cessaires: <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">IPTG</a> et <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">X-gal</a> , et le circuit fonctionne comme suit: <br><br><ul><li>  L'IPTG induit l'expression de lacZŒ±. <br></li><li>  Si lacZŒ± est d√©sactiv√© via l'ADN ins√©r√© au site de clonage multiple ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">MCS / polylinker</a> ) dans lacZŒ±, alors le plasmide ne peut pas hydrolyser X-gal et ces colonies seront blanches au lieu de bleu. <br></li><li>  Par cons√©quent, une insertion r√©ussie produit des colonies blanches et une insertion √©chou√©e produit des colonies bleues. </li></ul><br><img src="https://habrastorage.org/getpro/habr/post_images/123/aac/aca/123aacacac400c6f96966d2f2caeb918.jpg"><br>  <i><font color="gray">La s√©lection en bleu et blanc montre o√π l'expression lacZŒ± a √©t√© d√©sactiv√©e ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Wikipedia</a> )</font></i> <br><br>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">La documentation openwetware</a> indique: <br><br><blockquote>  E. coli DH5Œ± ne n√©cessite pas d'IPTG pour induire l'expression du promoteur lac, m√™me si un r√©presseur Lac est exprim√© dans la souche.  Le nombre de copies de la plupart des plasmides d√©passe le nombre de r√©presseurs dans les cellules.  Si vous avez besoin d'une expression maximale, ajoutez IPTG √† une concentration finale de 1 mM. </blockquote><br><h1>  S√©quences d'ADN synth√©tique </h1><br><h3>  S√©quence d'ADN SfGFP </h3><br>  Il est facile d'obtenir la s√©quence d'ADN de sfGFP en prenant <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">la s√©quence prot√©ique</a> et en la <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">codant avec des codons</a> appropri√©s pour l'organisme h√¥te (ici, <i>E. coli</i> ).  Il s'agit d'une prot√©ine de taille moyenne avec 236 acides amin√©s, donc √† 10 cents la synth√®se d'ADN co√ªte environ <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">70 $</a> par base. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/296/9da/a41/2969daa417296779220366c36c02a0e5.png"><br>  <i><font color="gray">Wolfram Alpha, calcul des co√ªts de synth√®se</font></i> <br><br>  Les 12 premi√®res bases de notre sfGFP sont la <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">s√©quence Shine-Delgarno</a> , que j'ai ajout√©e moi-m√™me, qui devrait th√©oriquement augmenter l'expression (AGGAGGACAGCT, puis ATG ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">start codon</a> ) d√©marre la prot√©ine).  Selon un outil de calcul d√©velopp√© par <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Salis Lab</a> ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">diapositives de cours</a> ), on peut s'attendre √† une expression moyenne √† √©lev√©e de notre prot√©ine (taux d'initiation de la traduction de 10 000 ¬´unit√©s arbitraires¬ª). <br><br><pre> <code class="python hljs">sfGFP_plus_SD = clean_seq(<span class="hljs-string"><span class="hljs-string">""" AGGAGGACAGCTATGTCGAAAGGAGAAGAACTGTTTACCGGTGTGGTTCCGATTCTGGTAGAACTGGA TGGGGACGTGAACGGCCATAAATTTAGCGTCCGTGGTGAGGGTGAAGGGGATGCCACAAATGGCAAAC TTACCCTTAAATTCATTTGCACTACCGGCAAGCTGCCGGTCCCTTGGCCGACCTTGGTCACCACACTG ACGTACGGGGTTCAGTGTTTTTCGCGTTATCCAGATCACATGAAACGCCATGACTTCTTCAAAAGCGC CATGCCCGAGGGCTATGTGCAGGAACGTACGATTAGCTTTAAAGATGACGGGACCTACAAAACCCGGG CAGAAGTGAAATTCGAGGGTGATACCCTGGTTAATCGCATTGAACTGAAGGGTATTGATTTCAAGGAA GATGGTAACATTCTCGGTCACAAATTAGAATACAACTTTAACAGTCATAACGTTTATATCACCGCCGA CAAACAGAAAAACGGTATCAAGGCGAATTTCAAAATCCGGCACAACGTGGAGGACGGGAGTGTACAAC TGGCCGACCATTACCAGCAGAACACACCGATCGGCGACGGCCCGGTGCTGCTCCCGGATAATCACTAT TTAAGCACCCAGTCAGTGCTGAGCAAAGATCCGAACGAAAAACGTGACCATATGGTGCTGCTGGAGTT CGTGACCGCCGCGGGCATTACCCATGGAATGGATGAACTGTATAAA"""</span></span>) print(<span class="hljs-string"><span class="hljs-string">"Read in sfGFP plus Shine-Dalgarno: {} bases long"</span></span>.format(len(sfGFP_plus_SD))) sfGFP_aas = clean_aas(<span class="hljs-string"><span class="hljs-string">"""MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPTLVTTLTYG VQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKN GIKANFKIRHNVEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK"""</span></span>) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> sfGFP_plus_SD[<span class="hljs-number"><span class="hljs-number">12</span></span>:].translate() == sfGFP_aas print(<span class="hljs-string"><span class="hljs-string">"Translation matches protein with accession 532528641"</span></span>)</code> </pre> <br><pre>  Lire dans sfGFP plus Shine-Dalgarno: 726 bases de long
 La traduction correspond √† la prot√©ine avec l'accession 532528641 </pre><br><h3>  S√©quence d'ADN PUC19 </h3><br>  Tout d'abord, je v√©rifie que la <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">s√©quence pUC19 que j'ai t√©l√©charg√©e de l'ON√â a la</a> bonne longueur et qu'elle inclut le <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">polylinker</a> attendu. <br><br><pre> <code class="python hljs">pUC19_fasta = !cat puc19fsa.txt pUC19_fwd = clean_seq(<span class="hljs-string"><span class="hljs-string">''</span></span>.join(pUC19_fasta[<span class="hljs-number"><span class="hljs-number">1</span></span>:])) pUC19_rev = pUC19_fwd.reverse_complement() <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACGT"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> pUC19_fwd) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> len(pUC19_fwd) == <span class="hljs-number"><span class="hljs-number">2686</span></span> pUC19_MCS = clean_seq(<span class="hljs-string"><span class="hljs-string">"GAATTCGAGCTCGGTACCCGGGGATCCTCTAGAGTCGACCTGCAGGCATGCAAGCTT"</span></span>) print(<span class="hljs-string"><span class="hljs-string">"Read in pUC19: {} bases long"</span></span>.format(len(pUC19_fwd))) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> pUC19_fwd print(<span class="hljs-string"><span class="hljs-string">"Found MCS/polylinker"</span></span>)</code> </pre> <br><pre>  Lire en pUC19: 2686 bases de long
 Trouv√© MCS / polylinker </pre><br>  Nous faisons quelques QC de base pour nous assurer que EcoRI et BamHI ne sont pr√©sents dans pUC19 qu'une seule fois (les enzymes de restriction suivantes sont disponibles dans l'inventaire transcriptique par d√©faut: <i>PstI</i> , <i>PvuII</i> , <i>EcoRI</i> , <i>BamHI</i> , <i>BbsI</i> , <i>BsmBI</i> ). <br><br><pre> <code class="python hljs">REs = {<span class="hljs-string"><span class="hljs-string">"EcoRI"</span></span>:<span class="hljs-string"><span class="hljs-string">"GAATTC"</span></span>, <span class="hljs-string"><span class="hljs-string">"BamHI"</span></span>:<span class="hljs-string"><span class="hljs-string">"GGATTC"</span></span>} <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> rename, res <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> REs.items(): <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> (pUC19_fwd.find(res) == pUC19_fwd.rfind(res) <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> pUC19_rev.find(res) == pUC19_rev.rfind(res)) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> (pUC19_fwd.find(res) == <span class="hljs-number"><span class="hljs-number">-1</span></span> <span class="hljs-keyword"><span class="hljs-keyword">or</span></span> pUC19_rev.find(res) == <span class="hljs-number"><span class="hljs-number">-1</span></span> <span class="hljs-keyword"><span class="hljs-keyword">or</span></span> pUC19_fwd.find(res) == len(pUC19_fwd) - pUC19_rev.find(res) - len(res)) print(<span class="hljs-string"><span class="hljs-string">"Asserted restriction enzyme sites present only once: {}"</span></span>.format(REs.keys()))</code> </pre> <br>  Maintenant, nous regardons la s√©quence lacZŒ± et v√©rifions qu'il n'y a rien d'inattendu.  Par exemple, il doit commencer par Met et se terminer par un codon stop.  Il est √©galement facile de confirmer qu'il s'agit de l'ORF lacZŒ± complet de 324 pb en chargeant la s√©quence pUC19 dans le <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">visualiseur Snapgene</a> gratuit. <br><br><pre> <code class="python hljs">lacZ = pUC19_rev[<span class="hljs-number"><span class="hljs-number">2217</span></span>:<span class="hljs-number"><span class="hljs-number">2541</span></span>] print(<span class="hljs-string"><span class="hljs-string">"lacZŒ± sequence:\t{}"</span></span>.format(lacZ)) print(<span class="hljs-string"><span class="hljs-string">"r_MCS sequence:\t{}"</span></span>.format(pUC19_MCS.reverse_complement())) lacZ_p = lacZ.translate() <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> lacZ_p[<span class="hljs-number"><span class="hljs-number">0</span></span>] == <span class="hljs-string"><span class="hljs-string">"M"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> <span class="hljs-keyword"><span class="hljs-keyword">not</span></span> <span class="hljs-string"><span class="hljs-string">"*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> lacZ_p[:<span class="hljs-number"><span class="hljs-number">-1</span></span>] <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> lacZ_p[<span class="hljs-number"><span class="hljs-number">-1</span></span>] == <span class="hljs-string"><span class="hljs-string">"*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS.reverse_complement() <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> lacZ <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS.reverse_complement() == pUC19_rev[<span class="hljs-number"><span class="hljs-number">2234</span></span>:<span class="hljs-number"><span class="hljs-number">2291</span></span>] print(<span class="hljs-string"><span class="hljs-string">"Found MCS once in lacZ sequence"</span></span>)</code> </pre> <br><pre>  s√©quence lacZ: ATGACCATGATTACGCCAAGCTTGCATGCCTGCAGGTCGACTCTAGAGGATCCCCGGGTACCGAGCTCGAATTCACTGGCCGTCGTTTTACAACGTCGTGACTGGGAAAACCCTGGCGTTACCCAACTTAATCGCCTTGCAGCACATCCCCCTTTCGCCAGCTGGCGTAATAGCGAAGAGGCCCGCACCGATCGCCCTTCCCAACAGTTGCGCAGCCTGAATGGCGAATGGCGCCTGATGCGGTATTTTCTCCTTACGCATCTGTGCGGTATTTCACACCGCATATGGTGCACTCTCAGTACAATCTGCTCTGATGCCGCATAG
 s√©quence r_MCS: AAGCTTGCATGCCTGCAGGTCGACTCTAGAGGATCCCCGGGTACCGAGCTCGAATTC
 MCS trouv√© une fois dans la s√©quence lacZ </pre><br><h1>  Assemblage Gibson </h1><br>  L'assemblage d'ADN signifie simplement des fragments de r√©ticulation.  Habituellement, vous collectez plusieurs fragments d'ADN dans un segment plus long, puis vous les clonez dans un plasmide ou un g√©nome.  Dans cette exp√©rience, je veux cloner un segment d'ADN dans le plasmide pUC19 sous le promoteur lac pour l'expression dans <i>E. coli</i> . <br><br>  Il existe de nombreuses m√©thodes de clonage (par exemple <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">NEB</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">openwetware</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">addgene</a> ).  Ici, je vais utiliser l'assemblage Gibson ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">d√©velopp√© par Daniel Gibson</a> dans Synthetic Genomics en 2009), qui n'est pas n√©cessairement la m√©thode la moins ch√®re, mais simple et flexible.  Il vous suffit de mettre l'ADN que vous souhaitez collecter (avec le chevauchement appropri√©) dans un tube √† essai avec le Gibson Assembly Master Mix, et il s'assemblera! <br><br><img src="https://habrastorage.org/getpro/habr/post_images/8df/946/db5/8df946db53b725f89277be267d2e3ab9.jpg"><br>  <i><font color="gray">Revue de l'Assembl√©e Gibson ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">ON√â</a> )</font></i> <br><br><h3>  Mat√©riel d'origine </h3><br>  On commence avec 100 ng d'ADN synth√©tique dans 10 ¬µl de liquide.  Cela √©quivaut √† 0,21 picomole d'ADN ou √† une concentration de 10 ng / Œºl. <br><br><pre> <code class="python hljs">pmol_sfgfp = convert_ug_to_pmol(<span class="hljs-number"><span class="hljs-number">0.1</span></span>, len(sfGFP_plus_SD)) print(<span class="hljs-string"><span class="hljs-string">"Insert: 100ng of DNA of length {:4d} equals {:.2f} pmol"</span></span>.format(len(sfGFP_plus_SD), pmol_sfgfp))</code> </pre> <br><pre>  Insert: 100ng d'ADN de longueur 726 est √©gal √† 0,21 pmol </pre><br>  Selon <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">le protocole d'assemblage</a> de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">l'ON√â</a> , c'est assez de mat√©riel source: <br><br><blockquote>  L'ON√â recommande un total de 0,02 √† 0,5 picomole de fragments d'ADN lorsque 1 ou 2 fragments sont assembl√©s dans le vecteur, ou 0,2 √† 1,0 picomole de fragments d'ADN lorsque 4 √† 6 fragments sont collect√©s. <br><br>  0,02-0,5 pmoles * X Œºl <br>  * L'efficacit√© de clonage optimis√©e est de 50 √† 100 ng de vecteurs avec 2-3 fois plus d'insertions.  Utilisez 5 fois plus d'insertions si la taille est inf√©rieure √† 200 bps.  Le volume total de fragments de PCR non filtr√©s dans la r√©action d'assemblage de Gibson ne doit pas d√©passer 20%. </blockquote><br><h3>  NEBuilder pour l'assemblage Gibson </h3><br>  NEBuilder de Biolab est un tr√®s bon outil pour cr√©er un protocole de construction Gibson.  Il vous g√©n√®re m√™me un PDF complet de quatre pages avec toutes les informations.  En utilisant cet outil, nous d√©veloppons un protocole pour couper pUC19 avec EcoRI, puis utilisons la PCR [PCR, la r√©action en cha√Æne par polym√©rase permet d'obtenir une augmentation significative des petites concentrations de certains fragments d'ADN dans le mat√©riel biologique - env.  par.] pour ajouter des fragments de la taille appropri√©e √† l'insertion. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/1f5/d52/171/1f5d52171cb60c8e8b4cb83f24b0c8c8.png"><br><br><h1>  Deuxi√®me partie: exp√©rimenter </h1><br>  L'exp√©rience comprend quatre √©tapes: <br><br><ol><li>  R√©action d'insertion de cha√Æne de polym√©rase pour ajouter du mat√©riel avec une s√©quence flanquante. <br></li><li>  Couper un plasmide pour permettre l'insertion. <br></li><li>  Assemblage par insertion de Gibson et plasmides. <br></li><li>  Transformation de bact√©ries √† l'aide du plasmide assembl√©. </li></ol><br><h3>  √âtape 1. Insertion PCR </h3><br>  L'assemblage Gibson d√©pend de la s√©quence d'ADN que vous collectez, ayant une s√©quence qui se chevauchent (voir le protocole NEB avec les instructions d√©taill√©es ci-dessus).  En plus d'une simple amplification, la PCR vous permet √©galement d'ajouter une s√©quence d'ADN flanquante en incluant simplement une s√©quence suppl√©mentaire dans les amorces (peut √©galement √™tre clon√©e en <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">utilisant uniquement OE-PCR</a> ). <br><br>  Nous synth√©tisons les amorces selon le protocole NEB ci-dessus.  J'ai essay√© <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">le protocole Quickstart</a> sur le site Transcriptic, mais il y a toujours <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">une commande auto-</a> <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">protocol</a> .  Le transcriptic lui-m√™me ne synth√©tise pas les oligonucl√©otides, donc apr√®s 1-2 jours d'attente, ces amorces apparaissent comme par magie dans mon inventaire (notez que la partie sp√©cifique au g√®ne des amorces est indiqu√©e en majuscule ci-dessous, mais ce ne sont que des choses cosm√©tiques). <br><br><pre> <code class="python hljs">insert_primers = [<span class="hljs-string"><span class="hljs-string">"aaacgacggccagtgTTTATACAGTTCATCCATTCCATG"</span></span>, <span class="hljs-string"><span class="hljs-string">"cgggtaccgagctcgAGGAGGACAGCTATGTCG"</span></span>]</code> </pre> <br><h3>  Analyse d'amorce </h3><br>  Vous pouvez analyser les propri√©t√©s de ces amorces √† l'aide de l' <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">IDT OligoAnalyzer</a> .    PCR         <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">primer dimer</a> ,   NEB       . <br><br><pre> Gene-specific portion of flank (uppercase)<font></font>
  Melt temperature: 51C, 53.5C<font></font>
Full sequence<font></font>
  Melt temperature: 64.5C, 68.5C<font></font>
  Hairpin: -.4dG, -5dG<font></font>
  Self-dimer: -9dG, -16dG<font></font>
  Heterodimer: -6dG </pre><br>      PCR,     ,        PCR.          (      ),      :       .            .   ,        ‚Äî    . <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">""" PCR overlap extension of sfGFP according to NEB protocol. v5: Use 3/10ths as much primer as the v4 protocol. v6: more complex touchdown pcr procedure. The Q5 temperature was probably too hot v7: more time at low temperature to allow gene-specific part to anneal v8: correct dNTP concentration, real touchdown """</span></span> p = Protocol() <span class="hljs-comment"><span class="hljs-comment"># --------------------------------------------------- # Set up experiment # experiment_name = "sfgfp_pcroe_v8" template_length = 740 _options = {'dilute_primers' : False, # if working stock has not been made 'dilute_template': False, # if working stock has not been made 'dilute_dNTP' : False, # if working stock has not been made 'run_gel' : True, # run a gel to see the plasmid size 'run_absorbance' : False, # check absorbance at 260/280/320 'run_sanger' : False} # sanger sequence the new sequence options = {k for k,v in _options.items() if v is True} # --------------------------------------------------- # Inventory and provisioning # https://developers.transcriptic.com/v1.0/docs/containers # # 'sfgfp2': 'ct17yx8h77tkme', # inventory; sfGFP tube #2, micro-1.5, cold_20 # 'sfgfp_puc19_primer1': 'ct17z9542mrcfv', # inventory; micro-2.0, cold_4 # 'sfgfp_puc19_primer2': 'ct17z9542m5ntb', # inventory; micro-2.0, cold_4 # 'sfgfp_idt_1ngul': 'ct184nnd3rbxfr', # inventory; micro-1.5, cold_4, (ERROR: no template) # inv = { 'Q5 Polymerase': 'rs16pcce8rdytv', # catalog; Q5 High-Fidelity DNA Polymerase 'Q5 Buffer': 'rs16pcce8rmke3', # catalog; Q5 Reaction Buffer 'dNTP Mixture': 'rs16pcb542c5rd', # catalog; dNTP Mixture (25mM?) 'water': 'rs17gmh5wafm5p', # catalog; Autoclaved MilliQ H2O 'sfgfp_pcroe_v5_puc19_primer1_10uM': 'ct186cj5cqzjmr', # inventory; micro-1.5, cold_4 'sfgfp_pcroe_v5_puc19_primer2_10uM': 'ct186cj5cq536x', # inventory; micro-1.5, cold_4 'sfgfp1': 'ct17yx8h759dk4', # inventory; sfGFP tube #1, micro-1.5, cold_20 } # Existing inventory template_tube = p.ref("sfgfp1", id=inv['sfgfp1'], cont_type="micro-1.5", storage="cold_4").well(0) dilute_primer_tubes = [p.ref('sfgfp_pcroe_v5_puc19_primer1_10uM', id=inv['sfgfp_pcroe_v5_puc19_primer1_10uM'], cont_type="micro-1.5", storage="cold_4").well(0), p.ref('sfgfp_pcroe_v5_puc19_primer2_10uM', id=inv['sfgfp_pcroe_v5_puc19_primer2_10uM'], cont_type="micro-1.5", storage="cold_4").well(0)] # New inventory resulting from this experiment dilute_template_tube = p.ref("sfgfp1_0.25ngul", cont_type="micro-1.5", storage="cold_4").well(0) dNTP_10uM_tube = p.ref("dNTP_10uM", cont_type="micro-1.5", storage="cold_4").well(0) sfgfp_pcroe_out_tube = p.ref(expid("amplified"), cont_type="micro-1.5", storage="cold_4").well(0) # Temporary tubes for use, then discarded mastermix_tube = p.ref("mastermix", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) water_tube = p.ref("water", cont_type="micro-1.5", storage="ambient", discard=True).well(0) pcr_plate = p.ref("pcr_plate", cont_type="96-pcr", storage="cold_4", discard=True) if 'run_absorbance' in options: abs_plate = p.ref("abs_plate", cont_type="96-flat", storage="cold_4", discard=True) # Initialize all existing inventory all_inventory_wells = [template_tube] + dilute_primer_tubes for well in all_inventory_wells: init_inventory_well(well) print(well.name, well.volume, well.properties) # ----------------------------------------------------- # Provision water once, for general use # p.provision(inv["water"], water_tube, ¬µl(500)) # ----------------------------------------------------- # Dilute primers 1/10 (100uM-&gt;10uM) and keep at 4C # if 'dilute_primers' in options: for primer_num in (0,1): p.transfer(water_tube, dilute_primer_tubes[primer_num], ¬µl(90)) p.transfer(primer_tubes[primer_num], dilute_primer_tubes[primer_num], ¬µl(10), mix_before=True, mix_vol=¬µl(50)) p.mix(dilute_primer_tubes[primer_num], volume=¬µl(50), repetitions=10) # ----------------------------------------------------- # Dilute template 1/10 (10ng/ul-&gt;1ng/ul) and keep at 4C # OR # Dilute template 1/40 (10ng/ul-&gt;0.25ng/ul) and keep at 4C # if 'dilute_template' in options: p.transfer(water_tube, dilute_template_tube, ¬µl(195)) p.mix(dilute_template_tube, volume=¬µl(100), repetitions=10) # Dilute dNTP to exactly 10uM if 'dilute_DNTP' in options: p.transfer(water_tube, dNTP_10uM_tube, ¬µl(6)) p.provision(inv["dNTP Mixture"], dNTP_10uM_tube, ¬µl(4)) # ----------------------------------------------------- # Q5 PCR protocol # www.neb.com/protocols/2013/12/13/pcr-using-q5-high-fidelity-dna-polymerase-m0491 # # 25ul reaction # ------------- # Q5 reaction buffer 5 ¬µl # Q5 polymerase 0.25 ¬µl # 10mM dNTP 0.5 ¬µl -- 1¬µl = 4x12.5mM # 10uM primer 1 1.25 ¬µl # 10uM primer 2 1.25 ¬µl # 1pg-1ng Template 1 ¬µl -- 0.5 or 1ng/ul concentration # ------------------------------- # Sum 9.25 ¬µl # # # Mastermix tube will have 96ul of stuff, leaving space for 4x1ul aliquots of template p.transfer(water_tube, mastermix_tube, ¬µl(64)) p.provision(inv["Q5 Buffer"], mastermix_tube, ¬µl(20)) p.provision(inv['Q5 Polymerase'], mastermix_tube, ¬µl(1)) p.transfer(dNTP_10uM_tube, mastermix_tube, ¬µl(1), mix_before=True, mix_vol=¬µl(2)) p.transfer(dilute_primer_tubes[0], mastermix_tube, ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(dilute_primer_tubes[1], mastermix_tube, ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.mix(mastermix_tube, volume="48:microliter", repetitions=10) # Transfer mastermix to pcr_plate without template p.transfer(mastermix_tube, pcr_plate.wells(["A1","B1","C1"]), ¬µl(24)) p.transfer(mastermix_tube, pcr_plate.wells(["A2"]), ¬µl(24)) # acknowledged dead volume problems p.mix(pcr_plate.wells(["A1","B1","C1","A2"]), volume=¬µl(12), repetitions=10) # Finally add template p.transfer(template_tube, pcr_plate.wells(["A1","B1","C1"]), ¬µl(1)) p.mix(pcr_plate.wells(["A1","B1","C1"]), volume=¬µl(12.5), repetitions=10) # --------------------------------------------------------- # Thermocycle with Q5 and hot start # 61.1 annealing temperature is recommended by NEB protocol # p.seal is enforced by transcriptic # extension_time = int(max(2, np.ceil(template_length * (11.0/1000)))) assert 0 &lt; extension_time &lt; 60, "extension time should be reasonable for PCR" cycles = [{"cycles": 1, "steps": [{"temperature": "98:celsius", "duration": "30:second"}]}] + \ touchdown(70, 61, [8, 25, extension_time], stepsize=0.5) + \ [{"cycles": 16, "steps": [{"temperature": "98:celsius", "duration": "8:second"}, {"temperature": "61.1:celsius", "duration": "25:second"}, {"temperature": "72:celsius", "duration": "{:d}:second".format(extension_time)}]}, {"cycles": 1, "steps": [{"temperature": "72:celsius", "duration": "2:minute"}]}] p.seal(pcr_plate) p.thermocycle(pcr_plate, cycles, volume=¬µl(25)) # -------------------------------------------------------- # Run a gel to hopefully see a 740bp fragment # if 'run_gel' in options: p.unseal(pcr_plate) p.mix(pcr_plate.wells(["A1","B1","C1","A2"]), volume=¬µl(12.5), repetitions=10) p.transfer(pcr_plate.wells(["A1","B1","C1","A2"]), pcr_plate.wells(["D1","E1","F1","D2"]), [¬µl(2), ¬µl(4), ¬µl(8), ¬µl(8)]) p.transfer(water_tube, pcr_plate.wells(["D1","E1","F1","D2"]), [¬µl(18),¬µl(16),¬µl(12),¬µl(12)], mix_after=True, mix_vol=¬µl(10)) p.gel_separate(pcr_plate.wells(["D1","E1","F1","D2"]), ¬µl(20), "agarose(10,2%)", "ladder1", "10:minute", expid("gel")) #--------------------------------------------------------- # Absorbance dilution series. Take 1ul out of the 25ul pcr plate wells # if 'run_absorbance' in options: p.unseal(pcr_plate) abs_wells = ["A1","B1","C1","A2","B2","C2","A3","B3","C3"] p.transfer(water_tube, abs_plate.wells(abs_wells[0:6]), ¬µl(10)) p.transfer(water_tube, abs_plate.wells(abs_wells[6:9]), ¬µl(9)) p.transfer(pcr_plate.wells(["A1","B1","C1"]), abs_plate.wells(["A1","B1","C1"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) p.transfer(abs_plate.wells(["A1","B1","C1"]), abs_plate.wells(["A2","B2","C2"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) p.transfer(abs_plate.wells(["A2","B2","C2"]), abs_plate.wells(["A3","B3","C3"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) for wavelength in [260, 280, 320]: p.absorbance(abs_plate, abs_plate.wells(abs_wells), "{}:nanometer".format(wavelength), exp_id("abs_{}".format(wavelength)), flashes=25) # ----------------------------------------------------------------------------- # Sanger sequencing: https://developers.transcriptic.com/docs/sanger-sequencing # "Each reaction should have a total volume of 15 ¬µl and we recommend the following composition of DNA and primer: # PCR product (40 ng), primer (1 ¬µl of a 10 ¬µM stock)" # # By comparing to the gel ladder concentration (175ng/lane), it looks like 5ul of PCR product has approximately 30ng of DNA # if 'run_sanger' in options: p.unseal(pcr_plate) seq_wells = ["G1","G2"] for primer_num, seq_well in [(0, seq_wells[0]),(1, seq_wells[1])]: p.transfer(dilute_primer_tubes[primer_num], pcr_plate.wells([seq_well]), ¬µl(1), mix_before=True, mix_vol=¬µl(50)) p.transfer(pcr_plate.wells(["A1"]), pcr_plate.wells([seq_well]), ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(water_tube, pcr_plate.wells([seq_well]), ¬µl(9)) p.mix(pcr_plate.wells(seq_wells), volume=¬µl(7.5), repetitions=10) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[0]).indices(), expid("seq1")) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[1]).indices(), expid("seq2")) # ------------------------------------------------------------------------- # Then consolidate to one tube. Leave at least 3ul dead volume in each tube # remaining_volumes = [well.volume - dead_volume['96-pcr'] for well in pcr_plate.wells(["A1","B1","C1"])] print("Consolidated volume", sum(remaining_volumes, ¬µl(0))) p.consolidate(pcr_plate.wells(["A1","B1","C1"]), sfgfp_pcroe_out_tube, remaining_volumes, allow_carryover=True) uprint("\nProtocol 1. Amplify the insert (oligos previously synthesized)") jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> WARNING:root:Low volume for well sfGFP 1 /sfGFP 1 : 2.0:microliter </pre><br><pre> sfGFP 1 /sfGFP 1 2.0:microliter {'dilution': '0.25ng/ul'}<font></font>
sfgfp_pcroe_v5_puc19_primer1_10uM 75.0:microliter {}<font></font>
sfgfp_pcroe_v5_puc19_primer2_10uM 75.0:microliter {}<font></font>
Consolidated volume 52.0:microliter<font></font>
<font></font>
Protocol 1. Amplify the insert (oligos previously synthesized)<font></font>
---------------------------------------------------------------<font></font>
<font></font>
  <font></font>
‚úì Protocol analyzed<font></font>
  11 instructions<font></font>
  8 containers<font></font>
  Total Cost: $32.18<font></font>
  Workcell Time: $4.32<font></font>
  Reagents &amp; Consumables: $27.86 </pre><br><h1> : PCR  </h1><br><h3>     </h3><br>      <b></b>     (   )   <b></b> ( ).    ,      ,     . <br><br>      D1, E1, F1   2 , 4   8   .               (50      ). ,     . <br><br>          <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">GelEval</a> ,   ,    ,    ,    .                .            GelEval  40 /. <br><br>  ,   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">    </a> ,    dNTP  , ,     12,5   ,     6   740bp  25 .         GelEval  40  x 25  (1   2 ),        ,       . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/74a/f5d/284/74af5d28463dbbd2279353a4651087f6.jpg"><br> <i><font color="gray">- EcoRI- pUC19,   (D1, E1, F1),   (D2)</font></i> <br><br><h3>   PCR </h3><br>  Transcriptic          .        ,           . <br><br>   ,     .    35  PCR,             <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> PCR</a> .        ‚Äî   ! ‚Äî     ,   PCR       ,    . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/8dc/9eb/1b8/8dc9eb1b85c382df68cc064583512707.jpg"><br> <i><font color="gray">    PCR:  ,      35   42 </font></i> <br><br><h2>  2.   </h2><br>     sfGFP  pUC19,    .   NEB,        <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">EcoRI</a> .    Transcriptic  ,   :  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">NEB EcoRI  10x CutSmart </a> ,    <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">NEB pUC19</a> . <br><br>  ,     .       ,   Transcriptic      : <br><br><pre> Item ID Amount Concentration Price<font></font>
------------ ------ ------------- ----------------- ------<font></font>
CutSmart 10x B7204S 5 ml 10 X $19.00<font></font>
EcoRI R3101L 50,000 units 20,000 units/ml $225.00<font></font>
pUC19 N3041L 250 ¬µg 1,000 ¬µg/ml $268.00 </pre><br>     NEB: <br><br><blockquote>       .   10X  dH2O    1X.   ,  ,   , , .   50    5  10x NEBuffer       ,   dH2O. <br><br>      ,    1  Œª    1   37¬∞C     50 .  ,   5-10       10-20     1- . <br><br>   1      50 . </blockquote><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Protocol for cutting pUC19 with EcoRI."""</span></span> p = Protocol() experiment_name = <span class="hljs-string"><span class="hljs-string">"puc19_ecori_v3"</span></span> options = {} inv = { <span class="hljs-string"><span class="hljs-string">'water'</span></span>: <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "pUC19": "rs17tcqmncjfsh", # catalog; pUC19; cold_20 "EcoRI": "rs17ta8xftpdk6", # catalog; EcoRI-HF; cold_20 "CutSmart": "rs17ta93g3y85t", # catalog; CutSmart Buffer 10x; cold_20 "ecori_p10x": "ct187v4ea85k2h", # inventory; EcoRI diluted 10x } # Tubes and plates I use then discard re_tube = p.ref("re_tube", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) water_tube = p.ref("water_tube", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) pcr_plate = p.ref("pcr_plate", cont_type="96-pcr", storage="cold_4", discard=True) # The result of the experiment, a pUC19 cut by EcoRI, goes in this tube for storage puc19_cut_tube = p.ref(expid("puc19_cut"), cont_type="micro-1.5", storage="cold_20").well(0) # ------------------------------------------------------------- # Provisioning and diluting. # Diluted EcoRI can be used more than once # p.provision(inv["water"], water_tube, ¬µl(500)) if 'dilute_ecori' in options: ecori_p10x_tube = p.ref("ecori_p10x", cont_type="micro-1.5", storage="cold_20").well(0) p.transfer(water_tube, ecori_p10x_tube, ¬µl(45)) p.provision(inv["EcoRI"], ecori_p10x_tube, ¬µl(5)) else: # All "inventory" (stuff I own at transcriptic) must be initialized ecori_p10x_tube = p.ref("ecori_p10x", id=inv["ecori_p10x"], cont_type="micro-1.5", storage="cold_20").well(0) init_inventory_well(ecori_p10x_tube) # ------------------------------------------------------------- # Restriction enzyme cutting pUC19 # # 50ul total reaction volume for cutting 1ug of DNA: # 5ul CutSmart 10x # 1ul pUC19 (1ug of DNA) # 1ul EcoRI (or 10ul diluted EcoRI, 20 units, &gt;10 units per ug DNA) # p.transfer(water_tube, re_tube, ¬µl(117)) p.provision(inv["CutSmart"], re_tube, ¬µl(15)) p.provision(inv["pUC19"], re_tube, ¬µl(3)) p.mix(re_tube, volume=¬µl(60), repetitions=10) assert re_tube.volume == ¬µl(120) + dead_volume["micro-1.5"] print("Volumes: re_tube:{} water_tube:{} EcoRI:{}".format(re_tube.volume, water_tube.volume, ecori_p10x_tube.volume)) p.distribute(re_tube, pcr_plate.wells(["A1","B1","A2"]), ¬µl(40)) p.distribute(water_tube, pcr_plate.wells(["A2"]), ¬µl(10)) p.distribute(ecori_p10x_tube, pcr_plate.wells(["A1","B1"]), ¬µl(10)) assert all(well.volume == ¬µl(50) for well in pcr_plate.wells(["A1","B1","A2"])) p.mix(pcr_plate.wells(["A1","B1","A2"]), volume=¬µl(25), repetitions=10) # Incubation to induce cut, then heat inactivation of EcoRI p.seal(pcr_plate) p.incubate(pcr_plate, "warm_37", "60:minute", shaking=False) p.thermocycle(pcr_plate, [{"cycles": 1, "steps": [{"temperature": "65:celsius", "duration": "21:minute"}]}], volume=¬µl(50)) # -------------------------------------------------------------- # Gel electrophoresis, to ensure the cutting worked # p.unseal(pcr_plate) p.mix(pcr_plate.wells(["A1","B1","A2"]), volume=¬µl(25), repetitions=5) p.transfer(pcr_plate.wells(["A1","B1","A2"]), pcr_plate.wells(["D1","E1","D2"]), ¬µl(8)) p.transfer(water_tube, pcr_plate.wells(["D1","E1","D2"]), ¬µl(15), mix_after=True, mix_vol=¬µl(10)) assert all(well.volume == ¬µl(20) + dead_volume["96-pcr"] for well in pcr_plate.wells(["D1","E1","D2"])) p.gel_separate(pcr_plate.wells(["D1","E1","D2"]), ¬µl(20), "agarose(10,2%)", "ladder2", "15:minute", expid("gel")) # ---------------------------------------------------------------------------- # Then consolidate all cut plasmid to one tube (puc19_cut_tube). # remaining_volumes = [well.volume - dead_volume['96-pcr'] for well in pcr_plate.wells(["A1","B1"])] print("Consolidated volume: {}".format(sum(remaining_volumes, ¬µl(0)))) p.consolidate(pcr_plate.wells(["A1","B1"]), puc19_cut_tube, remaining_volumes, allow_carryover=True) assert all(tube.volume &gt;= dead_volume['micro-1.5'] for tube in [water_tube, re_tube, puc19_cut_tube, ecori_p10x_tube]) # --------------------------------------------------------------- # Test protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("Protocol {}\n\n{}".format(experiment_name, jprotocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> Volumes: re_tube:135.0:microliter water_tube:383.0:microliter EcoRI:30.0:microliter<font></font>
Consolidated volume: 78.0:microliter<font></font>
<font></font>
  <font></font>
‚úì Protocol analyzed<font></font>
  12 instructions<font></font>
  5 containers<font></font>
  Total Cost: $30.72<font></font>
  Workcell Time: $3.38<font></font>
  Reagents &amp; Consumables: $27.34 </pre><br><h3> :   </h3><br>              ,    .    . <br><br>        <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">¬´¬ª </a> (  1,5    15 !). ,     D1  E1 (     ).     ,      EcoRI   . <br><br>    ,     D1  E1      2,6kb.   D2  :   ,           . <br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Deux photos sur gel sont tr√®s diff√©rentes. </font><font style="vertical-align: inherit;">Cela est d√ª en partie au fait que cette √©tape Transcriptic n'a pas encore √©t√© automatis√©e. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/8f8/b9c/e3c/8f8b9ce3cc42de99099976a3c60a9ada.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Deux gels montrant pUC19 coup√© (2.6kb) dans les bandes D1 et E1, et pUC19 non coup√© en D2</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> √âtape 3. Assemblage Gibson </font></font></h2><br>    ,       ‚Äî    ,    <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> M13</a> (  )          <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">qPCR</a>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> ,  ,   .      ,  ,     ,       . <br><br>     ,   M13  ,         M13. <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Debugging transformation protocol: Gibson assembly followed by qPCR and a gel v2: include v3 Gibson assembly"""</span></span> p = Protocol() options = {} experiment_name = <span class="hljs-string"><span class="hljs-string">"debug_sfgfp_puc19_gibson_seq_v2"</span></span> inv = { <span class="hljs-string"><span class="hljs-string">"water"</span></span> : <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "M13_F" : "rs17tcpqwqcaxe", # catalog; M13 Forward (-41); cold_20 (1ul = 100pmol) "M13_R" : "rs17tcph6e2qzh", # catalog; M13 Reverse (-48); cold_20 (1ul = 100pmol) "SensiFAST_SYBR_No-ROX" : "rs17knkh7526ha", # catalog; SensiFAST SYBR for qPCR "sfgfp_puc19_gibson_v1_clone" : "ct187rzdq9kd7q", # inventory; assembled sfGFP; cold_4 "sfgfp_puc19_gibson_v3_clone" : "ct188ejywa8jcv", # inventory; assembled sfGFP; cold_4 } # --------------------------------------------------------------- # First get my sfGFP pUC19 clones, assembled with Gibson assembly # clone_plate1 = p.ref("sfgfp_puc19_gibson_v1_clone", id=inv["sfgfp_puc19_gibson_v1_clone"], cont_type="96-pcr", storage="cold_4", discard=False) clone_plate2 = p.ref("sfgfp_puc19_gibson_v3_clone", id=inv["sfgfp_puc19_gibson_v3_clone"], cont_type="96-pcr", storage="cold_4", discard=False) water_tube = p.ref("water", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) master_tube = p.ref("master", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) primer_tube = p.ref("primer", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) pcr_plate = p.ref(expid("pcr_plate"), cont_type="96-pcr", storage="cold_4", discard=False) init_inventory_well(clone_plate1.well("A1")) init_inventory_well(clone_plate2.well("A1")) seq_wells = ["B2","B4","B6", # clone_plate1 "D2","D4","D6", # clone_plate2 "F2","F4"] # control # clone_plate2 was diluted 4X (20ul-&gt;80ul), according to NEB instructions assert clone_plate1.well("A1").volume == ¬µl(18), clone_plate1.well("A1").volume assert clone_plate2.well("A1").volume == ¬µl(78), clone_plate2.well("A1").volume # -------------------------------------------------------------- # Provisioning # p.provision(inv["water"], water_tube, ¬µl(500)) # primers, diluted 2X, discarded at the end p.provision(inv["M13_F"], primer_tube, ¬µl(13)) p.provision(inv["M13_R"], primer_tube, ¬µl(13)) p.transfer(water_tube, primer_tube, ¬µl(26), mix_after=True, mix_vol=¬µl(20), repetitions=10) # ------------------------------------------------------------------- # PCR Master mix -- 10ul SYBR mix, plus 1ul each undiluted primer DNA (100pmol) # Also add 15ul of dead volume # p.provision(inv['SensiFAST_SYBR_No-ROX'], master_tube, ¬µl(11+len(seq_wells)*10)) p.transfer(primer_tube, master_tube, ¬µl(4+len(seq_wells)*4)) p.mix(master_tube, volume=¬µl(63), repetitions=10) assert master_tube.volume == ¬µl(127) # 15ul dead volume p.distribute(master_tube, pcr_plate.wells(seq_wells), ¬µl(14), allow_carryover=True) p.distribute(water_tube, pcr_plate.wells(seq_wells), [¬µl(ul) for ul in [5,4,2, 4,2,0, 6,6]], allow_carryover=True) # Template -- starting with some small, unknown amount of DNA produced by Gibson p.transfer(clone_plate1.well("A1"), pcr_plate.wells(seq_wells[0:3]), [¬µl(1),¬µl(2),¬µl(4)], one_tip=True) p.transfer(clone_plate2.well("A1"), pcr_plate.wells(seq_wells[3:6]), [¬µl(2),¬µl(4),¬µl(6)], one_tip=True) assert all(pcr_plate.well(w).volume == ¬µl(20) for w in seq_wells) assert clone_plate1.well("A1").volume == ¬µl(11) assert clone_plate2.well("A1").volume == ¬µl(66) # -------------------------------------------------------------- # qPCR # standard melting curve parameters # p.seal(pcr_plate) p.thermocycle(pcr_plate, [{"cycles": 1, "steps": [{"temperature": "95:celsius","duration": "2:minute"}]}, {"cycles": 40, "steps": [{"temperature": "95:celsius","duration": "5:second"}, {"temperature": "60:celsius","duration": "20:second"}, {"temperature": "72:celsius","duration": "15:second", "read": True}]}], volume=¬µl(20), # volume is optional dataref=expid("qpcr"), dyes={"SYBR": seq_wells}, # dye must be specified (tells transcriptic what aborbance to use?) melting_start="65:celsius", melting_end="95:celsius", melting_increment="0.5:celsius", melting_rate="5:second") # -------------------------------------------------------------- # Gel -- 20ul required # Dilute such that I have 11ul for sequencing # p.unseal(pcr_plate) p.distribute(water_tube, pcr_plate.wells(seq_wells), ¬µl(11)) p.gel_separate(pcr_plate.wells(seq_wells), ¬µl(20), "agarose(8,0.8%)", "ladder1", "10:minute", expid("gel")) # This appears to be a bug in Transcriptic. The actual volume should be 11ul # but it is not updating after running a gel with 20ul. # Primer tube should be equal to dead volume, or it's a waste assert all(pcr_plate.well(w).volume==¬µl(31) for w in seq_wells) assert primer_tube.volume == ¬µl(16) == dead_volume['micro-1.5'] + ¬µl(1) assert water_tube.volume &gt; ¬µl(25) # --------------------------------------------------------------- # Test and run protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> WARNING:root:Low volume for well sfgfp_puc19_gibson_v1_clone/sfgfp_puc19_gibson_v1_clone : 11.0:microliter </pre><br><pre> ‚úì Protocol analyzed<font></font>
  11 instructions<font></font>
  6 containers<font></font>
  Total Cost: $32.09<font></font>
  Workcell Time: $6.98<font></font>
  Reagents &amp; Consumables: $25.11 </pre><br><h3> : qPCR    </h3><br>       qPCR   JSON  Transcriptic API.      <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> ,     . API          ,     . <br><br> -,    : <br><br><pre> <code class="python hljs">project_id, run_id = <span class="hljs-string"><span class="hljs-string">"p16x6gna8f5e9"</span></span>, <span class="hljs-string"><span class="hljs-string">"r18mj3cz3fku7"</span></span> api_url = <span class="hljs-string"><span class="hljs-string">"https://secure.transcriptic.com/hgbrian/{}/runs/{}/data.json"</span></span>.format(project_id, run_id) data_response = requests.get(api_url, headers=tsc_headers) data = data_response.json()</code> </pre> <br>    id,    ¬´¬ª qPCR: <br><br><pre> <code class="python hljs">qpcr_id = data[<span class="hljs-string"><span class="hljs-string">'debug_sfgfp_puc19_gibson_seq_v1_qpcr'</span></span>][<span class="hljs-string"><span class="hljs-string">'id'</span></span>] pp_api_url = <span class="hljs-string"><span class="hljs-string">"https://secure.transcriptic.com/data/{}.json?key=postprocessed_data"</span></span>.format(qpcr_id) data_response = requests.get(pp_api_url, headers=tsc_headers) pp_data = data_response.json()</code> </pre> <br>   Ct ( )   . Ct ‚Äî   ,      .   ,        (, , ,    ). <br><br><pre> <code class="python hljs"><span class="hljs-comment"><span class="hljs-comment"># Simple util to convert wellnum to wellname n_w = {str(wellnum):'ABCDEFGH'[wellnum//12]+str(1+wellnum%12) for wellnum in range(96)} w_n = {v: k for k, v in n_w.items()} ct_vals = {n_w[k]:v for k,v in pp_data["amp0"]["SYBR"]["cts"].items()} ct_df = pd.DataFrame(ct_vals, index=["Ct"]).T ct_df["well"] = ct_df.index f, ax = plt.subplots(figsize=(16,6)) _ = sns.barplot(y="well", x="Ct", data=ct_df)</span></span></code> </pre> <br><img src="https://habrastorage.org/getpro/habr/post_images/32f/a42/de3/32fa42de3ad35392ac0bbf7ae6f595b3.png"><br><br>  ,       D2/4/6 (      ¬´v3¬ª),  B2/4/6 (  ¬´v1¬ª).   v1  v3     ,   v3  4X     NEB,     .      30    (F2, F4),   -,    ,         . <br><br>       qPCR,    . <br><br><pre> <code class="python hljs">f, ax = plt.subplots(figsize=(<span class="hljs-number"><span class="hljs-number">16</span></span>,<span class="hljs-number"><span class="hljs-number">6</span></span>)) ax.set_color_cycle([<span class="hljs-string"><span class="hljs-string">'#fb6a4a'</span></span>, <span class="hljs-string"><span class="hljs-string">'#de2d26'</span></span>, <span class="hljs-string"><span class="hljs-string">'#a50f15'</span></span>, <span class="hljs-string"><span class="hljs-string">'#74c476'</span></span>, <span class="hljs-string"><span class="hljs-string">'#31a354'</span></span>, <span class="hljs-string"><span class="hljs-string">'#006d2c'</span></span>, <span class="hljs-string"><span class="hljs-string">'#08519c'</span></span>, <span class="hljs-string"><span class="hljs-string">'#6baed6'</span></span>]) amp0 = pp_data[<span class="hljs-string"><span class="hljs-string">'amp0'</span></span>][<span class="hljs-string"><span class="hljs-string">'SYBR'</span></span>][<span class="hljs-string"><span class="hljs-string">'baseline_subtracted'</span></span>] _ = [plt.plot(amp0[w_n[well]], label=well) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> well <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> [<span class="hljs-string"><span class="hljs-string">'B2'</span></span>, <span class="hljs-string"><span class="hljs-string">'B4'</span></span>, <span class="hljs-string"><span class="hljs-string">'B6'</span></span>, <span class="hljs-string"><span class="hljs-string">'D2'</span></span>, <span class="hljs-string"><span class="hljs-string">'D4'</span></span>, <span class="hljs-string"><span class="hljs-string">'D6'</span></span>, <span class="hljs-string"><span class="hljs-string">'F2'</span></span>, <span class="hljs-string"><span class="hljs-string">'F4'</span></span>]] _ = ax.set_ylim(<span class="hljs-number"><span class="hljs-number">0</span></span>,) _ = plt.title(<span class="hljs-string"><span class="hljs-string">"qPCR (reds=Gibson v1, greens=Gibson v3, blues=control)"</span></span>) _ = plt.legend(bbox_to_anchor=(<span class="hljs-number"><span class="hljs-number">1</span></span>, <span class="hljs-number"><span class="hljs-number">.75</span></span>), bbox_transform=plt.gcf().transFigure)</code> </pre> <br><img src="https://habrastorage.org/getpro/habr/post_images/88e/d2a/a19/88ed2aa19537b77b3214e6b36f55fdde.png"><br><br>  ,  qPCR  ,               .   v3    ,  v1,    . <br><br><h3> :     </h3><br>    ,      1kb   B2, B4, B6, D2, D4, D6:      (   740bp,   M13 ‚Äî  40bp   ).    .     ,    F2  F4    . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/cf7/509/629/cf750962951b700c8a9333ea1d747a94.jpg"><br> <i><font color="gray">   :    v3     (D2, D4, D6),       qPCR</font></i> <br><br><h2>  4.  </h2><br> <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> ‚Äî       .     <b></b> <i>E. coli</i>   sfGFP-  pUC19. <br><br>      <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Zymo DH5Œ± Mix&amp;Go</a>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> </a> .   ‚Äî    Transcriptic.  ,    ,     ,      ,  .        , ,       . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/951/9a6/a57/9519a6a578f11d26c3c9a2ddda039539.jpg"><br> <i><font color="gray"> Zymo Mix &amp; Go   </font></i> <br><br><h4>    </h4><br>   ‚Äî  ,               .      (¬´     ¬ª),      ,         (¬´,   ¬ª).      ,      . <br><br>        .   ,                  37¬∞C. ,      ,      ,  ,   Transcriptic     ‚Äî   ,      .     , ,       -    ,    . .      . <br><br>    :        (,   ,   Mix&amp;Go );        (,     );        (,   PCR   ). <br><br> ,    , ,     ,         .     ,   ,     ! <br><br><h2>   </h2><br>        ,    ,  ,      pUC19 (. .        sfGFP) . pUC19     ,         ,   . <br><br>       (¬´6-flat¬ª   Transcriptic),    ,  .  ,         , ,  .     . <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Simple transformation protocol: transformation with unaltered pUC19"""</span></span> p = Protocol() experiment_name = <span class="hljs-string"><span class="hljs-string">"debug_sfgfp_puc19_gibson_v1"</span></span> inv = { <span class="hljs-string"><span class="hljs-string">"water"</span></span> : <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "DH5a" : "rs16pbj944fnny", # catalog; Zymo DH5Œ±; cold_80 "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "pUC19" : "rs17tcqmncjfsh", # catalog; pUC19; cold_20 } # Catalog transform_plate = p.ref("transform_plate", cont_type="96-pcr", storage="ambient", discard=True) transform_tube = transform_plate.well(0) # ------------------------------------------------------------------------------------ # Plating transformed bacteria according to Tali's protocol (requires different code!) # http://learn.transcriptic.com/blog/2015/9/9/provisioning-commercial-reagents # Add 1-5ul plasmid and pre-warm culture plates to 37C before starting. # # # Extra inventory for plating # inv["lb-broth-100ug-ml-amp_6-flat"] = "ki17sbb845ssx9" # (kit, not normal ref) from blogpost inv["noAB-amp_6-flat"] = "ki17reefwqq3sq" # kit id inv["LB Miller"] = "rs17bafcbmyrmh" # # Ampicillin and no ampicillin plates # amp_6_flat = Container(None, p.container_type('6-flat')) p.refs["amp_6_flat"] = Ref('amp_6_flat', {"reserve": inv['lb-broth-100ug-ml-amp_6-flat'], "store": {"where": 'cold_4'}}, amp_6_flat) noAB_6_flat = Container(None, p.container_type('6-flat')) p.refs["noAB_6_flat"] = Ref('noAB_6_flat', {"reserve": inv['noAB-amp_6-flat'], "store": {"where": 'cold_4'}}, noAB_6_flat) # # Provision competent bacteria # p.provision(inv["DH5a"], transform_tube, ¬µl(50)) p.provision(inv["pUC19"], transform_tube, ¬µl(2)) # # Heatshock the bacteria to transform using a PCR machine # p.seal(transform_plate) p.thermocycle(transform_plate, [{"cycles": 1, "steps": [{"temperature": "4:celsius", "duration": "5:minute"}]}, {"cycles": 1, "steps": [{"temperature": "37:celsius", "duration": "30:minute"}]}], volume=¬µl(50)) p.unseal(transform_plate) # # Then dilute bacteria and spread onto 6-flat plates # Put more on ampicillin plates for more opportunities to get a colony # p.provision(inv["LB Miller"], transform_tube, ¬µl(355)) p.mix(transform_tube, ¬µl(150), repetitions=5) for i in range(6): p.spread(transform_tube, amp_6_flat.well(i), ¬µl(55)) p.spread(transform_tube, noAB_6_flat.well(i), ¬µl(10)) assert transform_tube.volume &gt;= ¬µl(15), transform_tube.volume # # Incubate and image 6-flat plates over 18 hours # for flat_name, flat in [("amp_6_flat", amp_6_flat), ("noAB_6_flat", noAB_6_flat)]: for timepoint in [6,12,18]: p.cover(flat) p.incubate(flat, "warm_37", "6:hour") p.uncover(flat) p.image_plate(flat, mode="top", dataref=expid("{}_t{}".format(flat_name, timepoint))) # --------------------------------------------------------------- # Analyze protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("Protocol {}\n\n{}".format(experiment_name, protocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> ‚úì Protocol analyzed<font></font>
  43 instructions<font></font>
  3 containers<font></font>
  $45.43 </pre><br><h3> :   </h3><br>     ,    ( )      ,     ,   . ,  Transcriptic      ,    . <br><br>    ( )   ,    .       ,   , , ,    55        10     .               .        ,    .        ,       ,         . <br><br> (          ,        ,    ,        <i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">E. coli</font></font></i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">La croissance est beaucoup plus faible sur les plaques d'ampicilline, bien qu'il y ait beaucoup plus de bact√©ries, comme pr√©vu). </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Dans l'ensemble, la transformation a assez bien fonctionn√© pour se poursuivre, bien qu'il y ait quelques d√©fauts. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/296/c9b/7e2/296c9b7e28b653015159907b8a2bb59e.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Plaques de cellules transform√©es avec pUC19 apr√®s 18 heures: sans antibiotique (√† gauche) et avec antibiotique (√† droite)</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Transformation du produit apr√®s assemblage </font></font></h2><br>       pUC19, , ,        ,  sfGFP. <br><br>   ,       IPTG  X-gal,      <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">- </a> .    ,        pUC19,    sfGFP,       . <br><br><h4>    </h4><br>   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> , sfGFP        485  / 510 .  ,   Transcriptic   485/535. ,   485  510  .      600  ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">OD600</a> ). <br><br><img src="https://habrastorage.org/getpro/habr/post_images/6ab/0d3/453/6ab0d345359a5b1d15f8f6341f1bee08.png"><br> <i><font color="gray"> GFP ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">biotek</a> )</font></i> <br><br><h4> IPTG  X-gal </h4><br>  IPTG    1M    1:1000.   , X-gal   20 /    1:1000 (20 /). ,  2000¬µl LB    2  . <br><br>   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a>    40  X-gal   20 /  40  IPTG  0,1 mM ( 4  IPTG  1M),   30 .      ,     IPTG, X-gal        . <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Full Gibson assembly and transformation protocol for sfGFP and pUC19 v1: Spread IPTG and X-gal onto plates, then spread cells v2: Mix IPTG, X-gal and cells; spread the mixture v3: exclude X-gal so I can do colony picking better v4: repeat v3 to try other excitation/emission wavelengths"""</span></span> p = Protocol() options = { <span class="hljs-string"><span class="hljs-string">"gibson"</span></span> : <span class="hljs-keyword"><span class="hljs-keyword">False</span></span>, <span class="hljs-comment"><span class="hljs-comment"># do a new gibson assembly "sanger" : False, # sanger sequence product "control_pUC19" : True, # unassembled pUC19 "XGal" : False # excluding X-gal should make the colony picking easier } for k, v in list(options.items()): if v is False: del options[k] experiment_name = "sfgfp_puc19_gibson_plates_v4" # ----------------------------------------------------------------------- # Inventory # inv = { # catalog "water" : "rs17gmh5wafm5p", # catalog; Autoclaved MilliQ H2O; ambient "DH5a" : "rs16pbj944fnny", # catalog; Zymo DH5Œ±; cold_80 "Gibson Mix" : "rs16pfatkggmk5", # catalog; Gibson Mix (2X); cold_20 "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "pUC19" : "rs17tcqmncjfsh", # catalog; pUC19; cold_20 # my inventory "puc19_cut_v2": "ct187v4ea7vvca", # inventory; pUC19 cut with EcoRI; cold_20 "IPTG" : "ct18a2r5wn6tqz", # inventory; IPTG at 1M (conc semi-documented); cold_20 "XGal" : "ct18a2r5wp5hcv", # inventory; XGal at 0.1M (conc not documented); cold_20 "sfgfp_pcroe_v8_amplified" : "ct1874zqh22pab", # inventory; sfGFP amplified to 40ng/ul; cold_4 "sfgfp_puc19_gibson_v3_clone" : "ct188ejywa8jcv", # inventory; assembled sfGFP; cold_4 # kits (must be used differently) "lb-broth-100ug-ml-amp_6-flat" : "ki17sbb845ssx9", # catalog; ampicillin plates "noAB-amp_6-flat" : "ki17reefwqq3sq" # catalog; no antibiotic plates } # # Catalog (all to be discarded afterward) # water_tube = p.ref("water", cont_type="micro-1.5", storage="ambient", discard=True).well(0) transform_plate = p.ref("trn_plate", cont_type="96-pcr", storage="ambient", discard=True) transform_tube = transform_plate.well(39) # experiment transform_tube_L = p.ref("trn_tubeL", cont_type="micro-1.5", storage="ambient", discard=True).well(0) transctrl_tube = transform_plate.well(56) # control transctrl_tube_L = p.ref("trc_tubeL", cont_type="micro-1.5", storage="ambient", discard=True).well(0) # # Plating according to Tali's protocol # http://learn.transcriptic.com/blog/2015/9/9/provisioning-commercial-reagents # amp_6_flat = Container(None, p.container_type('6-flat')) p.refs[expid("amp_6_flat")] = Ref(expid("amp_6_flat"), {"reserve": inv['lb-broth-100ug-ml-amp_6-flat'], "store": {"where": 'cold_4'}}, amp_6_flat) noAB_6_flat = Container(None, p.container_type('6-flat')) p.refs[expid("noAB_6_flat")] = Ref(expid("noAB_6_flat"), {"reserve": inv['noAB-amp_6-flat'], "store": {"where": 'cold_4'}}, noAB_6_flat) # # My inventory: EcoRI-cut pUC19, oePCR'd sfGFP, Gibson-assembled pUC19, IPTG and X-Gal # if "gibson" in options: puc19_cut_tube = p.ref("puc19_ecori_v2_puc19_cut", id=inv["puc19_cut_v2"], cont_type="micro-1.5", storage="cold_20").well(0) sfgfp_pcroe_amp_tube = p.ref("sfgfp_pcroe_v8_amplified", id=inv["sfgfp_pcroe_v8_amplified"], cont_type="micro-1.5", storage="cold_4").well(0) clone_plate = p.ref(expid("clone"), cont_type="96-pcr", storage="cold_4", discard=False) else: clone_plate = p.ref("sfgfp_puc19_gibson_v3_clone", id=inv["sfgfp_puc19_gibson_v3_clone"], cont_type="96-pcr", storage="cold_4", discard=False) IPTG_tube = p.ref("IPTG", id=inv["IPTG"], cont_type="micro-1.5", storage="cold_20").well(0) if "XGal" in options: XGal_tube = p.ref("XGal", id=inv["XGal"], cont_type="micro-1.5", storage="cold_20").well(0) # # Initialize inventory # if "gibson" in options: all_inventory_wells = [puc19_cut_tube, sfgfp_pcroe_amp_tube, IPTG_tube] assert puc19_cut_tube.volume == ¬µl(66), puc19_cut_tube.volume assert sfgfp_pcroe_amp_tube.volume == ¬µl(36), sfgfp_pcroe_amp_tube.volume else: all_inventory_wells = [IPTG_tube, clone_plate.well(0)] if "XGal" in options: all_inventory_wells.append(XGal_tube) for well in all_inventory_wells: init_inventory_well(well) print("Inventory: {} {} {}".format(well.name, well.volume, well.properties)) # # Provisioning. Water is used all over the protocol. Provision an excess since it's cheap # p.provision(inv["water"], water_tube, ¬µl(500)) # ----------------------------------------------------------------------------- # Cloning/assembly (see NEBuilder protocol above) # # "Optimized efficiency is 50‚Äì100 ng of vectors with 2 fold excess of inserts." # pUC19 is 20ng/ul (78ul total). # sfGFP is ~40ng/ul (48ul total) # Therefore 4ul of each gives 80ng and 160ng of vector and insert respectively # def do_gibson_assembly(): # # Combine all the Gibson reagents in one tube and thermocycle # p.provision(inv["Gibson Mix"], clone_plate.well(0), ¬µl(10)) p.transfer(water_tube, clone_plate.well(0), ¬µl(2)) p.transfer(puc19_cut_tube, clone_plate.well(0), ¬µl(4)) p.transfer(sfgfp_pcroe_amp_tube, clone_plate.well(0), ¬µl(4), mix_after=True, mix_vol=¬µl(10), repetitions=10) p.seal(clone_plate) p.thermocycle(clone_plate, [{"cycles": 1, "steps": [{"temperature": "50:celsius", "duration": "16:minute"}]}], volume=¬µl(50)) # # Dilute assembled plasmid 4X according to the NEB Gibson assembly protocol (20ul-&gt;80ul) # p.unseal(clone_plate) p.transfer(water_tube, clone_plate.well(0), ¬µl(60), mix_after=True, mix_vol=¬µl(40), repetitions=5) return # -------------------------------------------------------------------------------------------------- # Transformation # "Transform NEB 5-alpha Competent E. coli cells with 2 Œºl of the # assembled product, following the appropriate transformation protocol." # # Mix &amp; Go http://www.zymoresearch.com/downloads/dl/file/id/173/t3015i.pdf # "[After mixing] Immediately place on ice and incubate for 2-5 minutes" # "The highest transformation efficiencies can be obtained by incubating Mix &amp; Go cells with DNA on # ice for 2-5 minutes (60 minutes maximum) prior to plating." # "It is recommended that culture plates be pre-warmed to &gt;20¬∞C (preferably 37¬∞C) prior to plating." # "Avoid exposing the cells to room temperature for more than a few seconds at a time." # # "If competent cells are purchased from other manufacture, dilute assembled products 4-fold # with H2O prior transformation. This can be achieved by mixing 5 Œºl of assembled products with # 15 Œºl of H2O. Add 2 Œºl of the diluted assembled product to competent cells." # def _do_transformation(): # # Combine plasmid and competent bacteria in a pcr_plate and shock # p.provision(inv["DH5a"], transform_tube, ¬µl(50)) p.transfer(clone_plate.well(0), transform_tube, ¬µl(3), dispense_speed="10:microliter/second") assert clone_plate.well(0).volume == ¬µl(54), clone_plate.well(0).volume if 'control_pUC19' in options: p.provision(inv["DH5a"], transctrl_tube, ¬µl(50)) p.provision(inv["pUC19"], transctrl_tube, ¬µl(1)) # # Heatshock the bacteria to transform using a PCR machine # p.seal(transform_plate) p.thermocycle(transform_plate, [{"cycles": 1, "steps": [{"temperature": "4:celsius", "duration": "5:minute"}]}, {"cycles": 1, "steps": [{"temperature": "37:celsius", "duration": "30:minute"}]}], volume=¬µl(50)) return def _transfer_transformed_to_plates(): assert transform_tube.volume == ¬µl(53), transform_tube.volume p.unseal(transform_plate) num_ab_plates = 4 # antibiotic places # # Transfer bacteria to a bigger tube for diluting # Then spread onto 6-flat plates # Generally you would spread 50-100ul of diluted bacteria # Put more on ampicillin plates for more opportunities to get a colony # I use a dilution series since it's unclear how much to plate # p.provision(inv["LB Miller"], transform_tube_L, ¬µl(429)) # # Add all IPTG and XGal to the master tube # 4ul (1M) IPTG on each plate; 40ul XGal on each plate # p.transfer(IPTG_tube, transform_tube_L, ¬µl(4*num_ab_plates)) if 'XGal' in options: p.transfer(XGal_tube, transform_tube_L, ¬µl(40*num_ab_plates)) # # Add the transformed cells and mix (use new mix op in case of different pipette) # p.transfer(transform_tube, transform_tube_L, ¬µl(50)) p.mix(transform_tube_L, volume=transform_tube_L.volume/2, repetitions=10) assert transform_tube.volume == dead_volume['96-pcr'] == ¬µl(3), transform_tube.volume assert transform_tube_L.volume == ¬µl(495), transform_tube_L.volume # # Spread an average of 60ul on each plate == 480ul total # for i in range(num_ab_plates): p.spread(transform_tube_L, amp_6_flat.well(i), ¬µl(51+i*6)) p.spread(transform_tube_L, noAB_6_flat.well(i), ¬µl(51+i*6)) assert transform_tube_L.volume == dead_volume["micro-1.5"], transform_tube_L.volume # # Controls: include 2 ordinary pUC19-transformed plates as a control # if 'control_pUC19' in options: num_ctrl = 2 assert num_ab_plates + num_ctrl &lt;= 6 p.provision(inv["LB Miller"], transctrl_tube_L, ¬µl(184)+dead_volume["micro-1.5"]) p.transfer(IPTG_tube, transctrl_tube_L, ¬µl(4*num_ctrl)) if "XGal" in options: p.transfer(XGal_tube, transctrl_tube_L, ¬µl(40*num_ctrl)) p.transfer(transctrl_tube, transctrl_tube_L, ¬µl(48)) p.mix(transctrl_tube_L, volume=transctrl_tube_L.volume/2, repetitions=10) for i in range(num_ctrl): p.spread(transctrl_tube_L, amp_6_flat.well(num_ab_plates+i), ¬µl(55+i*10)) p.spread(transctrl_tube_L, noAB_6_flat.well(num_ab_plates+i), ¬µl(55+i*10)) assert transctrl_tube_L.volume == dead_volume["micro-1.5"], transctrl_tube_L.volume assert IPTG_tube.volume == ¬µl(808), IPTG_tube.volume if "XGal" in options: assert XGal_tube.volume == ¬µl(516), XGal_tube.volume return def do_transformation(): _do_transformation() _transfer_transformed_to_plates() # ------------------------------------------------------ # Measure growth in plates (photograph) # def measure_growth(): # # Incubate and photograph 6-flat plates over 18 hours # to see blue or white colonies # for flat_name, flat in [(expid("amp_6_flat"), amp_6_flat), (expid("noAB_6_flat"), noAB_6_flat)]: for timepoint in [9,18]: p.cover(flat) p.incubate(flat, "warm_37", "9:hour") p.uncover(flat) p.image_plate(flat, mode="top", dataref=expid("{}_t{}".format(flat_name, timepoint))) return # --------------------------------------------------------------- # Sanger sequencing, TURNED OFF # Sequence to make sure assembly worked # 500ng plasmid, 1 ¬µl of a 10 ¬µM stock primer # "M13_F" : "rs17tcpqwqcaxe", # catalog; M13 Forward (-41); cold_20 (1ul = 100pmol) # "M13_R" : "rs17tcph6e2qzh", # catalog; M13 Reverse (-48); cold_20 (1ul = 100pmol) # def do_sanger_seq(): seq_primers = [inv["M13_F"], inv["M13_R"]] seq_wells = ["G1","G2"] p.unseal(pcr_plate) for primer_num, seq_well in [(0, seq_wells[0]),(1, seq_wells[1])]: p.provision(seq_primers[primer_num], pcr_plate.wells([seq_well]), ¬µl(1)) p.transfer(pcr_plate.wells(["A1"]), pcr_plate.wells(seq_wells), ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(water_tube, pcr_plate.wells(seq_wells), ¬µl(9)) p.mix(pcr_plate.wells(seq_wells), volume=¬µl(7.5), repetitions=10) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[0]).indices(), expid("seq1")) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[1]).indices(), expid("seq2")) return # --------------------------------------------------------------- # Generate protocol # # Skip Gibson since I already did it if 'gibson' in options: do_gibson_assembly() do_transformation() measure_growth() if 'sanger' in options: do_sanger_seq() # --------------------------------------------------------------- # Output protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("\nProtocol {}\n\n{}".format(experiment_name, jprotocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> Inventory: IPTG/IPTG/IPTG/IPTG/IPTG/IPTG 832.0:microliter {}<font></font>
Inventory: sfgfp_puc19_gibson_v3_clone/sfgfp_puc19_gibson_v3_clone/sfgfp_puc19_gibson_v3_clone/sfgfp_puc19_gibson_v3_clone/sfgfp_puc19_gibson_v3_clone 57.0:microliter {}<font></font>
<font></font>
  <font></font>
‚úì Protocol analyzed<font></font>
  40 instructions<font></font>
  8 containers<font></font>
  Total Cost: $53.20<font></font>
  Workcell Time: $17.35<font></font>
  Reagents &amp; Consumables: $35.86 </pre><br><h2>   </h2><br>      ,   ¬´¬ª       96- .        ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">autopick</a> ). <br><br><div class="spoiler">  <b class="spoiler_title">Code</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Pick colonies from plates and grow in amp media and check for fluorescence. v2: try again with a new plate (no blue colonies) v3: repeat with different emission and excitation wavelengths"""</span></span> p = Protocol() options = {} <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> k, v <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> list(options.items()): <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> v <span class="hljs-keyword"><span class="hljs-keyword">is</span></span> <span class="hljs-keyword"><span class="hljs-keyword">False</span></span>: <span class="hljs-keyword"><span class="hljs-keyword">del</span></span> options[k] experiment_name = <span class="hljs-string"><span class="hljs-string">"sfgfp_puc19_gibson_pick_v3"</span></span> <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">plate_expid</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(val)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-string"><span class="hljs-string">"""refer to the previous plating experiment's outputs"""</span></span> plate_exp = <span class="hljs-string"><span class="hljs-string">"sfgfp_puc19_gibson_plates_v4"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> <span class="hljs-string"><span class="hljs-string">"{}_{}"</span></span>.format(plate_exp, val) <span class="hljs-comment"><span class="hljs-comment"># ----------------------------------------------------------------------- # Inventory # inv = { # catalog "water" : "rs17gmh5wafm5p", # catalog; Autoclaved MilliQ H2O; ambient "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "IPTG" : "ct18a2r5wn6tqz", # inventory; IPTG at 1M (conc semi-documented); cold_20 # plates from previous experiment, must be changed every new experiment plate_expid("amp_6_flat") : "ct18snmr9avvg9", # inventory; Ampicillin plates with blue-white screening of pUC19 plate_expid("noAB_6_flat") : "ct18snmr9dxfw2", # inventory; no AB plates with blue-white screening of pUC19 } # Tubes and plates lb_amp_tubes = [p.ref("lb_amp_{}".format(i+1), cont_type="micro-2.0", storage="ambient", discard=True).well(0) for i in range(4)] lb_xab_tube = p.ref("lb_xab", cont_type="micro-2.0", storage="ambient", discard=True).well(0) growth_plate = p.ref(expid("growth"), cont_type="96-flat", storage="cold_4", discard=False) # My inventory IPTG_tube = p.ref("IPTG", id=inv["IPTG"], cont_type="micro-1.5", storage="cold_20").well(0) # ampicillin plate amp_6_flat = Container(None, p.container_type('6-flat')) p.refs[plate_expid("amp_6_flat")] = Ref(plate_expid("amp_6_flat"), {"id":inv[plate_expid("amp_6_flat")], "store": {"where": 'cold_4'}}, amp_6_flat) # Use a total of 50 wells abs_wells = ["{}{}".format(row,col) for row in "BCDEF" for col in range(1,11)] abs_wells_T = ["{}{}".format(row,col) for col in range(1,11) for row in "BCDEF"] assert abs_wells[:3] == ["B1","B2","B3"] and abs_wells_T[:3] == ["B1","C1","D1"] def prepare_growth_wells(): # # To LB, add ampicillin at ~1/1000 concentration # Mix slowly in case of overflow # p.provision(inv["LB Miller"], lb_xab_tube, ¬µl(1913)) for lb_amp_tube in lb_amp_tubes: p.provision(inv["Amp 100mgml"], lb_amp_tube, ¬µl(2)) p.provision(inv["LB Miller"], lb_amp_tube, ¬µl(1911)) p.mix(lb_amp_tube, volume=¬µl(800), repetitions=10) # # Add IPTG but save on X-Gal # http://openwetware.org/images/f/f1/Dh5a_sub.pdf # "If you are concerned about obtaining maximal levels of expression, add IPTG to a final concentration of 1 mM." # 2ul of IPTG in 2000ul equals 1mM # p.transfer(IPTG_tube, [lb_xab_tube] + lb_amp_tubes, ¬µl(2), one_tip=True) # # Distribute LB among wells, row D is control (no ampicillin) # cols = range(1,11) row = "D" # control, no AB cwells = ["{}{}".format(row,col) for col in cols] assert set(cwells).issubset(set(abs_wells)) p.distribute(lb_xab_tube, growth_plate.wells(cwells), ¬µl(190), allow_carryover=True) rows = "BCEF" for row, lb_amp_tube in zip(rows, lb_amp_tubes): cwells = ["{}{}".format(row,col) for col in cols] assert set(cwells).issubset(set(abs_wells)) p.distribute(lb_amp_tube, growth_plate.wells(cwells), ¬µl(190), allow_carryover=True) assert all(lb_amp_tube.volume == lb_xab_tube.volume == dead_volume['micro-2.0'] for lb_amp_tube in lb_amp_tubes) return def measure_growth_wells(): # # Growth: absorbance and fluorescence over 24 hours # Absorbance at 600nm: cell growth # Absorbance at 615nm: X-gal, in theory # Fluorescence at 485nm/510nm: sfGFP # or 450nm/508nm (http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2695656/) # hr = 4 for t in range(0,24,hr): if t &gt; 0: p.cover(growth_plate) p.incubate(growth_plate, "warm_37", "{}:hour".format(hr), shaking=True) p.uncover(growth_plate) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="485:nanometer", emission="535:nanometer", dataref=expid("fl2_{}".format(t)), flashes=25) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="450:nanometer", emission="508:nanometer", dataref=expid("fl1_{}".format(t)), flashes=25) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="395:nanometer", emission="508:nanometer", dataref=expid("fl0_{}".format(t)), flashes=25) p.absorbance(growth_plate, growth_plate.wells(abs_wells).indices(), wavelength="600:nanometer", dataref=expid("abs_{}".format(t)), flashes=25) return # --------------------------------------------------------------- # Protocol steps # prepare_growth_wells() batch = 10 for i in range(5): p.autopick(amp_6_flat.well(i), growth_plate.wells(abs_wells_T[i*batch:i*batch+batch]), dataref=expid("autopick_{}".format(i))) p.image_plate(amp_6_flat, mode="top", dataref=expid("autopicked_{}".format(i))) measure_growth_wells() # --------------------------------------------------------------- # Output protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> ‚úì Protocol analyzed<font></font>
  62 instructions<font></font>
  8 containers<font></font>
  Total Cost: $66.38<font></font>
  Workcell Time: $57.59<font></font>
  Reagents &amp; Consumables: $8.78 </pre><br><h2> :   </h2><br> ‚Äì   ,  ,       (1-4)        (5-6).   ,   ,      ,  ,    IPTG  X-gal,     Transcriptic. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/bc4/d2b/c10/bc4d2bc102d423993907a021598257ef.jpg"><br> <i><font color="gray">  -     (1-4)    (5-6)</font></i> <br><br>  -      - .                  ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">GraphicsMagick</a> ).  ,   ,     (   ,      ). <br><br>      ,   Transcriptic. ,      10     .   ,      ,      .             .      ,  ,  ,    ,    . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/c8f/ee9/b71/c8fee9b710a214fdb7421dbc2d74e1b4.jpg"><br> <i><font color="gray">   -    (1-4)    (5-6),     </font></i> <br><br> ‚Äì    .  ,     .   ,   .          X-gal. <br><br>     -           .  ,        . <br><br><img src="https://habrastorage.org/getpro/habr/post_images/2cb/6f1/835/2cb6f183502e484f66e1f95ba1b44ed7.jpg"><br> <i><font color="gray">,      (1-4)    (5-6)</font></i> <br><br><h2> :     </h2><br>   50    96-    20    ,    sfGFP.      ( ,  ) Transcriptic   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">Tecan Infinite</a> . <br><br>  ,         ,         ,      sfGFP.      ,      ,     - ,      sfGFP       . ,    sfGFP,       ,       ,       ,   . <br><br>     (OD600)        20  ( 60 ). <br><br><pre> <code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">for</span></span> t <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> [<span class="hljs-number"><span class="hljs-number">0</span></span>,<span class="hljs-number"><span class="hljs-number">4</span></span>,<span class="hljs-number"><span class="hljs-number">8</span></span>,<span class="hljs-number"><span class="hljs-number">12</span></span>,<span class="hljs-number"><span class="hljs-number">16</span></span>,<span class="hljs-number"><span class="hljs-number">20</span></span>]: abs_data = pd.read_csv(<span class="hljs-string"><span class="hljs-string">"glow/sfgfp_puc19_gibson_pick_v3_abs_{}.csv"</span></span>.format(t), index_col=<span class="hljs-string"><span class="hljs-string">"Well"</span></span>) flr_data = pd.read_csv(<span class="hljs-string"><span class="hljs-string">"glow/sfgfp_puc19_gibson_pick_v3_fl2_{}.csv"</span></span>.format(t), index_col=<span class="hljs-string"><span class="hljs-string">"Well"</span></span>) <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> t == <span class="hljs-number"><span class="hljs-number">0</span></span>: new_data = abs_data.join(flr_data) <span class="hljs-keyword"><span class="hljs-keyword">else</span></span>: new_data = new_data.join(abs_data, rsuffix=<span class="hljs-string"><span class="hljs-string">'_{}'</span></span>.format(t)) new_data = new_data.join(flr_data, rsuffix=<span class="hljs-string"><span class="hljs-string">'_{}'</span></span>.format(t)) new_data.columns = [<span class="hljs-string"><span class="hljs-string">"OD 600:nanometer_0"</span></span>, <span class="hljs-string"><span class="hljs-string">"Fluorescence_0"</span></span>] + list(new_data.columns[<span class="hljs-number"><span class="hljs-number">2</span></span>:])</code> </pre> <br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Nous pla√ßons sur la carte les donn√©es de la 20√®me heure et les traces des mesures pr√©c√©dentes. En fait, je ne suis int√©ress√© que par les derni√®res donn√©es, car c'est alors qu'il faut observer un pic de fluorescence.</font></font><br><br><pre> <code class="python hljs">svg = [] W, H = <span class="hljs-number"><span class="hljs-number">800</span></span>, <span class="hljs-number"><span class="hljs-number">500</span></span> min_x, max_x = <span class="hljs-number"><span class="hljs-number">0</span></span>, <span class="hljs-number"><span class="hljs-number">0.8</span></span> min_y, max_y = <span class="hljs-number"><span class="hljs-number">0</span></span>, <span class="hljs-number"><span class="hljs-number">50000</span></span> <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">_toxy</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(x, y)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> W*(x-min_x)/(max_x-min_x), HH*(y-min_y)/(max_y-min_y) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">_topt</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(x, y)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> <span class="hljs-string"><span class="hljs-string">','</span></span>.join(map(str,_toxy(x,y))) ab_fls = [[row[<span class="hljs-number"><span class="hljs-number">0</span></span>]] + [list(row[<span class="hljs-number"><span class="hljs-number">1</span></span>])] <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> row <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> new_data.iterrows()] <span class="hljs-comment"><span class="hljs-comment"># axes svg.append('&lt;g fill="#888" font-size="18" transform="translate(20,0),scale(.95)"&gt;') svg.append('&lt;text x="0" y="{}"&gt;OD600 ‚Üí&lt;/text&gt;'.format(H+20)) svg.append('&lt;text x="0" y="0" transform="rotate(-90),translate(-{},-8)"&gt;Fluorescence ‚Üí&lt;/text&gt;'.format(H)) svg.append('&lt;line x1="0" y1="{}" x2="{}" y2="{}" style="stroke:#888;stroke-width:2" /&gt;'.format(H,W,H)) svg.append('&lt;line x1="0" y1="0" x2="0" y2="{}" style="stroke:#888;stroke-width:2" /&gt;'.format(H)) # glow filter svg.append("""&lt;filter id="glow" x="-200%" y="-200%" height="400%" width="400%"&gt; &lt;feColorMatrix type="matrix" values="0 0 0 0 0 255 0 0 0 0 0 0 0 0 0 0 0 0 1 0"/&gt; &lt;feGaussianBlur stdDeviation="10" result="coloredBlur"/&gt; &lt;feMerge&gt;&lt;feMergeNode in="coloredBlur"/&gt;&lt;feMergeNode in="SourceGraphic"/&gt;&lt;/feMerge&gt; &lt;/filter&gt;""") for n, (well, vals) in enumerate(ab_fls): fill = "#444" if not well.startswith("D") else "#aaa" gfilter = 'filter="url(#glow)"' if well in ["C3", "D1", "D3"] else "" cx, cy = _toxy(*vals[-2:]) svg.append('''&lt;g id="point{n:d}"&gt;&lt;circle {gfilter:s} r="12" cx="{cx:f}" cy="{cy:f}" fill="{fill:s}" /&gt; &lt;text x="{cx:f}" y="{cy:f}" font-size="10" text-anchor="middle" fill="#fff" alignment-baseline="middle"&gt;{txt:s}&lt;/text&gt;&lt;/g&gt; '''.format(n=n, cx=cx, cy=cy, fill=fill, txt=well, gfilter=gfilter)) pathd = 'M{} '.format(_topt(*vals[:2])) pathd += ' '.join("L{}".format(_topt(*vals[i:i+2])) for i in range(2,len(vals),2)) svg.append('''&lt;path d="{pathd:}" stroke="#ccc" stroke-width=".2" fill="none" id="path{n:d}"/&gt;'''.format(pathd=pathd, n=n)) svg.append("&lt;/g&gt;") # entire chart group show_svg(''.join(svg), w=W, h=H)</span></span></code> </pre> <br><br><img src="https://habrastorage.org/webt/03/z3/kk/03z3kkczqokp7nhgplib_q1bhia.png"><br> <i><font color="gray">  OD600:    ,     .   ,       sfGFP</font></i> <br><br>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">miniprep</a> ,    ,       13.  ,  -  miniprep       - Transcriptic,    .          (C1, D1, D3)    (B1, B3, E1),    sfGFP   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">muscle</a> . <br><br>   C1, D3  D3      sfGFP,      B1, B3  E1       . <br><br><h4>    </h4><br>  ,    . ,           0     (40 000 ).  20-            OD600   (, -    ),      . ,     ,  , , 11-15 . <br><br>            (. .  ,  ),  ,       ,     ). <br><br>       , ,     50   sfGFP  .    ,   . ,       ( ,  ,  miniprep),       200  ,        . <br><br>       ,          .   ,         GFP,    Python! <br><br><h1>  :  </h1><br><h4>  Prix </h4><br>    ,  ,      $360,     : <br><br><ul><li> <b>$70</b>    <br></li><li> <b>$32</b>  PCR       <br></li><li> <b>$31</b>    <br></li><li> <b>$32</b>    <br></li><li> <b>$53</b>   <br></li><li> <b>$67</b>    <br></li><li> <b>$75</b>  3 miniprep'   </li></ul><br> ,      $250-300   . ,   50     , ,    . <br><br>   ,       ( )     (  IT).  Transcriptic         ,       .        ,       ,   .  ,   ,  ,     ,     . <br><br>  ,       . ,    -     ,         .         ,       :      ,  ,  IDT       . <br><br><h4>  :    </h4><br> ,       .            ,                  : <br><br><ul><li> <b>!</b> ,   .              autoprotocol,     . <br></li><li> <b></b> .      100    ,    . <br></li><li> <b>  </b> , ,  PCR.       ,          ,      ?    /       ?    ,   ,      ,      ¬´  2-3 ¬ª.  ? <br></li><li> <b> </b> .          .     ,   . <br></li><li> <b></b> .       . <br></li><li> <b></b> .           . ,      1  96    (96‚àíx)   96- ,    . <br></li><li> <b> </b> .         csv   ,    . <br></li><li> <b></b> .                -    ,    . </li></ul><br> ,    ,  ,     .    ,     1994 : <br><br><ul><li>    Transcriptic   ‚Äî  .   ,    ,         ,    .         , ,     . <br></li><li>         ‚Äî        Transcriptic. <br></li><li>   ,      .     Transcriptic     ( ,   ,     ). <br></li><li>         ,        (   : ~$0).    ,       . <br></li><li> Transcriptic      .   ,    ,      . </li></ul><br><h4>     </h4><br>         ,  ,    -  ,      ,         . <br><br>   ,    : <br><br><ul><li>   Twist/IDT/Gen9  Transcriptic (,     -     ). <br></li><li>    ,    ,       , ,    . . <br></li><li>     (    NEB,    IDT)       (, <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">primer3</a> ). </li></ul><br>          ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> ) , ,    <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a> . ,               in vivo (. .   ). <br><br>     ,        , :    RBS   ,    ;        ;   . <br><br><h4>  Pourquoi tout √ßa? </h4><br>        ,    .   : <br><br><ul><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a>   - //,  . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="></a>   ,   ,  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> </a> . <br></li><li>    <i>in vivo</i>   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">split-GFP</a>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> </a> . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">scFv</a>     .  scFvs   -  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">  </a> . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">BiTE</a>     ,      (   ,  ). <br></li><li>   ,     <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u=">  </a> (    ). <br></li><li>        ,  .    1000  10 000 ? ,   GFP? </li></ul><br>    ,     ,    <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> iGEM</a> . <br><br>      <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=fr&amp;u="> </a>  Transcriptic      . </div></div><p>Source: <a rel="nofollow" href="https://habr.com/ru/post/fr451124/">https://habr.com/ru/post/fr451124/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../fr451112/index.html">Jetpack compose</a></li>
<li><a href="../fr451114/index.html">Recherche sur les microbes indig√®nes et les produits dans les magasins</a></li>
<li><a href="../fr451116/index.html">Comment j'ai lutt√© avec la transition de l'√©l√©ment partag√© et √©crit ma premi√®re biblioth√®que open source</a></li>
<li><a href="../fr451118/index.html">Les tests ne sont pas pour les d√©butants</a></li>
<li><a href="../fr451120/index.html">√Ä propos des d√©fis du portage de Dead Cells sur des plateformes mobiles</a></li>
<li><a href="../fr451126/index.html">Bo√Æte √† outils pour les chercheurs - Num√©ro un: auto-organisation et visualisation des donn√©es</a></li>
<li><a href="../fr451130/index.html">Swift: ARC et gestion de la m√©moire</a></li>
<li><a href="../fr451132/index.html">Contrats ax√©s sur le consommateur ou automatisation des tests d'assurance qualit√© Gitlab CI-eyed</a></li>
<li><a href="../fr451136/index.html">Introduit par .NET 5</a></li>
<li><a href="../fr451138/index.html">Symfony CLI - Nouvel outil de d√©veloppement local</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter57283870 = new Ya.Metrika({
                  id:57283870,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/57283870" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134228602-6', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

<footer class="page-footer">
  <div class="page-footer-legal-info-container page-footer-element">
    <p>
      Weekly-Geekly ES | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
    </p>
  </div>
  <div class="page-footer-counters-container page-footer-element">
    <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=9uU9J9pq8z7k8xEBHYSfs6DenIBAHs3vLIHcPIJW9d0&co=3a3a3a&ct=ffffff'/></a>
  </div>
</footer>
  
</body>

</html>