<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134228602-6"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134228602-6');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>üöΩ üöü üö¥üèº Desenvolvimento de prote√≠nas na nuvem usando Python e Transcriptic ou Como criar qualquer prote√≠na por US $ 360 ‚úçüèº üîù üëáüèø</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="E se voc√™ tiver uma id√©ia para uma prote√≠na fresca e saud√°vel e quiser realiz√°-la? Por exemplo, voc√™ gostaria de criar uma vacina contra H. pylori (co...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <link href="https://fonts.googleapis.com/css?family=Quicksand&display=swap" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script>document.write('<script src="//pagea' + 'd2.googles' + 'yndication.com/pagea' + 'd/js/a' + 'dsby' + 'google.js"><\/script>')</script>
  <script>
        var superSpecialObject = {};
        superSpecialObject['google_a' + 'd_client'] = 'ca-p' + 'ub-6974184241884155';
        superSpecialObject['enable_page_level_a' + 'ds'] = true;
       (window['a' + 'dsbygoogle'] = window['a' + 'dsbygoogle'] || []).push(superSpecialObject);
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly-es.github.io/index.html"></a>
    <div class="page-header-text">Geekly articles weekly</div>
  </header>
  <section class="page js-page"><h1>Desenvolvimento de prote√≠nas na nuvem usando Python e Transcriptic ou Como criar qualquer prote√≠na por US $ 360</h1><div class="post__body post__body_full"><div class="post__text post__text-html" id="post-content-body" data-io-article-url="https://habr.com/ru/post/451124/">  E se voc√™ tiver uma id√©ia para uma prote√≠na fresca e saud√°vel e quiser realiz√°-la?  Por exemplo, voc√™ gostaria de criar uma vacina contra <i>H. pylori</i> (como a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">equipe eslovena da iGEM 2008</a> ) criando uma prote√≠na h√≠brida que combina fragmentos de flagelina de <i>E. coli</i> que estimulam a resposta imune com a habitual flagelina de <i>H. pylori</i> ? <br><br><div style="text-align:center;"><img src="https://habrastorage.org/getpro/habr/post_images/e9c/dd5/691/e9cdd5691f1bab368bb400a8818b6507.png"></div>  <i><font color="gray">H. pylori Hybrid Flagellin Design Apresentado pela Equipe Eslovena no iGEM 2008</font></i> <br><br>  Surpreendentemente, estamos muito perto de criar qualquer prote√≠na que desejamos sem sair do notebook de Jupyter, gra√ßas aos mais recentes desenvolvimentos em gen√¥mica, biologia sint√©tica e, mais recentemente, em laborat√≥rios em nuvem. <br><br>  Neste artigo, mostrarei o c√≥digo Python da id√©ia de uma prote√≠na √† sua express√£o em uma c√©lula bacteriana, sem tocar uma pipeta ou conversar com algu√©m.  O custo total ser√° de apenas algumas centenas de d√≥lares!  Usando a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">terminologia de Vijaya Pande da A16Z</a> , isso √© Biology 2.0. <br><a name="habracut"></a><br>  Mais especificamente, no artigo, o c√≥digo Python do laborat√≥rio em nuvem faz o seguinte: <br><br><ul><li>  <b>S√≠ntese de</b> uma sequ√™ncia de DNA que codifica qualquer prote√≠na que eu quero. <br></li><li>  <b>Clonar</b> esse DNA sint√©tico em um <b>vetor</b> que possa express√°-lo. <br></li><li>  <b>Transforma√ß√£o de</b> bact√©rias com esse <b>vetor</b> e confirma√ß√£o de que a express√£o est√° ocorrendo. </li></ul><br><h1>  Configura√ß√£o do Python </h1><br>  Primeiro, as configura√ß√µes gerais de Python necess√°rias para qualquer bloco de notas Jupyter.  Importamos alguns m√≥dulos Python √∫teis e criamos algumas fun√ß√µes utilit√°rias, principalmente para visualiza√ß√£o de dados. <br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre><code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">import</span></span> re <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> json <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> logging <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> requests <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> itertools <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> numpy <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> np <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> seaborn <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> sns <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> pandas <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> pd <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> matplotlib <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> mpl <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> matplotlib.pyplot <span class="hljs-keyword"><span class="hljs-keyword">as</span></span> plt <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> io <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> StringIO <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> pprint <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> pprint <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> Bio.Seq <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Seq <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> Bio.Alphabet <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> generic_dna <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> IPython.display <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> display, Image, HTML, SVG <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">uprint</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr)</span></span></span><span class="hljs-function">:</span></span> print(astr + <span class="hljs-string"><span class="hljs-string">"\n"</span></span> + <span class="hljs-string"><span class="hljs-string">"-"</span></span>*len(astr)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">show_html</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> display(HTML(<span class="hljs-string"><span class="hljs-string">'{}'</span></span>.format(astr))) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">show_svg</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(astr, w=</span></span><span class="hljs-number"><span class="hljs-function"><span class="hljs-params"><span class="hljs-number">1000</span></span></span></span><span class="hljs-function"><span class="hljs-params">, h=</span></span><span class="hljs-number"><span class="hljs-function"><span class="hljs-params"><span class="hljs-number">1000</span></span></span></span><span class="hljs-function"><span class="hljs-params">)</span></span></span><span class="hljs-function">:</span></span> SVG_HEAD = <span class="hljs-string"><span class="hljs-string">'''&lt;?xml version="1.0" standalone="no"?&gt;&lt;!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"&gt;'''</span></span> SVG_START = <span class="hljs-string"><span class="hljs-string">'''&lt;svg viewBox="0 0 {w:} {h:}" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink= "http://www.w3.org/1999/xlink"&gt;'''</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> display(SVG(SVG_HEAD + SVG_START.format(w=w, h=h) + astr + <span class="hljs-string"><span class="hljs-string">'&lt;/svg&gt;'</span></span>)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">table_print</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(rows, header=True)</span></span></span><span class="hljs-function">:</span></span> html = [<span class="hljs-string"><span class="hljs-string">"&lt;table&gt;"</span></span>] html_row = <span class="hljs-string"><span class="hljs-string">"&lt;/td&gt;&lt;td&gt;"</span></span>.join(k <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> k <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> rows[<span class="hljs-number"><span class="hljs-number">0</span></span>]) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;tr style='font-weight:{}'&gt;&lt;td&gt;{}&lt;/td&gt;&lt;/tr&gt;"</span></span>.format(<span class="hljs-string"><span class="hljs-string">'bold'</span></span> <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> header <span class="hljs-keyword"><span class="hljs-keyword">is</span></span> <span class="hljs-keyword"><span class="hljs-keyword">True</span></span> <span class="hljs-keyword"><span class="hljs-keyword">else</span></span> <span class="hljs-string"><span class="hljs-string">'normal'</span></span>, html_row)) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> row <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> rows[<span class="hljs-number"><span class="hljs-number">1</span></span>:]: html_row = <span class="hljs-string"><span class="hljs-string">"&lt;/td&gt;&lt;td&gt;"</span></span>.join(row) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;tr style='font-family:monospace;'&gt;&lt;td&gt;{:}&lt;/td&gt;&lt;/tr&gt;"</span></span>.format(html_row)) html.append(<span class="hljs-string"><span class="hljs-string">"&lt;/table&gt;"</span></span>) show_html(<span class="hljs-string"><span class="hljs-string">''</span></span>.join(html)) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">clean_seq</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(dna)</span></span></span><span class="hljs-function">:</span></span> dna = re.sub(<span class="hljs-string"><span class="hljs-string">"\s"</span></span>,<span class="hljs-string"><span class="hljs-string">""</span></span>,dna) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACGTN"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> dna) <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> Seq(dna, generic_dna) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">clean_aas</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(aas)</span></span></span><span class="hljs-function">:</span></span> aas = re.sub(<span class="hljs-string"><span class="hljs-string">"\s"</span></span>,<span class="hljs-string"><span class="hljs-string">""</span></span>,aas) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(aa <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACDEFGHIKLMNPQRSTVWY*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> aa <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> aas) <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> aas <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">Images</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(images, header=None, width=</span></span><span class="hljs-string"><span class="hljs-function"><span class="hljs-params"><span class="hljs-string">"100%"</span></span></span></span><span class="hljs-function"><span class="hljs-params">)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-comment"><span class="hljs-comment"># to match Image syntax if type(width)==type(1): width = "{}px".format(width) html = ["&lt;table style='width:{}'&gt;&lt;tr&gt;".format(width)] if header is not None: html += ["&lt;th&gt;{}&lt;/th&gt;".format(h) for h in header] + ["&lt;/tr&gt;&lt;tr&gt;"] for image in images: html.append("&lt;td&gt;&lt;img src='{}' /&gt;&lt;/td&gt;".format(image)) html.append("&lt;/tr&gt;&lt;/table&gt;") show_html(''.join(html)) def new_section(title, color="#66aa33", padding="120px"): style = "text-align:center;background:{};padding:{} 10px {} 10px;".format(color,padding,padding) style += "color:#ffffff;font-size:2.55em;line-height:1.2em;" return HTML('&lt;div style="{}"&gt;{}&lt;/div&gt;'.format(style, title)) # Show or hide text HTML(""" &lt;style&gt; .section { display:flex;align-items:center;justify-content:center;width:100%; height:400px; background:#6a3;color:#eee;font-size:275%; } .showhide_label { display:block; cursor:pointer; } .showhide { position: absolute; left: -999em; } .showhide + div { display: none; } .showhide:checked + div { display: block; } .shown_or_hidden { font-size:85%; } &lt;/style&gt; """) # Plotting style plt.rc("axes", titlesize=20, labelsize=15, linewidth=.25, edgecolor='#444444') sns.set_context("notebook", font_scale=1.2, rc={}) %matplotlib inline %config InlineBackend.figure_format = 'retina' # or 'svg'</span></span></code> </pre> </div></div><br><h1>  Laborat√≥rios em nuvem </h1><br>  Como a AWS ou qualquer nuvem de computa√ß√£o, o laborat√≥rio de nuvem possui equipamentos de biologia molecular, al√©m de rob√¥s que aluga pela Internet.  Voc√™ pode emitir instru√ß√µes para seus rob√¥s clicando em alguns bot√µes na interface ou escrevendo c√≥digo que os programa por conta pr√≥pria.  N√£o √© necess√°rio escrever seus pr√≥prios protocolos, como farei aqui, uma parte significativa da biologia molecular s√£o tarefas rotineiras padr√£o; portanto, geralmente √© melhor confiar em um protocolo alien√≠gena confi√°vel que mostre boa intera√ß√£o com os rob√¥s. <br><br>  Recentemente, surgiram v√°rias empresas com laborat√≥rios em nuvem: <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Transcriptic</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Autodesk Wet Lab Accelerator</a> (beta, e constru√≠do com base no Transcriptic), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Arcturus BioCloud</a> (beta), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Emerald Cloud Lab</a> (beta), <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Synthego</a> (ainda n√£o foi iniciado).  Existem at√© empresas criadas em cima de laborat√≥rios em nuvem, como a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Desktop Genetics</a> , especializada em CRISPR.  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Artigos cient√≠ficos</a> sobre o uso de laborat√≥rios em nuvem na ci√™ncia real est√£o come√ßando a aparecer. <br><br>  No momento da reda√ß√£o deste artigo, apenas o Transcriptic est√° em dom√≠nio p√∫blico, portanto, o usaremos.  Pelo que entendi, a maioria dos neg√≥cios da Transcriptic √© criada para automatizar protocolos comuns, e escrever seus pr√≥prios protocolos em Python (como farei neste artigo) √© menos comum. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/94e/38f/0a6/94e38f0a6f9ea9af3509a0278242b9b4.jpg"><br>  <i><font color="gray">‚ÄúC√©lula de trabalho‚Äù transcrita com refrigeradores na parte inferior e v√°rios equipamentos de laborat√≥rio no estande</font></i> <br><br>  Vou dar instru√ß√µes aos rob√¥s de transcri√ß√£o no <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">protocolo autom√°tico</a> .  Autoprotocol √© uma linguagem baseada em JSON para escrever protocolos para rob√¥s de laborat√≥rio (e humanos, por assim dizer).  O protocolo autom√°tico √© feito principalmente <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">nesta biblioteca Python</a> .  O idioma foi criado originalmente e ainda √© suportado pelo Transcriptic, mas, pelo que entendi, √© completamente aberto.  Existe uma boa <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">documenta√ß√£o</a> . <br><br>  Uma ideia interessante √© que voc√™ pode escrever instru√ß√µes para pessoas em laborat√≥rios remotos na, digamos, China ou √çndia, no auto-protocolo e potencialmente obter algumas vantagens ao usar pessoas (seu julgamento) e rob√¥s (falta de julgamento).  Precisamos mencionar <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">protocolocols.io aqui</a> , esta √© uma tentativa de padronizar protocolos para melhorar a reprodutibilidade, mas para humanos, n√£o para rob√¥s. <br><br><pre> <code class="json hljs"><span class="hljs-string"><span class="hljs-string">"instructions"</span></span>: [ { <span class="hljs-attr"><span class="hljs-attr">"to"</span></span>: [ { <span class="hljs-attr"><span class="hljs-attr">"well"</span></span>: <span class="hljs-string"><span class="hljs-string">"water/0"</span></span>, <span class="hljs-attr"><span class="hljs-attr">"volume"</span></span>: <span class="hljs-string"><span class="hljs-string">"500.0:microliter"</span></span> } ], <span class="hljs-attr"><span class="hljs-attr">"op"</span></span>: <span class="hljs-string"><span class="hljs-string">"provision"</span></span>, <span class="hljs-attr"><span class="hljs-attr">"resource_id"</span></span>: <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span> }, ... ]</code> </pre> <br>  <i><font color="gray">Exemplo de fragmento de protocolo autom√°tico</font></i> <br><br><h1>  Configura√ß√µes de Python para biologia molecular </h1><br>  Al√©m de importar bibliotecas padr√£o, precisarei de alguns utilit√°rios biol√≥gicos moleculares espec√≠ficos.  Este c√≥digo √© principalmente para protocolo autom√°tico e transcriptic. <br><br>  O conceito de "volume morto" √© frequentemente encontrado no c√≥digo.  Isso significa a √∫ltima gota de l√≠quido que os rob√¥s da Transcriptic n√£o conseguem tirar com uma pipeta dos tubos (porque eles n√£o podem v√™-la!).  Voc√™ precisa gastar muito tempo para garantir que os frascos tenham material suficiente. <br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">import</span></span> autoprotocol <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Unit <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.container <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Container <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.protocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Protocol <span class="hljs-keyword"><span class="hljs-keyword">from</span></span> autoprotocol.protocol <span class="hljs-keyword"><span class="hljs-keyword">import</span></span> Ref <span class="hljs-comment"><span class="hljs-comment"># "Link a ref name (string) to a Container instance." import requests import logging # Transcriptic authorization org_name = 'hgbrian' tsc_headers = {k:v for k,v in json.load(open("auth.json")).items() if k in ["X_User_Email","X_User_Token"]} # Transcriptic-specific dead volumes _dead_volume = [("96-pcr",3), ("96-flat",25), ("96-flat-uv",25), ("96-deep",15), ("384-pcr",2), ("384-flat",5), ("384-echo",15), ("micro-1.5",15), ("micro-2.0",15)] dead_volume = {k:Unit(v,"microliter") for k,v in _dead_volume} def init_inventory_well(well, headers=tsc_headers, org_name=org_name): """Initialize well (set volume etc) for Transcriptic""" def _container_url(container_id): return 'https://secure.transcriptic.com/{}/samples/{}.json'.format(org_name, container_id) response = requests.get(_container_url(well.container.id), headers=headers) response.raise_for_status() container = response.json() well_data = container['aliquots'][well.index] well.name = "{}/{}".format(container["label"], well_data['name']) if well_data['name'] is not None else container["label"] well.properties = well_data['properties'] well.volume = Unit(well_data['volume_ul'], 'microliter') if 'ERROR' in well.properties: raise ValueError("Well {} has ERROR property: {}".format(well, well.properties["ERROR"])) if well.volume &lt; Unit(20, "microliter"): logging.warn("Low volume for well {} : {}".format(well.name, well.volume)) return True def touchdown(fromC, toC, durations, stepsize=2, meltC=98, extC=72): """Touchdown PCR protocol generator""" assert 0 &lt; stepsize &lt; toC &lt; fromC def td(temp, dur): return {"temperature":"{:2g}:celsius".format(temp), "duration":"{:d}:second".format(dur)} return [{"cycles": 1, "steps": [td(meltC, durations[0]), td(C, durations[1]), td(extC, durations[2])]} for C in np.arange(fromC, toC-stepsize, -stepsize)] def convert_ug_to_pmol(ug_dsDNA, num_nts): """Convert ug dsDNA to pmol""" return float(ug_dsDNA)/num_nts * (1e6 / 660.0) def expid(val): """Generate a unique ID per experiment""" return "{}_{}".format(experiment_name, val) def ¬µl(microliters): """Unicode function name for creating microliter volumes""" return Unit(microliters,"microliter")</span></span></code> </pre> </div></div><br><h1>  S√≠ntese de DNA e biologia sint√©tica </h1><br>  Apesar de sua conex√£o com a moderna biologia sint√©tica, a s√≠ntese de DNA √© uma tecnologia bastante antiga.  Durante d√©cadas, conseguimos produzir oligonucleot√≠deos (isto √©, sequ√™ncias de DNA de at√© 200 bases).  No entanto, sempre foi caro, e a qu√≠mica nunca permitiu longas sequ√™ncias de DNA.  Recentemente, tornou-se poss√≠vel, a um pre√ßo razo√°vel, sintetizar genes inteiros (at√© milhares de bases).  Essa conquista realmente abre a era da "biologia sint√©tica". <br><br>  A <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Synthetic Genomics de</a> Craig Venter levou a biologia sint√©tica mais longe ao <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">sintetizar um organismo inteiro</a> - com mais de um milh√£o de bases de comprimento.  √Ä medida que o comprimento do DNA aumenta, o problema n√£o √© mais a s√≠ntese, mas a montagem (isto √©, a costura de sequ√™ncias de DNA sintetizadas).  Com cada montagem, voc√™ pode dobrar o comprimento do DNA (ou mais); portanto, ap√≥s uma d√∫zia ou mais de itera√ß√µes, voc√™ obt√©m uma <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">mol√©cula bastante longa</a> !  A distin√ß√£o entre s√≠ntese e montagem deve ficar clara em breve para o usu√°rio final. <br><br><h1>  Lei de Moore? </h1><br>  O pre√ßo da s√≠ntese de DNA est√° caindo muito r√°pido, de mais de US $ 0,30 h√° um ano, dois para cerca de US $ 0,10 hoje, mas est√° se desenvolvendo mais como bact√©rias do que como processadores.  Por outro lado, os pre√ßos do seq√ºenciamento de DNA est√£o caindo mais r√°pido que a lei de Moore.  Uma meta de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">US $ 0,02 por base √© definida como um ponto de inflex√£o</a> onde voc√™ pode substituir muitas manipula√ß√µes de DNA demoradas por s√≠ntese simples.  Por exemplo, a esse pre√ßo, voc√™ pode sintetizar um plasm√≠deo inteiro de 3kb por <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">US $ 60</a> e pular um monte de biologia molecular.  Espero que consigamos isso em alguns anos. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/373/b42/050/373b42050b870646857266bb3e07c7ed.png"><br>  <i><font color="gray">Pre√ßos de s√≠ntese de DNA em compara√ß√£o com os pre√ßos de seq√ºenciamento de DNA, pre√ßo para 1 base (Carlson, 2014)</font></i> <br><br><h1>  Empresas de s√≠ntese de DNA </h1><br>  Existem v√°rias grandes empresas no campo da s√≠ntese de DNA: O IDT √© o maior produtor de oligonucleot√≠deos e tamb√©m pode produzir "fragmentos de genes" ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">gBlocks</a> ) mais longos (at√© 2kb).  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Gen9</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Twist</a> e <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">DNA 2.0</a> geralmente se especializam em seq√º√™ncias de DNA mais longas - essas s√£o empresas de s√≠ntese de genes.  Existem tamb√©m algumas empresas novas e interessantes, como a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Cambrian Genomics</a> e o <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Genesis DNA</a> , que est√£o trabalhando nos m√©todos de s√≠ntese da pr√≥xima gera√ß√£o. <br><br>  Outras empresas, como <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Amyris</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Zymergen</a> e <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Ginkgo Bioworks</a> , usam o DNA sintetizado por essas empresas para trabalhar no n√≠vel do corpo.  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">A Genomics sint√©tica</a> tamb√©m faz isso, mas sintetiza o pr√≥prio DNA. <br><br>  Recentemente, a Ginkgo <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">fez um acordo com a Twist</a> para fazer 100 milh√µes de bases: o maior neg√≥cio que j√° vi.  Isso prova que vivemos no futuro. A Twist chegou a anunciar um c√≥digo promocional no Twitter: quando voc√™ compra 10 milh√µes de bases de DNA (quase todo o genoma do fermento!), Voc√™ recebe outros 10 milh√µes de gra√ßa. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/e07/bfb/ac9/e07bfbac9c53ddb79634b63efc3fc0e7.png"><br>  <i><font color="gray"><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Oferta de</a> nicho no Twitter <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Twist</a></font></i> <br><br><h1>  Parte Um: Design da Experi√™ncia </h1><br><h3>  Prote√≠na verde fluorescente </h3><br>  Nesta experi√™ncia, sintetizamos uma sequ√™ncia de DNA para uma <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">prote√≠na verde fluorescente</a> simples (GFP).  A prote√≠na GFP foi encontrada pela primeira vez em uma <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">√°gua</a> - <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">viva</a> que fluorescente sob luz ultravioleta.  Esta √© uma prote√≠na extremamente √∫til porque √© f√°cil detectar sua express√£o simplesmente medindo a fluoresc√™ncia.  Existem op√ß√µes de GFP que produzem amarelo, vermelho, laranja e outras cores. <br><br>  √â interessante ver como v√°rias muta√ß√µes afetam a cor de uma prote√≠na, e esse √© um problema de aprendizado de m√°quina potencialmente interessante.  Mais recentemente, voc√™ precisaria gastar muito tempo no laborat√≥rio para isso, mas agora vou mostrar que √© (quase) t√£o f√°cil quanto editar um arquivo de texto! <br><br>  Tecnicamente, meu GFP √© uma Super Folder Option (sfGFP) com algumas muta√ß√µes para melhorar a qualidade. <br><br><img src="https://habrastorage.org/webt/tb/tm/_v/tbtm_vcly3elxlwhbcoknhsjpuo.jpeg"><br>  <i><font color="gray">Na <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">superfolder-GFP</a> (sfGFP), algumas muta√ß√µes fornecem determinadas propriedades √∫teis.</font></i> <br><br><img src="https://habrastorage.org/getpro/habr/post_images/645/232/c67/645232c67631f484ed6e8b30f6fa91eb.jpg"><br>  <i><font color="gray">Estrutura GFP (visualizada usando <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">PV</a> )</font></i> <br><br><h3>  S√≠ntese de GFP em Twist </h3><br>  Eu tive a sorte de entrar no programa de testes alfa de Twist, ent√£o usei o servi√ßo de s√≠ntese de DNA deles (eles gentilmente fizeram meu pequeno pedido - obrigado, Twist!).  Esta √© uma nova empresa em nosso campo, com um novo processo de s√≠ntese simplificado.  Seus pre√ßos est√£o em torno de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">US $ 0,10 por base ou menos</a> , mas <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">ainda</a> est√£o <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">na vers√£o beta</a> , e o programa alfa do qual participei foi fechado.  A Twist levantou cerca de US $ 150 milh√µes, ent√£o sua tecnologia √© animada. <br><br>  Enviei minha sequ√™ncia de DNA para o Twist como uma planilha do Excel (ainda n√£o existe uma API, mas acho que ser√° em breve), e eles enviaram o DNA sintetizado diretamente para minha caixa no laborat√≥rio Transcriptic (eu tamb√©m usei o IDT para s√≠ntese, mas eles n√£o enviaram DNA na Transcriptic, que estraga um pouco a divers√£o). <br><br>  Obviamente, esse processo ainda n√£o se tornou um caso de uso t√≠pico e requer algum suporte, mas funcionou, para que todo o pipeline permanecesse virtual.  Sem isso, eu provavelmente precisaria ter acesso ao laborat√≥rio - muitas empresas n√£o enviar√£o DNA ou reagentes para seu endere√ßo residencial. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/734/378/b0b/734378b0b151d2ed64bbd3ebb6edf0f3.jpg"><br>  <i><font color="gray">O GFP √© inofensivo, ent√£o <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">qualquer tipo √© destacado</a></font></i> <br><br><h3>  Vetor plasm√≠deo </h3><br>  Para expressar essa prote√≠na em bact√©rias, o gene precisa viver em algum lugar; caso contr√°rio, o DNA sint√©tico que codifica o gene simplesmente se degrada instantaneamente.  Como regra, em biologia molecular, usamos um plasm√≠deo, um peda√ßo de DNA redondo que vive fora do genoma bacteriano e expressa prote√≠nas.  Os plasm√≠deos s√£o uma maneira conveniente de as bact√©rias compartilharem m√≥dulos funcionais √∫teis e independentes, como resist√™ncia a antibi√≥ticos.  Pode haver centenas de plasm√≠deos em uma c√©lula. <br><br>  A terminologia amplamente usada √© que um plasm√≠deo √© um <b>vetor</b> e o DNA sint√©tico √© uma inser√ß√£o (inser√ß√£o).  Ent√£o, aqui estamos tentando clonar a inser√ß√£o em um vetor e depois transformar as bact√©rias usando o vetor. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/66c/9ae/d03/66c9aed037ae408c6f642433d9a2a130.jpg"><br>  <i><font color="gray">Genoma bacteriano e plasm√≠deo (sem escala!) ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Wikipedia</a> )</font></i> <br><br><h3>  pUC19 </h3><br>  Eu escolhi um plasm√≠deo bastante padr√£o na s√©rie <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">pUC19</a> .  Esse plasm√≠deo √© usado com muita frequ√™ncia e, como est√° dispon√≠vel como parte do invent√°rio Transcriptic padr√£o, n√£o precisamos enviar nada a eles. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/4f3/576/8f1/4f35768f17d5ff5506f7d1aca0b064a0.png"><br>  <i><font color="gray">Estrutura da pUC19: os principais componentes s√£o o gene de resist√™ncia √† ampicilina, lacZŒ±, MCS / polylinker e a origem da replica√ß√£o (Wikipedia)</font></i> <br><br>  A PUC19 tem uma fun√ß√£o interessante: como cont√©m o gene lacZŒ±, voc√™ pode usar o m√©todo de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">sele√ß√£o azul e branco</a> e ver em quais col√¥nias a inser√ß√£o foi bem-sucedida.  S√£o necess√°rios dois produtos qu√≠micos: <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">IPTG</a> e <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">X-gal</a> , e o circuito funciona da seguinte maneira: <br><br><ul><li>  IPTG induz a express√£o de lacZŒ±. <br></li><li>  Se o lacZŒ± for desativado via DNA inserido no local de clonagem m√∫ltipla ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">MCS / poliligante</a> ) no lacZŒ±, o plasm√≠deo n√£o poder√° hidrolisar X-gal e essas col√¥nias ser√£o brancas em vez de azuis. <br></li><li>  Portanto, uma inser√ß√£o bem-sucedida produz col√¥nias brancas e uma inser√ß√£o com falha produz col√¥nias azuis. </li></ul><br><img src="https://habrastorage.org/getpro/habr/post_images/123/aac/aca/123aacacac400c6f96966d2f2caeb918.jpg"><br>  <i><font color="gray">A sele√ß√£o em azul e branco mostra onde a express√£o lacZŒ± foi desativada ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Wikipedia</a> )</font></i> <br><br>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">A documenta√ß√£o do openwetware</a> diz: <br><br><blockquote>  E. coli DH5Œ± n√£o requer IPTG para induzir a express√£o do promotor lac, mesmo se um repressor Lac for expresso na cepa.  O n√∫mero de c√≥pias da maioria dos plasm√≠deos excede o n√∫mero de repressores nas c√©lulas.  Se voc√™ precisar de express√£o m√°xima, adicione IPTG a uma concentra√ß√£o final de 1 mM. </blockquote><br><h1>  Sequ√™ncias de DNA sint√©tico </h1><br><h3>  Sequ√™ncia de DNA SfGFP </h3><br>  √â f√°cil obter a sequ√™ncia de DNA para sfGFP, pegando <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">a sequ√™ncia proteica</a> e <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">codificando-a com c√≥dons</a> adequados para o organismo hospedeiro (aqui, <i>E. coli</i> ).  Esta √© uma prote√≠na de tamanho m√©dio com 236 amino√°cidos, portanto, a 10 centavos de s√≠ntese de DNA custa cerca de <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">US $ 70</a> por base. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/296/9da/a41/2969daa417296779220366c36c02a0e5.png"><br>  <i><font color="gray">Wolfram Alpha, c√°lculo dos custos de s√≠ntese</font></i> <br><br>  As 12 primeiras bases de nosso sfGFP s√£o a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">sequ√™ncia Shine-Delgarno</a> , que eu adicionei, que teoricamente deveria aumentar a express√£o (AGGAGGACAGCT, ent√£o ATG ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">cod√£o de in√≠cio</a> ) lan√ßa a prote√≠na).  De acordo com uma ferramenta computacional desenvolvida pelo <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">Salis Lab</a> ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">slides da aula</a> ), podemos esperar uma express√£o m√©dia a alta de nossa prote√≠na (taxa de inicia√ß√£o da tradu√ß√£o de 10.000 "unidades arbitr√°rias"). <br><br><pre> <code class="python hljs">sfGFP_plus_SD = clean_seq(<span class="hljs-string"><span class="hljs-string">""" AGGAGGACAGCTATGTCGAAAGGAGAAGAACTGTTTACCGGTGTGGTTCCGATTCTGGTAGAACTGGA TGGGGACGTGAACGGCCATAAATTTAGCGTCCGTGGTGAGGGTGAAGGGGATGCCACAAATGGCAAAC TTACCCTTAAATTCATTTGCACTACCGGCAAGCTGCCGGTCCCTTGGCCGACCTTGGTCACCACACTG ACGTACGGGGTTCAGTGTTTTTCGCGTTATCCAGATCACATGAAACGCCATGACTTCTTCAAAAGCGC CATGCCCGAGGGCTATGTGCAGGAACGTACGATTAGCTTTAAAGATGACGGGACCTACAAAACCCGGG CAGAAGTGAAATTCGAGGGTGATACCCTGGTTAATCGCATTGAACTGAAGGGTATTGATTTCAAGGAA GATGGTAACATTCTCGGTCACAAATTAGAATACAACTTTAACAGTCATAACGTTTATATCACCGCCGA CAAACAGAAAAACGGTATCAAGGCGAATTTCAAAATCCGGCACAACGTGGAGGACGGGAGTGTACAAC TGGCCGACCATTACCAGCAGAACACACCGATCGGCGACGGCCCGGTGCTGCTCCCGGATAATCACTAT TTAAGCACCCAGTCAGTGCTGAGCAAAGATCCGAACGAAAAACGTGACCATATGGTGCTGCTGGAGTT CGTGACCGCCGCGGGCATTACCCATGGAATGGATGAACTGTATAAA"""</span></span>) print(<span class="hljs-string"><span class="hljs-string">"Read in sfGFP plus Shine-Dalgarno: {} bases long"</span></span>.format(len(sfGFP_plus_SD))) sfGFP_aas = clean_aas(<span class="hljs-string"><span class="hljs-string">"""MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPTLVTTLTYG VQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKN GIKANFKIRHNVEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK"""</span></span>) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> sfGFP_plus_SD[<span class="hljs-number"><span class="hljs-number">12</span></span>:].translate() == sfGFP_aas print(<span class="hljs-string"><span class="hljs-string">"Translation matches protein with accession 532528641"</span></span>)</code> </pre> <br><pre>  Leia em sfGFP plus Shine-Dalgarno: 726 bases de comprimento
 A tradu√ß√£o corresponde √† prote√≠na com a ades√£o 532528641 </pre><br><h3>  Sequ√™ncia de DNA PUC19 </h3><br>  Primeiro, verifico se a <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">sequ√™ncia de pUC19 que baixei do NEB tem o</a> comprimento correto e inclui o <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">poliligador</a> esperado. <br><br><pre> <code class="python hljs">pUC19_fasta = !cat puc19fsa.txt pUC19_fwd = clean_seq(<span class="hljs-string"><span class="hljs-string">''</span></span>.join(pUC19_fasta[<span class="hljs-number"><span class="hljs-number">1</span></span>:])) pUC19_rev = pUC19_fwd.reverse_complement() <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> all(nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> <span class="hljs-string"><span class="hljs-string">"ACGT"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> nt <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> pUC19_fwd) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> len(pUC19_fwd) == <span class="hljs-number"><span class="hljs-number">2686</span></span> pUC19_MCS = clean_seq(<span class="hljs-string"><span class="hljs-string">"GAATTCGAGCTCGGTACCCGGGGATCCTCTAGAGTCGACCTGCAGGCATGCAAGCTT"</span></span>) print(<span class="hljs-string"><span class="hljs-string">"Read in pUC19: {} bases long"</span></span>.format(len(pUC19_fwd))) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> pUC19_fwd print(<span class="hljs-string"><span class="hljs-string">"Found MCS/polylinker"</span></span>)</code> </pre> <br><pre>  Leia na pUC19: 2686 bases longas
 MCS / poliligante encontrado </pre><br>  Fazemos alguns CQs b√°sicos para garantir que EcoRI e BamHI estejam presentes na pUC19 apenas uma vez (as seguintes enzimas de restri√ß√£o est√£o dispon√≠veis no invent√°rio transcricional padr√£o: <i>PstI</i> , <i>PvuII</i> , <i>EcoRI</i> , <i>BamHI</i> , <i>BbsI</i> , <i>BsmBI</i> ). <br><br><pre> <code class="python hljs">REs = {<span class="hljs-string"><span class="hljs-string">"EcoRI"</span></span>:<span class="hljs-string"><span class="hljs-string">"GAATTC"</span></span>, <span class="hljs-string"><span class="hljs-string">"BamHI"</span></span>:<span class="hljs-string"><span class="hljs-string">"GGATTC"</span></span>} <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> rename, res <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> REs.items(): <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> (pUC19_fwd.find(res) == pUC19_fwd.rfind(res) <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> pUC19_rev.find(res) == pUC19_rev.rfind(res)) <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> (pUC19_fwd.find(res) == <span class="hljs-number"><span class="hljs-number">-1</span></span> <span class="hljs-keyword"><span class="hljs-keyword">or</span></span> pUC19_rev.find(res) == <span class="hljs-number"><span class="hljs-number">-1</span></span> <span class="hljs-keyword"><span class="hljs-keyword">or</span></span> pUC19_fwd.find(res) == len(pUC19_fwd) - pUC19_rev.find(res) - len(res)) print(<span class="hljs-string"><span class="hljs-string">"Asserted restriction enzyme sites present only once: {}"</span></span>.format(REs.keys()))</code> </pre> <br>  Agora, examinamos a sequ√™ncia lacZŒ± e verificamos que n√£o h√° nada inesperado.  Por exemplo, deve come√ßar com Met e terminar com um c√≥don de parada.  Tamb√©m √© f√°cil confirmar que esta √© a ORF completa de 324 pb lacZŒ± carregando a sequ√™ncia de pUC19 no <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">visualizador de snapgene</a> livre. <br><br><pre> <code class="python hljs">lacZ = pUC19_rev[<span class="hljs-number"><span class="hljs-number">2217</span></span>:<span class="hljs-number"><span class="hljs-number">2541</span></span>] print(<span class="hljs-string"><span class="hljs-string">"lacZŒ± sequence:\t{}"</span></span>.format(lacZ)) print(<span class="hljs-string"><span class="hljs-string">"r_MCS sequence:\t{}"</span></span>.format(pUC19_MCS.reverse_complement())) lacZ_p = lacZ.translate() <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> lacZ_p[<span class="hljs-number"><span class="hljs-number">0</span></span>] == <span class="hljs-string"><span class="hljs-string">"M"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> <span class="hljs-keyword"><span class="hljs-keyword">not</span></span> <span class="hljs-string"><span class="hljs-string">"*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> lacZ_p[:<span class="hljs-number"><span class="hljs-number">-1</span></span>] <span class="hljs-keyword"><span class="hljs-keyword">and</span></span> lacZ_p[<span class="hljs-number"><span class="hljs-number">-1</span></span>] == <span class="hljs-string"><span class="hljs-string">"*"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS.reverse_complement() <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> lacZ <span class="hljs-keyword"><span class="hljs-keyword">assert</span></span> pUC19_MCS.reverse_complement() == pUC19_rev[<span class="hljs-number"><span class="hljs-number">2234</span></span>:<span class="hljs-number"><span class="hljs-number">2291</span></span>] print(<span class="hljs-string"><span class="hljs-string">"Found MCS once in lacZ sequence"</span></span>)</code> </pre> <br><pre>  sequ√™ncia de lacZ: ATGACCATGATTACGCCAAGCTTGCATGCCTGCAGGTCGACTCTAGAGGATCCCCGGGTACCGAGCTCGAATTCACTGGCCGTCGTTTTACAACGTCGTGACTGGGAAAACCCTGGCGTTACCCAACTTAATCGCCTTGCAGCACATCCCCCTTTCGCCAGCTGGCGTAATAGCGAAGAGGCCCGCACCGATCGCCCTTCCCAACAGTTGCGCAGCCTGAATGGCGAATGGCGCCTGATGCGGTATTTTCTCCTTACGCATCTGTGCGGTATTTCACACCGCATATGGTGCACTCTCAGTACAATCTGCTCTGATGCCGCATAG
 Sequ√™ncia r_MCS: AAGCTTGCATGCCTGCAGGTCGACTCTAGAGGATCCCCGGGTACCGAGCTCGAATTC
 MCS encontrado uma vez na sequ√™ncia lacZ </pre><br><h1>  Gibson assembly </h1><br>  Montar DNA significa simplesmente reticula√ß√£o de fragmentos.  Geralmente, voc√™ coleta v√°rios fragmentos de DNA em um segmento mais longo e o clona em um plasm√≠deo ou genoma.  Nesta experi√™ncia, quero clonar um segmento de DNA no plasm√≠deo pUC19 abaixo do promotor lac para express√£o em <i>E. coli</i> . <br><br>  Existem muitos m√©todos de clonagem (por exemplo, <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">NEB</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">openwetware</a> , <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">addgene</a> ).  Aqui vou usar a montagem Gibson ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">desenvolvida por Daniel Gibson</a> na Synthetic Genomics em 2009), que n√£o √© necessariamente o m√©todo mais barato, mas simples e flex√≠vel.  Voc√™ s√≥ precisa colocar o DNA que deseja coletar (com a sobreposi√ß√£o apropriada) em um tubo de ensaio com o Gibson Assembly Master Mix, e ele se montar√°! <br><br><img src="https://habrastorage.org/getpro/habr/post_images/8df/946/db5/8df946db53b725f89277be267d2e3ab9.jpg"><br>  <i><font color="gray">Revis√£o da Assembl√©ia Gibson ( <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">NEB</a> )</font></i> <br><br><h3>  Material de origem </h3><br>  Come√ßamos com 100 ng de DNA sint√©tico em 10 mL de l√≠quido.  Isso equivale a 0,21 picomoles de DNA ou a uma concentra√ß√£o de 10 ng / Œºl. <br><br><pre> <code class="python hljs">pmol_sfgfp = convert_ug_to_pmol(<span class="hljs-number"><span class="hljs-number">0.1</span></span>, len(sfGFP_plus_SD)) print(<span class="hljs-string"><span class="hljs-string">"Insert: 100ng of DNA of length {:4d} equals {:.2f} pmol"</span></span>.format(len(sfGFP_plus_SD), pmol_sfgfp))</code> </pre> <br><pre>  Inser√ß√£o: 100ng de DNA de comprimento 726 √© igual a 0,21 pmol </pre><br>  De acordo com <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">o protocolo de montagem NEB</a> , isso √© material de origem suficiente: <br><br><blockquote>  A NEB recomenda um total de 0,02-0,5 picomoles de fragmentos de DNA quando 1 ou 2 fragmentos s√£o montados no vetor, ou 0,2-1,0 picomoles de fragmentos de DNA quando 4-6 fragmentos s√£o coletados. <br><br>  0,02-0,5 pmoles * X Œºl <br>  * A efici√™ncia de clonagem otimizada √© de 50 a 100 ng de vetores com excesso de 2-3 vezes de inser√ß√µes.  Use 5 vezes mais inser√ß√µes se o tamanho for menor que 200 bps.  O volume total de fragmentos de PCR n√£o filtrados na rea√ß√£o de montagem de Gibson n√£o deve exceder 20%. </blockquote><br><h3>  NEBuilder para montagem Gibson </h3><br>  O NEBuilder da Biolab √© uma ferramenta realmente √≥tima para criar o protocolo de compila√ß√£o Gibson.  Ele ainda gera um PDF abrangente de quatro p√°ginas com todas as informa√ß√µes.  Utilizando esta ferramenta, desenvolvemos um protocolo para cortar pUC19 com EcoRI e, em seguida, usamos PCR [PCR, a rea√ß√£o em cadeia da polimerase permite obter um aumento significativo em pequenas concentra√ß√µes de certos fragmentos de DNA no material biol√≥gico - aprox.  por.] para adicionar fragmentos do tamanho apropriado √† inser√ß√£o. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/1f5/d52/171/1f5d52171cb60c8e8b4cb83f24b0c8c8.png"><br><br><h1>  Parte dois: experimento </h1><br>  O experimento consiste em quatro etapas: <br><br><ol><li>  Rea√ß√£o de inser√ß√£o em cadeia da polimerase para adicionar material com uma sequ√™ncia de flanqueamento. <br></li><li>  Cortar um plasm√≠deo para acomodar a inser√ß√£o. <br></li><li>  Montagem por inser√ß√£o de Gibson e plasm√≠deos. <br></li><li>  Transforma√ß√£o de bact√©rias usando o plasm√≠deo montado. </li></ol><br><h3>  Etapa 1. Inser√ß√£o de PCR </h3><br>  A montagem de Gibson depende da sequ√™ncia de DNA que voc√™ coleta, tendo alguma sequ√™ncia sobreposta (consulte o protocolo NEB com instru√ß√µes detalhadas acima).  Al√©m da amplifica√ß√£o simples, a PCR tamb√©m permite adicionar uma sequ√™ncia de DNA de flanqueamento, incluindo simplesmente uma sequ√™ncia adicional nos primers (tamb√©m pode ser clonado <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">usando apenas OE-PCR</a> ). <br><br>  Sintetizamos os primers de acordo com o protocolo NEB acima.  Tentei <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">o protocolo de in√≠cio r√°pido</a> no site da Transcriptic, mas ainda existe <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">um comando de</a> <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">protocolo</a> <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">autom√°tico</a> .  A transcri√ß√£o em si n√£o sintetiza oligonucleot√≠deos; portanto, ap√≥s 1-2 dias de espera, esses iniciadores aparecem magicamente em meu invent√°rio (observe que a parte espec√≠fica dos genes dos iniciadores √© indicada em mai√∫scula abaixo, mas s√£o apenas coisas cosm√©ticas). <br><br><pre> <code class="python hljs">insert_primers = [<span class="hljs-string"><span class="hljs-string">"aaacgacggccagtgTTTATACAGTTCATCCATTCCATG"</span></span>, <span class="hljs-string"><span class="hljs-string">"cgggtaccgagctcgAGGAGGACAGCTATGTCG"</span></span>]</code> </pre> <br><h3>  An√°lise de Primer </h3><br>  Voc√™ pode analisar as propriedades desses primers usando o <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">IDT OligoAnalyzer</a> .<font style="vertical-align: inherit;"><font style="vertical-align: inherit;">ao depurar um experimento de PCR, √© √∫til conhecer os pontos de fus√£o e a probabilidade de um efeito colateral do </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">d√≠mero</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> do </font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;">iniciador</font></a><font style="vertical-align: inherit;"> , embora o protocolo NEB quase certamente selecione os primers com boas propriedades.</font></font><br><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Parte do flanco espec√≠fica do gene (mai√∫scula)</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Temperatura de derretimento: 51C, 53.5C</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
Sequ√™ncia completa</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Temperatura de derretimento: 64.5C, 68.5C</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Gancho de cabelo: -.4dG, -5dG</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Auto-d√≠mero: -9dG, -16dG</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Heterod√≠mero: -6dG </font></font></pre><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Eu passei por muitas itera√ß√µes de PCR antes de obter resultados satisfat√≥rios, incluindo experimentos com v√°rias marcas diferentes de misturas de PCR. </font><font style="vertical-align: inherit;">Como cada uma dessas itera√ß√µes pode levar v√°rios dias (dependendo da dura√ß√£o da fila do laborat√≥rio), vale a pena gastar tempo na depura√ß√£o antecipada: isso economiza muito tempo a longo prazo. </font><font style="vertical-align: inherit;">√Ä medida que o poder do laborat√≥rio em nuvem aumenta, esse problema deve se tornar menos agudo. </font><font style="vertical-align: inherit;">No entanto, √© improv√°vel que seu primeiro protocolo seja bem-sucedido - h√° muitas vari√°veis.</font></font><br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">""" PCR overlap extension of sfGFP according to NEB protocol. v5: Use 3/10ths as much primer as the v4 protocol. v6: more complex touchdown pcr procedure. The Q5 temperature was probably too hot v7: more time at low temperature to allow gene-specific part to anneal v8: correct dNTP concentration, real touchdown """</span></span> p = Protocol() <span class="hljs-comment"><span class="hljs-comment"># --------------------------------------------------- # Set up experiment # experiment_name = "sfgfp_pcroe_v8" template_length = 740 _options = {'dilute_primers' : False, # if working stock has not been made 'dilute_template': False, # if working stock has not been made 'dilute_dNTP' : False, # if working stock has not been made 'run_gel' : True, # run a gel to see the plasmid size 'run_absorbance' : False, # check absorbance at 260/280/320 'run_sanger' : False} # sanger sequence the new sequence options = {k for k,v in _options.items() if v is True} # --------------------------------------------------- # Inventory and provisioning # https://developers.transcriptic.com/v1.0/docs/containers # # 'sfgfp2': 'ct17yx8h77tkme', # inventory; sfGFP tube #2, micro-1.5, cold_20 # 'sfgfp_puc19_primer1': 'ct17z9542mrcfv', # inventory; micro-2.0, cold_4 # 'sfgfp_puc19_primer2': 'ct17z9542m5ntb', # inventory; micro-2.0, cold_4 # 'sfgfp_idt_1ngul': 'ct184nnd3rbxfr', # inventory; micro-1.5, cold_4, (ERROR: no template) # inv = { 'Q5 Polymerase': 'rs16pcce8rdytv', # catalog; Q5 High-Fidelity DNA Polymerase 'Q5 Buffer': 'rs16pcce8rmke3', # catalog; Q5 Reaction Buffer 'dNTP Mixture': 'rs16pcb542c5rd', # catalog; dNTP Mixture (25mM?) 'water': 'rs17gmh5wafm5p', # catalog; Autoclaved MilliQ H2O 'sfgfp_pcroe_v5_puc19_primer1_10uM': 'ct186cj5cqzjmr', # inventory; micro-1.5, cold_4 'sfgfp_pcroe_v5_puc19_primer2_10uM': 'ct186cj5cq536x', # inventory; micro-1.5, cold_4 'sfgfp1': 'ct17yx8h759dk4', # inventory; sfGFP tube #1, micro-1.5, cold_20 } # Existing inventory template_tube = p.ref("sfgfp1", id=inv['sfgfp1'], cont_type="micro-1.5", storage="cold_4").well(0) dilute_primer_tubes = [p.ref('sfgfp_pcroe_v5_puc19_primer1_10uM', id=inv['sfgfp_pcroe_v5_puc19_primer1_10uM'], cont_type="micro-1.5", storage="cold_4").well(0), p.ref('sfgfp_pcroe_v5_puc19_primer2_10uM', id=inv['sfgfp_pcroe_v5_puc19_primer2_10uM'], cont_type="micro-1.5", storage="cold_4").well(0)] # New inventory resulting from this experiment dilute_template_tube = p.ref("sfgfp1_0.25ngul", cont_type="micro-1.5", storage="cold_4").well(0) dNTP_10uM_tube = p.ref("dNTP_10uM", cont_type="micro-1.5", storage="cold_4").well(0) sfgfp_pcroe_out_tube = p.ref(expid("amplified"), cont_type="micro-1.5", storage="cold_4").well(0) # Temporary tubes for use, then discarded mastermix_tube = p.ref("mastermix", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) water_tube = p.ref("water", cont_type="micro-1.5", storage="ambient", discard=True).well(0) pcr_plate = p.ref("pcr_plate", cont_type="96-pcr", storage="cold_4", discard=True) if 'run_absorbance' in options: abs_plate = p.ref("abs_plate", cont_type="96-flat", storage="cold_4", discard=True) # Initialize all existing inventory all_inventory_wells = [template_tube] + dilute_primer_tubes for well in all_inventory_wells: init_inventory_well(well) print(well.name, well.volume, well.properties) # ----------------------------------------------------- # Provision water once, for general use # p.provision(inv["water"], water_tube, ¬µl(500)) # ----------------------------------------------------- # Dilute primers 1/10 (100uM-&gt;10uM) and keep at 4C # if 'dilute_primers' in options: for primer_num in (0,1): p.transfer(water_tube, dilute_primer_tubes[primer_num], ¬µl(90)) p.transfer(primer_tubes[primer_num], dilute_primer_tubes[primer_num], ¬µl(10), mix_before=True, mix_vol=¬µl(50)) p.mix(dilute_primer_tubes[primer_num], volume=¬µl(50), repetitions=10) # ----------------------------------------------------- # Dilute template 1/10 (10ng/ul-&gt;1ng/ul) and keep at 4C # OR # Dilute template 1/40 (10ng/ul-&gt;0.25ng/ul) and keep at 4C # if 'dilute_template' in options: p.transfer(water_tube, dilute_template_tube, ¬µl(195)) p.mix(dilute_template_tube, volume=¬µl(100), repetitions=10) # Dilute dNTP to exactly 10uM if 'dilute_DNTP' in options: p.transfer(water_tube, dNTP_10uM_tube, ¬µl(6)) p.provision(inv["dNTP Mixture"], dNTP_10uM_tube, ¬µl(4)) # ----------------------------------------------------- # Q5 PCR protocol # www.neb.com/protocols/2013/12/13/pcr-using-q5-high-fidelity-dna-polymerase-m0491 # # 25ul reaction # ------------- # Q5 reaction buffer 5 ¬µl # Q5 polymerase 0.25 ¬µl # 10mM dNTP 0.5 ¬µl -- 1¬µl = 4x12.5mM # 10uM primer 1 1.25 ¬µl # 10uM primer 2 1.25 ¬µl # 1pg-1ng Template 1 ¬µl -- 0.5 or 1ng/ul concentration # ------------------------------- # Sum 9.25 ¬µl # # # Mastermix tube will have 96ul of stuff, leaving space for 4x1ul aliquots of template p.transfer(water_tube, mastermix_tube, ¬µl(64)) p.provision(inv["Q5 Buffer"], mastermix_tube, ¬µl(20)) p.provision(inv['Q5 Polymerase'], mastermix_tube, ¬µl(1)) p.transfer(dNTP_10uM_tube, mastermix_tube, ¬µl(1), mix_before=True, mix_vol=¬µl(2)) p.transfer(dilute_primer_tubes[0], mastermix_tube, ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(dilute_primer_tubes[1], mastermix_tube, ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.mix(mastermix_tube, volume="48:microliter", repetitions=10) # Transfer mastermix to pcr_plate without template p.transfer(mastermix_tube, pcr_plate.wells(["A1","B1","C1"]), ¬µl(24)) p.transfer(mastermix_tube, pcr_plate.wells(["A2"]), ¬µl(24)) # acknowledged dead volume problems p.mix(pcr_plate.wells(["A1","B1","C1","A2"]), volume=¬µl(12), repetitions=10) # Finally add template p.transfer(template_tube, pcr_plate.wells(["A1","B1","C1"]), ¬µl(1)) p.mix(pcr_plate.wells(["A1","B1","C1"]), volume=¬µl(12.5), repetitions=10) # --------------------------------------------------------- # Thermocycle with Q5 and hot start # 61.1 annealing temperature is recommended by NEB protocol # p.seal is enforced by transcriptic # extension_time = int(max(2, np.ceil(template_length * (11.0/1000)))) assert 0 &lt; extension_time &lt; 60, "extension time should be reasonable for PCR" cycles = [{"cycles": 1, "steps": [{"temperature": "98:celsius", "duration": "30:second"}]}] + \ touchdown(70, 61, [8, 25, extension_time], stepsize=0.5) + \ [{"cycles": 16, "steps": [{"temperature": "98:celsius", "duration": "8:second"}, {"temperature": "61.1:celsius", "duration": "25:second"}, {"temperature": "72:celsius", "duration": "{:d}:second".format(extension_time)}]}, {"cycles": 1, "steps": [{"temperature": "72:celsius", "duration": "2:minute"}]}] p.seal(pcr_plate) p.thermocycle(pcr_plate, cycles, volume=¬µl(25)) # -------------------------------------------------------- # Run a gel to hopefully see a 740bp fragment # if 'run_gel' in options: p.unseal(pcr_plate) p.mix(pcr_plate.wells(["A1","B1","C1","A2"]), volume=¬µl(12.5), repetitions=10) p.transfer(pcr_plate.wells(["A1","B1","C1","A2"]), pcr_plate.wells(["D1","E1","F1","D2"]), [¬µl(2), ¬µl(4), ¬µl(8), ¬µl(8)]) p.transfer(water_tube, pcr_plate.wells(["D1","E1","F1","D2"]), [¬µl(18),¬µl(16),¬µl(12),¬µl(12)], mix_after=True, mix_vol=¬µl(10)) p.gel_separate(pcr_plate.wells(["D1","E1","F1","D2"]), ¬µl(20), "agarose(10,2%)", "ladder1", "10:minute", expid("gel")) #--------------------------------------------------------- # Absorbance dilution series. Take 1ul out of the 25ul pcr plate wells # if 'run_absorbance' in options: p.unseal(pcr_plate) abs_wells = ["A1","B1","C1","A2","B2","C2","A3","B3","C3"] p.transfer(water_tube, abs_plate.wells(abs_wells[0:6]), ¬µl(10)) p.transfer(water_tube, abs_plate.wells(abs_wells[6:9]), ¬µl(9)) p.transfer(pcr_plate.wells(["A1","B1","C1"]), abs_plate.wells(["A1","B1","C1"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) p.transfer(abs_plate.wells(["A1","B1","C1"]), abs_plate.wells(["A2","B2","C2"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) p.transfer(abs_plate.wells(["A2","B2","C2"]), abs_plate.wells(["A3","B3","C3"]), ¬µl(1), mix_after=True, mix_vol=¬µl(5)) for wavelength in [260, 280, 320]: p.absorbance(abs_plate, abs_plate.wells(abs_wells), "{}:nanometer".format(wavelength), exp_id("abs_{}".format(wavelength)), flashes=25) # ----------------------------------------------------------------------------- # Sanger sequencing: https://developers.transcriptic.com/docs/sanger-sequencing # "Each reaction should have a total volume of 15 ¬µl and we recommend the following composition of DNA and primer: # PCR product (40 ng), primer (1 ¬µl of a 10 ¬µM stock)" # # By comparing to the gel ladder concentration (175ng/lane), it looks like 5ul of PCR product has approximately 30ng of DNA # if 'run_sanger' in options: p.unseal(pcr_plate) seq_wells = ["G1","G2"] for primer_num, seq_well in [(0, seq_wells[0]),(1, seq_wells[1])]: p.transfer(dilute_primer_tubes[primer_num], pcr_plate.wells([seq_well]), ¬µl(1), mix_before=True, mix_vol=¬µl(50)) p.transfer(pcr_plate.wells(["A1"]), pcr_plate.wells([seq_well]), ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(water_tube, pcr_plate.wells([seq_well]), ¬µl(9)) p.mix(pcr_plate.wells(seq_wells), volume=¬µl(7.5), repetitions=10) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[0]).indices(), expid("seq1")) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[1]).indices(), expid("seq2")) # ------------------------------------------------------------------------- # Then consolidate to one tube. Leave at least 3ul dead volume in each tube # remaining_volumes = [well.volume - dead_volume['96-pcr'] for well in pcr_plate.wells(["A1","B1","C1"])] print("Consolidated volume", sum(remaining_volumes, ¬µl(0))) p.consolidate(pcr_plate.wells(["A1","B1","C1"]), sfgfp_pcroe_out_tube, remaining_volumes, allow_carryover=True) uprint("\nProtocol 1. Amplify the insert (oligos previously synthesized)") jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> AVISO: raiz: volume baixo para o po√ßo sfGFP 1 / sfGFP 1: 2.0: microlitro </font></font></pre><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> sfGFP 1 / sfGFP 1 2.0: microlitro {'dilui√ß√£o': '0,25 ng / ul'}</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
sfgfp_pcroe_v5_puc19_primer1_10uM 75.0: microlitro {}</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
sfgfp_pcroe_v5_puc19_primer2_10uM 75.0: microlitro {}</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
Volume consolidado 52.0: microlitro</font></font><font></font>
<font></font>
Protocol 1. Amplify the insert (oligos previously synthesized)<font></font>
---------------------------------------------------------------<font></font>
<font></font>
  <font></font>
‚úì Protocol analyzed<font></font>
  11 instructions<font></font>
  8 containers<font></font>
  Total Cost: $32.18<font></font>
  Workcell Time: $4.32<font></font>
  Reagents &amp; Consumables: $27.86 </pre><br><h1> : PCR  </h1><br><h3>     </h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">No gel, voc√™ pode avaliar o </font></font><b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">tamanho</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> correto </font><b><font style="vertical-align: inherit;">do</font></b><font style="vertical-align: inherit;"> produto ap√≥s aumentar a concentra√ß√£o (a posi√ß√£o da tira no gel) e a </font></font><b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">quantidade</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> correta </font><font style="vertical-align: inherit;">(tira escura). O gel possui uma escada correspondente a v√°rios comprimentos e quantidades de DNA que podem ser usados ‚Äã‚Äãpara compara√ß√£o. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Na fotografia de gel abaixo, as bandas D1, E1, F1 cont√™m, respectivamente, 2 Œºl, 4 Œºl e 8 Œºl do produto amplificado. Eu posso estimar a quantidade de DNA em cada faixa em compara√ß√£o com o DNA na escada (50 ng de DNA por faixa na escada). Eu acho que os resultados parecem muito limpos. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Tentei usar o </font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;">GelEval</font></a><font style="vertical-align: inherit;"> para an√°lise de imagens e </font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;">estimativa de</font></a><font style="vertical-align: inherit;"> concentra√ß√£o</font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">e com bastante sucesso, embora n√£o tenha certeza se isso √© muito mais preciso que o m√©todo mais ing√™nuo. No entanto, pequenas mudan√ßas na localiza√ß√£o e no tamanho das bandas levaram a grandes mudan√ßas na estimativa da quantidade de DNA. Minha melhor estimativa da quantidade de DNA no meu produto amplificado usando GelEval √© de 40 ng / Œºl. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Supondo que estamos </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">limitados pela quantidade de iniciador na mistura</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> e n√£o pela quantidade de dNTP ou enzima, uma vez que tenho 12,5 pmol de cada iniciador, isso significa um m√°ximo te√≥rico de 6 Œºg de 740bp de DNA em 25 Œºl. Como minha estimativa da quantidade total de DNA usando GelEval √© de 40 ng x 25 Œºl (1 Œºg ou 2 pmol), esses resultados s√£o muito razo√°veis ‚Äã‚Äãe pr√≥ximos do que eu deveria esperar em condi√ß√µes ideais.</font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/74a/f5d/284/74af5d28463dbbd2279353a4651087f6.jpg"><br> <i><font color="gray">- EcoRI- pUC19,   (D1, E1, F1),   (D2)</font></i> <br><br><h3>   PCR </h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">A Transcriptic come√ßou recentemente a fornecer dados de diagn√≥stico interessantes e √∫teis de seus rob√¥s. </font><font style="vertical-align: inherit;">No momento da reda√ß√£o deste artigo, eles n√£o estavam dispon√≠veis para download. Por enquanto, s√≥ tenho uma imagem das temperaturas durante o ciclo t√©rmico. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Os dados parecem bons, sem altos ou baixos inesperados. </font><font style="vertical-align: inherit;">Um total de 35 ciclos de PCR, mas alguns desses ciclos s√£o realizados em temperaturas muito altas como parte do </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">touchdown</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> da </font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;">PCR</font></a><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Nas minhas tentativas anteriores de amplificar esse segmento - dos quais havia v√°rios! </font><font style="vertical-align: inherit;">- houve problemas com a hibrida√ß√£o dos primers, ent√£o aqui a PCR trabalha por muito tempo em altas temperaturas, o que deve aumentar a precis√£o. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/8dc/9eb/1b8/8dc9eb1b85c382df68cc064583512707.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Diagn√≥sticos termoc√≠clicos para PCR de touchdown: temperaturas de bloqueio, amostra e cobertura por 35 ciclos e 42 minutos</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Etapa 2. Cortando o plasm√≠deo </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para inserir nosso DNA sfGFP em pUC19, voc√™ primeiro precisa cortar o plasm√≠deo. </font><font style="vertical-align: inherit;">Seguindo o protocolo NEB, fa√ßo isso usando a enzima de restri√ß√£o </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">EcoRI</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Existem reagentes que eu preciso no invent√°rio padr√£o da Transcriptic: esse √© o </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">tamp√£o NEB EcoRI e 10x CutSmart</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> , bem como o plasm√≠deo </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">NEB pUC19</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para informa√ß√µes, abaixo est√£o os pre√ßos de seu invent√°rio. </font><font style="vertical-align: inherit;">De fato, pago apenas parte do pre√ßo, j√° que a Transcriptic aceita o pagamento pelo valor realmente consumido:</font></font><br><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ID do item Quantidade Pre√ßo de concentra√ß√£o</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
------------ ------ ------------- ----------------- - ----</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
CutSmart 10x B7204S 5 ml 10 X R $ 19,00</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
EcoRI R3101L 50.000 unidades 20.000 unidades / ml US $ 225,00</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
pUC19 N3041L 250 Œºg 1.000 Œºg / ml $ 268,00 </font></font></pre><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Eu segui o protocolo NEB, tanto quanto poss√≠vel: </font></font><br><br><blockquote>       .   10X  dH2O    1X.   ,  ,   , , .   50    5  10x NEBuffer       ,   dH2O. <br><br>      ,    1  Œª    1   37¬∞C     50 .  ,   5-10       10-20     1- . <br><br>   1      50 . </blockquote><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Protocol for cutting pUC19 with EcoRI."""</span></span> p = Protocol() experiment_name = <span class="hljs-string"><span class="hljs-string">"puc19_ecori_v3"</span></span> options = {} inv = { <span class="hljs-string"><span class="hljs-string">'water'</span></span>: <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "pUC19": "rs17tcqmncjfsh", # catalog; pUC19; cold_20 "EcoRI": "rs17ta8xftpdk6", # catalog; EcoRI-HF; cold_20 "CutSmart": "rs17ta93g3y85t", # catalog; CutSmart Buffer 10x; cold_20 "ecori_p10x": "ct187v4ea85k2h", # inventory; EcoRI diluted 10x } # Tubes and plates I use then discard re_tube = p.ref("re_tube", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) water_tube = p.ref("water_tube", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) pcr_plate = p.ref("pcr_plate", cont_type="96-pcr", storage="cold_4", discard=True) # The result of the experiment, a pUC19 cut by EcoRI, goes in this tube for storage puc19_cut_tube = p.ref(expid("puc19_cut"), cont_type="micro-1.5", storage="cold_20").well(0) # ------------------------------------------------------------- # Provisioning and diluting. # Diluted EcoRI can be used more than once # p.provision(inv["water"], water_tube, ¬µl(500)) if 'dilute_ecori' in options: ecori_p10x_tube = p.ref("ecori_p10x", cont_type="micro-1.5", storage="cold_20").well(0) p.transfer(water_tube, ecori_p10x_tube, ¬µl(45)) p.provision(inv["EcoRI"], ecori_p10x_tube, ¬µl(5)) else: # All "inventory" (stuff I own at transcriptic) must be initialized ecori_p10x_tube = p.ref("ecori_p10x", id=inv["ecori_p10x"], cont_type="micro-1.5", storage="cold_20").well(0) init_inventory_well(ecori_p10x_tube) # ------------------------------------------------------------- # Restriction enzyme cutting pUC19 # # 50ul total reaction volume for cutting 1ug of DNA: # 5ul CutSmart 10x # 1ul pUC19 (1ug of DNA) # 1ul EcoRI (or 10ul diluted EcoRI, 20 units, &gt;10 units per ug DNA) # p.transfer(water_tube, re_tube, ¬µl(117)) p.provision(inv["CutSmart"], re_tube, ¬µl(15)) p.provision(inv["pUC19"], re_tube, ¬µl(3)) p.mix(re_tube, volume=¬µl(60), repetitions=10) assert re_tube.volume == ¬µl(120) + dead_volume["micro-1.5"] print("Volumes: re_tube:{} water_tube:{} EcoRI:{}".format(re_tube.volume, water_tube.volume, ecori_p10x_tube.volume)) p.distribute(re_tube, pcr_plate.wells(["A1","B1","A2"]), ¬µl(40)) p.distribute(water_tube, pcr_plate.wells(["A2"]), ¬µl(10)) p.distribute(ecori_p10x_tube, pcr_plate.wells(["A1","B1"]), ¬µl(10)) assert all(well.volume == ¬µl(50) for well in pcr_plate.wells(["A1","B1","A2"])) p.mix(pcr_plate.wells(["A1","B1","A2"]), volume=¬µl(25), repetitions=10) # Incubation to induce cut, then heat inactivation of EcoRI p.seal(pcr_plate) p.incubate(pcr_plate, "warm_37", "60:minute", shaking=False) p.thermocycle(pcr_plate, [{"cycles": 1, "steps": [{"temperature": "65:celsius", "duration": "21:minute"}]}], volume=¬µl(50)) # -------------------------------------------------------------- # Gel electrophoresis, to ensure the cutting worked # p.unseal(pcr_plate) p.mix(pcr_plate.wells(["A1","B1","A2"]), volume=¬µl(25), repetitions=5) p.transfer(pcr_plate.wells(["A1","B1","A2"]), pcr_plate.wells(["D1","E1","D2"]), ¬µl(8)) p.transfer(water_tube, pcr_plate.wells(["D1","E1","D2"]), ¬µl(15), mix_after=True, mix_vol=¬µl(10)) assert all(well.volume == ¬µl(20) + dead_volume["96-pcr"] for well in pcr_plate.wells(["D1","E1","D2"])) p.gel_separate(pcr_plate.wells(["D1","E1","D2"]), ¬µl(20), "agarose(10,2%)", "ladder2", "15:minute", expid("gel")) # ---------------------------------------------------------------------------- # Then consolidate all cut plasmid to one tube (puc19_cut_tube). # remaining_volumes = [well.volume - dead_volume['96-pcr'] for well in pcr_plate.wells(["A1","B1"])] print("Consolidated volume: {}".format(sum(remaining_volumes, ¬µl(0)))) p.consolidate(pcr_plate.wells(["A1","B1"]), puc19_cut_tube, remaining_volumes, allow_carryover=True) assert all(tube.volume &gt;= dead_volume['micro-1.5'] for tube in [water_tube, re_tube, puc19_cut_tube, ecori_p10x_tube]) # --------------------------------------------------------------- # Test protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("Protocol {}\n\n{}".format(experiment_name, jprotocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre> Volumes: re_tube:135.0:microliter water_tube:383.0:microliter EcoRI:30.0:microliter<font></font>
Consolidated volume: 78.0:microliter<font></font>
<font></font>
  <font></font>
‚úì Protocol analyzed<font></font>
  12 instructions<font></font>
  5 containers<font></font>
  Total Cost: $30.72<font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Tempo da c√©lula de trabalho: $ 3,38</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Reagentes e Consum√≠veis: $ 27,34 </font></font></pre><br><h3><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: corte de plasm√≠deo </font></font></h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Realizei esse experimento duas vezes sob condi√ß√µes ligeiramente diferentes e com g√©is de tamanhos diferentes, mas os resultados s√£o quase id√™nticos. </font><font style="vertical-align: inherit;">Eu gosto dos dois g√©is. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Inicialmente, n√£o aloquei espa√ßo suficiente para o </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">volume "morto"</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> (em tubos de ensaio de 1,5 ml, o volume morto √© de 15 ¬µl!). </font><font style="vertical-align: inherit;">Eu acho que isso explica a diferen√ßa entre D1 e E1 (as duas bandas devem ser id√™nticas). </font><font style="vertical-align: inherit;">O problema do volume morto √© f√°cil de resolver, criando o suprimento de trabalho adequado do EcoRI dilu√≠do no in√≠cio do protocolo. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Apesar desse erro, em ambos os g√©is, as bandas D1 e E1 t√™m bandas fortes na posi√ß√£o correta de 2,6kb. </font><font style="vertical-align: inherit;">Na banda D2, um plasm√≠deo n√£o cortado: como esperado, n√£o √© vis√≠vel em um gel e quase invis√≠vel em outro.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Duas fotos em gel parecem bem diferentes. </font><font style="vertical-align: inherit;">Isso se deve em parte ao fato de que essa etapa da Transcriptic ainda n√£o foi automatizada. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/8f8/b9c/e3c/8f8b9ce3cc42de99099976a3c60a9ada.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Dois g√©is mostrando pUC19 cortado (2,6kb) nas bandas D1 e E1 e pUC19 n√£o cortado em D2</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Etapa 3. Montagem Gibson </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">A maneira mais f√°cil de verificar se minha montagem funciona usando o m√©todo Gibson √© coletar a inser√ß√£o e o plasm√≠deo, usar os </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">primers M13</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> padr√£o </font><font style="vertical-align: inherit;">(que flanqueiam a inser√ß√£o) para amplificar parte do plasm√≠deo e o DNA inserido e executar </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">qPCR</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> e </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">gel</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> para garantir que a amplifica√ß√£o funcione. </font><font style="vertical-align: inherit;">Voc√™ tamb√©m pode executar uma rea√ß√£o de seq√ºenciamento para confirmar que tudo est√° inserido conforme o esperado, mas decidi deix√°-lo para mais tarde. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Se o conjunto de Gibson n√£o funcionar, a amplifica√ß√£o de M13 n√£o funcionar√° porque o plasm√≠deo foi cortado entre duas sequ√™ncias M13.</font></font><br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Debugging transformation protocol: Gibson assembly followed by qPCR and a gel v2: include v3 Gibson assembly"""</span></span> p = Protocol() options = {} experiment_name = <span class="hljs-string"><span class="hljs-string">"debug_sfgfp_puc19_gibson_seq_v2"</span></span> inv = { <span class="hljs-string"><span class="hljs-string">"water"</span></span> : <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "M13_F" : "rs17tcpqwqcaxe", # catalog; M13 Forward (-41); cold_20 (1ul = 100pmol) "M13_R" : "rs17tcph6e2qzh", # catalog; M13 Reverse (-48); cold_20 (1ul = 100pmol) "SensiFAST_SYBR_No-ROX" : "rs17knkh7526ha", # catalog; SensiFAST SYBR for qPCR "sfgfp_puc19_gibson_v1_clone" : "ct187rzdq9kd7q", # inventory; assembled sfGFP; cold_4 "sfgfp_puc19_gibson_v3_clone" : "ct188ejywa8jcv", # inventory; assembled sfGFP; cold_4 } # --------------------------------------------------------------- # First get my sfGFP pUC19 clones, assembled with Gibson assembly # clone_plate1 = p.ref("sfgfp_puc19_gibson_v1_clone", id=inv["sfgfp_puc19_gibson_v1_clone"], cont_type="96-pcr", storage="cold_4", discard=False) clone_plate2 = p.ref("sfgfp_puc19_gibson_v3_clone", id=inv["sfgfp_puc19_gibson_v3_clone"], cont_type="96-pcr", storage="cold_4", discard=False) water_tube = p.ref("water", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) master_tube = p.ref("master", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) primer_tube = p.ref("primer", cont_type="micro-1.5", storage="cold_4", discard=True).well(0) pcr_plate = p.ref(expid("pcr_plate"), cont_type="96-pcr", storage="cold_4", discard=False) init_inventory_well(clone_plate1.well("A1")) init_inventory_well(clone_plate2.well("A1")) seq_wells = ["B2","B4","B6", # clone_plate1 "D2","D4","D6", # clone_plate2 "F2","F4"] # control # clone_plate2 was diluted 4X (20ul-&gt;80ul), according to NEB instructions assert clone_plate1.well("A1").volume == ¬µl(18), clone_plate1.well("A1").volume assert clone_plate2.well("A1").volume == ¬µl(78), clone_plate2.well("A1").volume # -------------------------------------------------------------- # Provisioning # p.provision(inv["water"], water_tube, ¬µl(500)) # primers, diluted 2X, discarded at the end p.provision(inv["M13_F"], primer_tube, ¬µl(13)) p.provision(inv["M13_R"], primer_tube, ¬µl(13)) p.transfer(water_tube, primer_tube, ¬µl(26), mix_after=True, mix_vol=¬µl(20), repetitions=10) # ------------------------------------------------------------------- # PCR Master mix -- 10ul SYBR mix, plus 1ul each undiluted primer DNA (100pmol) # Also add 15ul of dead volume # p.provision(inv['SensiFAST_SYBR_No-ROX'], master_tube, ¬µl(11+len(seq_wells)*10)) p.transfer(primer_tube, master_tube, ¬µl(4+len(seq_wells)*4)) p.mix(master_tube, volume=¬µl(63), repetitions=10) assert master_tube.volume == ¬µl(127) # 15ul dead volume p.distribute(master_tube, pcr_plate.wells(seq_wells), ¬µl(14), allow_carryover=True) p.distribute(water_tube, pcr_plate.wells(seq_wells), [¬µl(ul) for ul in [5,4,2, 4,2,0, 6,6]], allow_carryover=True) # Template -- starting with some small, unknown amount of DNA produced by Gibson p.transfer(clone_plate1.well("A1"), pcr_plate.wells(seq_wells[0:3]), [¬µl(1),¬µl(2),¬µl(4)], one_tip=True) p.transfer(clone_plate2.well("A1"), pcr_plate.wells(seq_wells[3:6]), [¬µl(2),¬µl(4),¬µl(6)], one_tip=True) assert all(pcr_plate.well(w).volume == ¬µl(20) for w in seq_wells) assert clone_plate1.well("A1").volume == ¬µl(11) assert clone_plate2.well("A1").volume == ¬µl(66) # -------------------------------------------------------------- # qPCR # standard melting curve parameters # p.seal(pcr_plate) p.thermocycle(pcr_plate, [{"cycles": 1, "steps": [{"temperature": "95:celsius","duration": "2:minute"}]}, {"cycles": 40, "steps": [{"temperature": "95:celsius","duration": "5:second"}, {"temperature": "60:celsius","duration": "20:second"}, {"temperature": "72:celsius","duration": "15:second", "read": True}]}], volume=¬µl(20), # volume is optional dataref=expid("qpcr"), dyes={"SYBR": seq_wells}, # dye must be specified (tells transcriptic what aborbance to use?) melting_start="65:celsius", melting_end="95:celsius", melting_increment="0.5:celsius", melting_rate="5:second") # -------------------------------------------------------------- # Gel -- 20ul required # Dilute such that I have 11ul for sequencing # p.unseal(pcr_plate) p.distribute(water_tube, pcr_plate.wells(seq_wells), ¬µl(11)) p.gel_separate(pcr_plate.wells(seq_wells), ¬µl(20), "agarose(8,0.8%)", "ladder1", "10:minute", expid("gel")) # This appears to be a bug in Transcriptic. The actual volume should be 11ul # but it is not updating after running a gel with 20ul. # Primer tube should be equal to dead volume, or it's a waste assert all(pcr_plate.well(w).volume==¬µl(31) for w in seq_wells) assert primer_tube.volume == ¬µl(16) == dead_volume['micro-1.5'] + ¬µl(1) assert water_tube.volume &gt; ¬µl(25) # --------------------------------------------------------------- # Test and run protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> AVISO: raiz: volume baixo para o po√ßo sfgfp_puc19_gibson_v1_clone / sfgfp_puc19_gibson_v1_clone: ‚Äã‚Äã11.0: microlitro </font></font></pre><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ‚úì Protocolo analisado</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  11 instru√ß√µes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  6 recipientes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Custo total: $ 32.09</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Tempo de Workcell: $ 6,98</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Reagentes e Consum√≠veis: $ 25,11 </font></font></pre><br><h3><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: qPCR para montagem Gibson </font></font></h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Eu posso acessar dados qPCR no formato JSON por meio da API Transcriptic. </font><font style="vertical-align: inherit;">Esse recurso n√£o est√° bem </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">documentado</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> , mas pode ser extremamente √∫til. </font><font style="vertical-align: inherit;">As APIs ainda oferecem acesso a alguns dados de diagn√≥stico de rob√¥s, o que pode ajudar na depura√ß√£o. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Primeiro, solicitamos dados de inicializa√ß√£o:</font></font><br><br><pre> <code class="python hljs">project_id, run_id = <span class="hljs-string"><span class="hljs-string">"p16x6gna8f5e9"</span></span>, <span class="hljs-string"><span class="hljs-string">"r18mj3cz3fku7"</span></span> api_url = <span class="hljs-string"><span class="hljs-string">"https://secure.transcriptic.com/hgbrian/{}/runs/{}/data.json"</span></span>.format(project_id, run_id) data_response = requests.get(api_url, headers=tsc_headers) data = data_response.json()</code> </pre> <br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Em seguida, especificamos esse ID para obter os dados de "p√≥s-processamento" do qPCR: </font></font><br><br><pre> <code class="python hljs">qpcr_id = data[<span class="hljs-string"><span class="hljs-string">'debug_sfgfp_puc19_gibson_seq_v1_qpcr'</span></span>][<span class="hljs-string"><span class="hljs-string">'id'</span></span>] pp_api_url = <span class="hljs-string"><span class="hljs-string">"https://secure.transcriptic.com/data/{}.json?key=postprocessed_data"</span></span>.format(qpcr_id) data_response = requests.get(pp_api_url, headers=tsc_headers) pp_data = data_response.json()</code> </pre> <br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Aqui est√£o os valores de Ct (limite do ciclo) para cada tubo. </font><font style="vertical-align: inherit;">Ct √© simplesmente o ponto em que a fluoresc√™ncia excede um determinado valor. </font><font style="vertical-align: inherit;">Ela diz aproximadamente quanto DNA existe no momento (e, portanto, aproximadamente onde come√ßamos).</font></font><br><br><pre> <code class="python hljs"><span class="hljs-comment"><span class="hljs-comment"># Simple util to convert wellnum to wellname n_w = {str(wellnum):'ABCDEFGH'[wellnum//12]+str(1+wellnum%12) for wellnum in range(96)} w_n = {v: k for k, v in n_w.items()} ct_vals = {n_w[k]:v for k,v in pp_data["amp0"]["SYBR"]["cts"].items()} ct_df = pd.DataFrame(ct_vals, index=["Ct"]).T ct_df["well"] = ct_df.index f, ax = plt.subplots(figsize=(16,6)) _ = sns.barplot(y="well", x="Ct", data=ct_df)</span></span></code> </pre> <br><img src="https://habrastorage.org/getpro/habr/post_images/32f/a42/de3/32fa42de3ad35392ac0bbf7ae6f595b3.png"><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Como voc√™ pode ver, a amplifica√ß√£o ocorre principalmente nos tubos de teste D2 / 4/6 (onde o DNA do meu conjunto Gibson √© "v3") e, em seguida, B2 / 4/6 (o conjunto Gibson √© "v1"). </font><font style="vertical-align: inherit;">As diferen√ßas entre v1 e v3 s√£o basicamente que o DNA v3 √© dilu√≠do 4X de acordo com o protocolo NEB, mas ambas as op√ß√µes devem funcionar. </font><font style="vertical-align: inherit;">Existe alguma amplifica√ß√£o ap√≥s o ciclo 30 em tubos de controle (F2, F4) sem modelo de DNA, mas isso n√£o √© incomum, pois inclui muito DNA iniciador. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Tamb√©m posso tra√ßar a curva de amplifica√ß√£o de qPCR para ver a din√¢mica da amplifica√ß√£o.</font></font><br><br><pre> <code class="python hljs">f, ax = plt.subplots(figsize=(<span class="hljs-number"><span class="hljs-number">16</span></span>,<span class="hljs-number"><span class="hljs-number">6</span></span>)) ax.set_color_cycle([<span class="hljs-string"><span class="hljs-string">'#fb6a4a'</span></span>, <span class="hljs-string"><span class="hljs-string">'#de2d26'</span></span>, <span class="hljs-string"><span class="hljs-string">'#a50f15'</span></span>, <span class="hljs-string"><span class="hljs-string">'#74c476'</span></span>, <span class="hljs-string"><span class="hljs-string">'#31a354'</span></span>, <span class="hljs-string"><span class="hljs-string">'#006d2c'</span></span>, <span class="hljs-string"><span class="hljs-string">'#08519c'</span></span>, <span class="hljs-string"><span class="hljs-string">'#6baed6'</span></span>]) amp0 = pp_data[<span class="hljs-string"><span class="hljs-string">'amp0'</span></span>][<span class="hljs-string"><span class="hljs-string">'SYBR'</span></span>][<span class="hljs-string"><span class="hljs-string">'baseline_subtracted'</span></span>] _ = [plt.plot(amp0[w_n[well]], label=well) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> well <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> [<span class="hljs-string"><span class="hljs-string">'B2'</span></span>, <span class="hljs-string"><span class="hljs-string">'B4'</span></span>, <span class="hljs-string"><span class="hljs-string">'B6'</span></span>, <span class="hljs-string"><span class="hljs-string">'D2'</span></span>, <span class="hljs-string"><span class="hljs-string">'D4'</span></span>, <span class="hljs-string"><span class="hljs-string">'D6'</span></span>, <span class="hljs-string"><span class="hljs-string">'F2'</span></span>, <span class="hljs-string"><span class="hljs-string">'F4'</span></span>]] _ = ax.set_ylim(<span class="hljs-number"><span class="hljs-number">0</span></span>,) _ = plt.title(<span class="hljs-string"><span class="hljs-string">"qPCR (reds=Gibson v1, greens=Gibson v3, blues=control)"</span></span>) _ = plt.legend(bbox_to_anchor=(<span class="hljs-number"><span class="hljs-number">1</span></span>, <span class="hljs-number"><span class="hljs-number">.75</span></span>), bbox_transform=plt.gcf().transFigure)</code> </pre> <br><img src="https://habrastorage.org/getpro/habr/post_images/88e/d2a/a19/88ed2aa19537b77b3214e6b36f55fdde.png"><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Em geral, os resultados de qPCR s√£o √≥timos, com boa amplifica√ß√£o de ambas as vers√µes do meu conjunto Gibson e sem amplifica√ß√£o real no grupo de controle. </font><font style="vertical-align: inherit;">Como o assembly v3 mostrou um resultado um pouco melhor que o v1, agora vamos us√°-lo.</font></font><br><br><h3><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: Montagem Gibson em gel </font></font></h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">O gel tamb√©m √© muito limpo, mostrando bandas fortes logo abaixo de 1kb nas bandas B2, B4, B6, D2, D4, D6: esse √© o tamanho que esperamos (a inser√ß√£o √© de cerca de 740 pb e os primers M13 t√™m cerca de 40 pb para cima e para baixo). </font><font style="vertical-align: inherit;">A segunda faixa corresponde aos primers. </font><font style="vertical-align: inherit;">Voc√™ pode ter certeza disso, pois as bandas F2 e F4 cont√™m apenas DNA iniciador. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/cf7/509/629/cf750962951b700c8a9333ea1d747a94.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Eletroforese em gel de poliacrilamida: c O conjunto Gibson v3 mostra bandas mais fortes (D2, D4, D6), de acordo com os dados qPCR acima</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Etapa 4. Transforma√ß√£o </font></font></h2><br> <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Transforma√ß√£o</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> √© o processo de mudar o corpo adicionando DNA. </font><font style="vertical-align: inherit;">Nesta experi√™ncia, </font></font><b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">transformamos </font></font></b> <i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">E. coli</font></font></i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> usando o plasm√≠deo pUC19 que expressa sfGFP. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Utilizamos a cepa </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Zymo DH5Œ± Mix &amp; Go</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> f√°cil de usar </font><font style="vertical-align: inherit;">e o </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">protocolo recomendado</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Essa linhagem faz parte do invent√°rio Transcriptic padr√£o. </font><font style="vertical-align: inherit;">Em geral, as transforma√ß√µes podem ser complexas porque as c√©lulas competentes s√£o bastante fr√°geis; portanto, quanto mais simples e confi√°vel o protocolo, melhor. </font><font style="vertical-align: inherit;">Nos laborat√≥rios comuns de biologia molecular, essas c√©lulas competentes provavelmente seriam caras demais para uso geral. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/951/9a6/a57/9519a6a578f11d26c3c9a2ddda039539.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">C√©lulas Zymo Mix &amp; Go com protocolo simples</font></font></font></i> <br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Problema com rob√¥s </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Este protocolo √© um bom exemplo de como √© dif√≠cil adaptar protocolos humanos para uso por rob√¥s e como ele pode falhar inesperadamente. Os protocolos √†s vezes s√£o surpreendentemente vagos ("giram o tubo de um lado para o outro"), com base no contexto geral dos bi√≥logos moleculares, ou podem, de repente, solicitar processamento avan√ßado de imagem ("verifique se o gr√¢nulo est√° misturado"). As pessoas n√£o se importam com essas tarefas, mas os rob√¥s precisam de instru√ß√µes mais claras.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Essa transforma√ß√£o mostrou quest√µes interessantes de tempo. O protocolo de transforma√ß√£o aconselha que as c√©lulas n√£o permane√ßam em temperatura ambiente por mais de alguns segundos e a placa com tubos deve ser pr√©-aquecida a 37 ¬∞ C. Teoricamente, voc√™ gostaria de come√ßar o pr√©-aquecimento para que termine simultaneamente com a transforma√ß√£o, mas n√£o est√° claro como os rob√¥s Transcriptic v√£o lidar com essa situa√ß√£o - at√© onde eu sei, n√£o h√° como sincronizar com precis√£o as etapas do protocolo. A falta de controle preciso do tempo parece ser um problema comum nos protocolos rob√≥ticos devido √† inflexibilidade relativa do bra√ßo rob√≥tico, conflitos de agendamento, etc. Teremos que ajustar os protocolos adequadamente.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Geralmente existem solu√ß√µes razo√°veis: √†s vezes voc√™ s√≥ precisa usar reagentes diferentes (por exemplo, c√©lulas mais resistentes, como Mix &amp; Go acima); </font><font style="vertical-align: inherit;">√†s vezes voc√™ apenas hipoteca a√ß√µes com uma margem (por exemplo, sacode dez vezes em vez de tr√™s); </font><font style="vertical-align: inherit;">√†s vezes voc√™ precisa criar truques especiais para rob√¥s (por exemplo, use uma m√°quina de PCR para insola√ß√£o). </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Obviamente, a grande vantagem √© que, uma vez que o protocolo funcionou uma vez, geralmente voc√™ pode confiar nele novamente. </font><font style="vertical-align: inherit;">Voc√™ pode at√© quantificar a confiabilidade do protocolo e melhor√°-lo com o tempo!</font></font><br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Transforma√ß√£o de teste </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Antes de iniciar a transforma√ß√£o com o plasm√≠deo totalmente montado, conduzo um experimento simples para garantir que a transforma√ß√£o usando o pUC19 usual (isto √©, sem montagem de Gibson e sem inser√ß√£o de DNA sfGFP) funcione. </font><font style="vertical-align: inherit;">A pUC19 cont√©m o gene de resist√™ncia √† ampicilina; portanto, uma transforma√ß√£o bem-sucedida deve permitir que as bact√©rias cres√ßam em placas contendo esse antibi√≥tico. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Transfiro as bact√©rias diretamente para o tablet ("6 planos" na terminologia da Transcriptic), onde existe ou n√£o ampicilina. </font><font style="vertical-align: inherit;">Espero que as bact√©rias transformadas contenham o gene de resist√™ncia √† ampicilina e, portanto, cres√ßam. </font><font style="vertical-align: inherit;">Bact√©rias n√£o transformadas n√£o devem crescer.</font></font><br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Simple transformation protocol: transformation with unaltered pUC19"""</span></span> p = Protocol() experiment_name = <span class="hljs-string"><span class="hljs-string">"debug_sfgfp_puc19_gibson_v1"</span></span> inv = { <span class="hljs-string"><span class="hljs-string">"water"</span></span> : <span class="hljs-string"><span class="hljs-string">"rs17gmh5wafm5p"</span></span>, <span class="hljs-comment"><span class="hljs-comment"># catalog; Autoclaved MilliQ H2O; ambient "DH5a" : "rs16pbj944fnny", # catalog; Zymo DH5Œ±; cold_80 "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "pUC19" : "rs17tcqmncjfsh", # catalog; pUC19; cold_20 } # Catalog transform_plate = p.ref("transform_plate", cont_type="96-pcr", storage="ambient", discard=True) transform_tube = transform_plate.well(0) # ------------------------------------------------------------------------------------ # Plating transformed bacteria according to Tali's protocol (requires different code!) # http://learn.transcriptic.com/blog/2015/9/9/provisioning-commercial-reagents # Add 1-5ul plasmid and pre-warm culture plates to 37C before starting. # # # Extra inventory for plating # inv["lb-broth-100ug-ml-amp_6-flat"] = "ki17sbb845ssx9" # (kit, not normal ref) from blogpost inv["noAB-amp_6-flat"] = "ki17reefwqq3sq" # kit id inv["LB Miller"] = "rs17bafcbmyrmh" # # Ampicillin and no ampicillin plates # amp_6_flat = Container(None, p.container_type('6-flat')) p.refs["amp_6_flat"] = Ref('amp_6_flat', {"reserve": inv['lb-broth-100ug-ml-amp_6-flat'], "store": {"where": 'cold_4'}}, amp_6_flat) noAB_6_flat = Container(None, p.container_type('6-flat')) p.refs["noAB_6_flat"] = Ref('noAB_6_flat', {"reserve": inv['noAB-amp_6-flat'], "store": {"where": 'cold_4'}}, noAB_6_flat) # # Provision competent bacteria # p.provision(inv["DH5a"], transform_tube, ¬µl(50)) p.provision(inv["pUC19"], transform_tube, ¬µl(2)) # # Heatshock the bacteria to transform using a PCR machine # p.seal(transform_plate) p.thermocycle(transform_plate, [{"cycles": 1, "steps": [{"temperature": "4:celsius", "duration": "5:minute"}]}, {"cycles": 1, "steps": [{"temperature": "37:celsius", "duration": "30:minute"}]}], volume=¬µl(50)) p.unseal(transform_plate) # # Then dilute bacteria and spread onto 6-flat plates # Put more on ampicillin plates for more opportunities to get a colony # p.provision(inv["LB Miller"], transform_tube, ¬µl(355)) p.mix(transform_tube, ¬µl(150), repetitions=5) for i in range(6): p.spread(transform_tube, amp_6_flat.well(i), ¬µl(55)) p.spread(transform_tube, noAB_6_flat.well(i), ¬µl(10)) assert transform_tube.volume &gt;= ¬µl(15), transform_tube.volume # # Incubate and image 6-flat plates over 18 hours # for flat_name, flat in [("amp_6_flat", amp_6_flat), ("noAB_6_flat", noAB_6_flat)]: for timepoint in [6,12,18]: p.cover(flat) p.incubate(flat, "warm_37", "6:hour") p.uncover(flat) p.image_plate(flat, mode="top", dataref=expid("{}_t{}".format(flat_name, timepoint))) # --------------------------------------------------------------- # Analyze protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("Protocol {}\n\n{}".format(experiment_name, protocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ‚úì Protocolo analisado</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  43 instru√ß√µes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  3 recipientes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  $ 45,43 </font></font></pre><br><h3><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: transforma√ß√£o de teste </font></font></h3><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Nas fotografias a seguir, vemos que sem um antibi√≥tico (placa √† esquerda), observa-se crescimento nas seis placas, embora em um grau vari√°vel, o que causa preocupa√ß√£o. Parece que os rob√¥s transcriptic realmente n√£o lidam com a distribui√ß√£o uniforme, o que requer certa destreza.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Na presen√ßa do antibi√≥tico (placa √† direita), tamb√©m h√° crescimento, embora novamente seja inconsistente. As duas primeiras placas com antibi√≥ticos parecem estranhas, com grande crescimento, o que √© provavelmente o resultado da adi√ß√£o de 55 ¬µl nessas placas, em compara√ß√£o com 10 ¬µl em placas sem antibi√≥ticos. Existem v√°rias col√¥nias na terceira placa e, em ess√™ncia, √© isso que eu esperava ver em todas as placas. Deve haver algum crescimento nas √∫ltimas tr√™s placas, mas isso n√£o acontece. Minha √∫nica explica√ß√£o para esses resultados estranhos √© que eu n√£o misturei as c√©lulas e o meio o suficiente, ent√£o quase todas as c√©lulas entraram nas duas primeiras placas. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">(Eu ainda tinha que fazer um controle positivo da ampicilina com uma bact√©ria inalterada, mas j√° fiz isso em um experimento anterior, por isso sei que a ampicilina estoque deve matar essa cepa</font></font><i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">E. coli</font></font></i><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">O crescimento √© muito mais fraco nas placas de ampicilina, embora haja muito mais bact√©rias, como esperado). </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">No geral, a transforma√ß√£o funcionou bem o suficiente para continuar, embora haja algumas falhas. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/296/c9b/7e2/296c9b7e28b653015159907b8a2bb59e.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Placas de c√©lulas transformadas com pUC19 ap√≥s 18 horas: sem antibi√≥tico (esquerda) e com antibi√≥tico (direita)</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Transforma√ß√£o do produto ap√≥s a montagem </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Como a montagem de Gibson e a simples transforma√ß√£o de pUC19 parecem funcionar, agora voc√™ pode tentar a transforma√ß√£o com um plasm√≠deo totalmente montado que expressa sfGFP. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Al√©m da inser√ß√£o coletada, tamb√©m adicionarei um pouco de IPTG e X-gal √†s placas para ver a convers√£o bem-sucedida usando o m√©todo de </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">sele√ß√£o branco-azul</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Esta informa√ß√£o adicional √© √∫til, porque se a transforma√ß√£o ocorrer com a pUC19 usual, que n√£o cont√©m sfGFP, ela ainda dar√° resist√™ncia a antibi√≥ticos.</font></font><br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Absor√ß√£o e fluoresc√™ncia </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">De acordo com esta </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">tabela</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> , o sfGFP brilha melhor em comprimentos de onda de excita√ß√£o de 485 nm / 510 nm. </font><font style="vertical-align: inherit;">Descobri que no Transcriptic, o 485/535 funciona melhor. </font><font style="vertical-align: inherit;">Eu acho que porque 485 e 510 s√£o muito semelhantes. </font><font style="vertical-align: inherit;">Medo o crescimento bacteriano a 600 nm ( </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">OD600</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ). </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/6ab/0d3/453/6ab0d345359a5b1d15f8f6341f1bee08.png"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Variedade de GFP ( </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">biotek</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> )</font></font></font></i> <br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> IPTG e X-gal </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Meu IPTG est√° na concentra√ß√£o de 1M e deve ser dilu√≠do 1: 1000. </font><font style="vertical-align: inherit;">Por sua vez, o X-gal na concentra√ß√£o de 20 mg / ml tamb√©m deve ser dilu√≠do 1: 1000 (20 mg / Œºl). </font><font style="vertical-align: inherit;">Portanto, a 2000 ¬µl de LB, adiciono 2 ¬µl cada. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">De acordo com um </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">protocolo,</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> voc√™ deve primeiro tomar 40 mL de X-gal a uma concentra√ß√£o de 20 mg / ml e 40 mL de concentra√ß√£o de IPTG de 0,1 mM (ou 4 mL de IPTG por 1M) e incubar por 30 minutos. </font><font style="vertical-align: inherit;">Esse procedimento n√£o funcionou para mim, ent√£o apenas misturei IPTG, X-gal e as c√©lulas correspondentes e usei essa mistura diretamente.</font></font><br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Full Gibson assembly and transformation protocol for sfGFP and pUC19 v1: Spread IPTG and X-gal onto plates, then spread cells v2: Mix IPTG, X-gal and cells; spread the mixture v3: exclude X-gal so I can do colony picking better v4: repeat v3 to try other excitation/emission wavelengths"""</span></span> p = Protocol() options = { <span class="hljs-string"><span class="hljs-string">"gibson"</span></span> : <span class="hljs-keyword"><span class="hljs-keyword">False</span></span>, <span class="hljs-comment"><span class="hljs-comment"># do a new gibson assembly "sanger" : False, # sanger sequence product "control_pUC19" : True, # unassembled pUC19 "XGal" : False # excluding X-gal should make the colony picking easier } for k, v in list(options.items()): if v is False: del options[k] experiment_name = "sfgfp_puc19_gibson_plates_v4" # ----------------------------------------------------------------------- # Inventory # inv = { # catalog "water" : "rs17gmh5wafm5p", # catalog; Autoclaved MilliQ H2O; ambient "DH5a" : "rs16pbj944fnny", # catalog; Zymo DH5Œ±; cold_80 "Gibson Mix" : "rs16pfatkggmk5", # catalog; Gibson Mix (2X); cold_20 "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "pUC19" : "rs17tcqmncjfsh", # catalog; pUC19; cold_20 # my inventory "puc19_cut_v2": "ct187v4ea7vvca", # inventory; pUC19 cut with EcoRI; cold_20 "IPTG" : "ct18a2r5wn6tqz", # inventory; IPTG at 1M (conc semi-documented); cold_20 "XGal" : "ct18a2r5wp5hcv", # inventory; XGal at 0.1M (conc not documented); cold_20 "sfgfp_pcroe_v8_amplified" : "ct1874zqh22pab", # inventory; sfGFP amplified to 40ng/ul; cold_4 "sfgfp_puc19_gibson_v3_clone" : "ct188ejywa8jcv", # inventory; assembled sfGFP; cold_4 # kits (must be used differently) "lb-broth-100ug-ml-amp_6-flat" : "ki17sbb845ssx9", # catalog; ampicillin plates "noAB-amp_6-flat" : "ki17reefwqq3sq" # catalog; no antibiotic plates } # # Catalog (all to be discarded afterward) # water_tube = p.ref("water", cont_type="micro-1.5", storage="ambient", discard=True).well(0) transform_plate = p.ref("trn_plate", cont_type="96-pcr", storage="ambient", discard=True) transform_tube = transform_plate.well(39) # experiment transform_tube_L = p.ref("trn_tubeL", cont_type="micro-1.5", storage="ambient", discard=True).well(0) transctrl_tube = transform_plate.well(56) # control transctrl_tube_L = p.ref("trc_tubeL", cont_type="micro-1.5", storage="ambient", discard=True).well(0) # # Plating according to Tali's protocol # http://learn.transcriptic.com/blog/2015/9/9/provisioning-commercial-reagents # amp_6_flat = Container(None, p.container_type('6-flat')) p.refs[expid("amp_6_flat")] = Ref(expid("amp_6_flat"), {"reserve": inv['lb-broth-100ug-ml-amp_6-flat'], "store": {"where": 'cold_4'}}, amp_6_flat) noAB_6_flat = Container(None, p.container_type('6-flat')) p.refs[expid("noAB_6_flat")] = Ref(expid("noAB_6_flat"), {"reserve": inv['noAB-amp_6-flat'], "store": {"where": 'cold_4'}}, noAB_6_flat) # # My inventory: EcoRI-cut pUC19, oePCR'd sfGFP, Gibson-assembled pUC19, IPTG and X-Gal # if "gibson" in options: puc19_cut_tube = p.ref("puc19_ecori_v2_puc19_cut", id=inv["puc19_cut_v2"], cont_type="micro-1.5", storage="cold_20").well(0) sfgfp_pcroe_amp_tube = p.ref("sfgfp_pcroe_v8_amplified", id=inv["sfgfp_pcroe_v8_amplified"], cont_type="micro-1.5", storage="cold_4").well(0) clone_plate = p.ref(expid("clone"), cont_type="96-pcr", storage="cold_4", discard=False) else: clone_plate = p.ref("sfgfp_puc19_gibson_v3_clone", id=inv["sfgfp_puc19_gibson_v3_clone"], cont_type="96-pcr", storage="cold_4", discard=False) IPTG_tube = p.ref("IPTG", id=inv["IPTG"], cont_type="micro-1.5", storage="cold_20").well(0) if "XGal" in options: XGal_tube = p.ref("XGal", id=inv["XGal"], cont_type="micro-1.5", storage="cold_20").well(0) # # Initialize inventory # if "gibson" in options: all_inventory_wells = [puc19_cut_tube, sfgfp_pcroe_amp_tube, IPTG_tube] assert puc19_cut_tube.volume == ¬µl(66), puc19_cut_tube.volume assert sfgfp_pcroe_amp_tube.volume == ¬µl(36), sfgfp_pcroe_amp_tube.volume else: all_inventory_wells = [IPTG_tube, clone_plate.well(0)] if "XGal" in options: all_inventory_wells.append(XGal_tube) for well in all_inventory_wells: init_inventory_well(well) print("Inventory: {} {} {}".format(well.name, well.volume, well.properties)) # # Provisioning. Water is used all over the protocol. Provision an excess since it's cheap # p.provision(inv["water"], water_tube, ¬µl(500)) # ----------------------------------------------------------------------------- # Cloning/assembly (see NEBuilder protocol above) # # "Optimized efficiency is 50‚Äì100 ng of vectors with 2 fold excess of inserts." # pUC19 is 20ng/ul (78ul total). # sfGFP is ~40ng/ul (48ul total) # Therefore 4ul of each gives 80ng and 160ng of vector and insert respectively # def do_gibson_assembly(): # # Combine all the Gibson reagents in one tube and thermocycle # p.provision(inv["Gibson Mix"], clone_plate.well(0), ¬µl(10)) p.transfer(water_tube, clone_plate.well(0), ¬µl(2)) p.transfer(puc19_cut_tube, clone_plate.well(0), ¬µl(4)) p.transfer(sfgfp_pcroe_amp_tube, clone_plate.well(0), ¬µl(4), mix_after=True, mix_vol=¬µl(10), repetitions=10) p.seal(clone_plate) p.thermocycle(clone_plate, [{"cycles": 1, "steps": [{"temperature": "50:celsius", "duration": "16:minute"}]}], volume=¬µl(50)) # # Dilute assembled plasmid 4X according to the NEB Gibson assembly protocol (20ul-&gt;80ul) # p.unseal(clone_plate) p.transfer(water_tube, clone_plate.well(0), ¬µl(60), mix_after=True, mix_vol=¬µl(40), repetitions=5) return # -------------------------------------------------------------------------------------------------- # Transformation # "Transform NEB 5-alpha Competent E. coli cells with 2 Œºl of the # assembled product, following the appropriate transformation protocol." # # Mix &amp; Go http://www.zymoresearch.com/downloads/dl/file/id/173/t3015i.pdf # "[After mixing] Immediately place on ice and incubate for 2-5 minutes" # "The highest transformation efficiencies can be obtained by incubating Mix &amp; Go cells with DNA on # ice for 2-5 minutes (60 minutes maximum) prior to plating." # "It is recommended that culture plates be pre-warmed to &gt;20¬∞C (preferably 37¬∞C) prior to plating." # "Avoid exposing the cells to room temperature for more than a few seconds at a time." # # "If competent cells are purchased from other manufacture, dilute assembled products 4-fold # with H2O prior transformation. This can be achieved by mixing 5 Œºl of assembled products with # 15 Œºl of H2O. Add 2 Œºl of the diluted assembled product to competent cells." # def _do_transformation(): # # Combine plasmid and competent bacteria in a pcr_plate and shock # p.provision(inv["DH5a"], transform_tube, ¬µl(50)) p.transfer(clone_plate.well(0), transform_tube, ¬µl(3), dispense_speed="10:microliter/second") assert clone_plate.well(0).volume == ¬µl(54), clone_plate.well(0).volume if 'control_pUC19' in options: p.provision(inv["DH5a"], transctrl_tube, ¬µl(50)) p.provision(inv["pUC19"], transctrl_tube, ¬µl(1)) # # Heatshock the bacteria to transform using a PCR machine # p.seal(transform_plate) p.thermocycle(transform_plate, [{"cycles": 1, "steps": [{"temperature": "4:celsius", "duration": "5:minute"}]}, {"cycles": 1, "steps": [{"temperature": "37:celsius", "duration": "30:minute"}]}], volume=¬µl(50)) return def _transfer_transformed_to_plates(): assert transform_tube.volume == ¬µl(53), transform_tube.volume p.unseal(transform_plate) num_ab_plates = 4 # antibiotic places # # Transfer bacteria to a bigger tube for diluting # Then spread onto 6-flat plates # Generally you would spread 50-100ul of diluted bacteria # Put more on ampicillin plates for more opportunities to get a colony # I use a dilution series since it's unclear how much to plate # p.provision(inv["LB Miller"], transform_tube_L, ¬µl(429)) # # Add all IPTG and XGal to the master tube # 4ul (1M) IPTG on each plate; 40ul XGal on each plate # p.transfer(IPTG_tube, transform_tube_L, ¬µl(4*num_ab_plates)) if 'XGal' in options: p.transfer(XGal_tube, transform_tube_L, ¬µl(40*num_ab_plates)) # # Add the transformed cells and mix (use new mix op in case of different pipette) # p.transfer(transform_tube, transform_tube_L, ¬µl(50)) p.mix(transform_tube_L, volume=transform_tube_L.volume/2, repetitions=10) assert transform_tube.volume == dead_volume['96-pcr'] == ¬µl(3), transform_tube.volume assert transform_tube_L.volume == ¬µl(495), transform_tube_L.volume # # Spread an average of 60ul on each plate == 480ul total # for i in range(num_ab_plates): p.spread(transform_tube_L, amp_6_flat.well(i), ¬µl(51+i*6)) p.spread(transform_tube_L, noAB_6_flat.well(i), ¬µl(51+i*6)) assert transform_tube_L.volume == dead_volume["micro-1.5"], transform_tube_L.volume # # Controls: include 2 ordinary pUC19-transformed plates as a control # if 'control_pUC19' in options: num_ctrl = 2 assert num_ab_plates + num_ctrl &lt;= 6 p.provision(inv["LB Miller"], transctrl_tube_L, ¬µl(184)+dead_volume["micro-1.5"]) p.transfer(IPTG_tube, transctrl_tube_L, ¬µl(4*num_ctrl)) if "XGal" in options: p.transfer(XGal_tube, transctrl_tube_L, ¬µl(40*num_ctrl)) p.transfer(transctrl_tube, transctrl_tube_L, ¬µl(48)) p.mix(transctrl_tube_L, volume=transctrl_tube_L.volume/2, repetitions=10) for i in range(num_ctrl): p.spread(transctrl_tube_L, amp_6_flat.well(num_ab_plates+i), ¬µl(55+i*10)) p.spread(transctrl_tube_L, noAB_6_flat.well(num_ab_plates+i), ¬µl(55+i*10)) assert transctrl_tube_L.volume == dead_volume["micro-1.5"], transctrl_tube_L.volume assert IPTG_tube.volume == ¬µl(808), IPTG_tube.volume if "XGal" in options: assert XGal_tube.volume == ¬µl(516), XGal_tube.volume return def do_transformation(): _do_transformation() _transfer_transformed_to_plates() # ------------------------------------------------------ # Measure growth in plates (photograph) # def measure_growth(): # # Incubate and photograph 6-flat plates over 18 hours # to see blue or white colonies # for flat_name, flat in [(expid("amp_6_flat"), amp_6_flat), (expid("noAB_6_flat"), noAB_6_flat)]: for timepoint in [9,18]: p.cover(flat) p.incubate(flat, "warm_37", "9:hour") p.uncover(flat) p.image_plate(flat, mode="top", dataref=expid("{}_t{}".format(flat_name, timepoint))) return # --------------------------------------------------------------- # Sanger sequencing, TURNED OFF # Sequence to make sure assembly worked # 500ng plasmid, 1 ¬µl of a 10 ¬µM stock primer # "M13_F" : "rs17tcpqwqcaxe", # catalog; M13 Forward (-41); cold_20 (1ul = 100pmol) # "M13_R" : "rs17tcph6e2qzh", # catalog; M13 Reverse (-48); cold_20 (1ul = 100pmol) # def do_sanger_seq(): seq_primers = [inv["M13_F"], inv["M13_R"]] seq_wells = ["G1","G2"] p.unseal(pcr_plate) for primer_num, seq_well in [(0, seq_wells[0]),(1, seq_wells[1])]: p.provision(seq_primers[primer_num], pcr_plate.wells([seq_well]), ¬µl(1)) p.transfer(pcr_plate.wells(["A1"]), pcr_plate.wells(seq_wells), ¬µl(5), mix_before=True, mix_vol=¬µl(10)) p.transfer(water_tube, pcr_plate.wells(seq_wells), ¬µl(9)) p.mix(pcr_plate.wells(seq_wells), volume=¬µl(7.5), repetitions=10) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[0]).indices(), expid("seq1")) p.sangerseq(pcr_plate, pcr_plate.wells(seq_wells[1]).indices(), expid("seq2")) return # --------------------------------------------------------------- # Generate protocol # # Skip Gibson since I already did it if 'gibson' in options: do_gibson_assembly() do_transformation() measure_growth() if 'sanger' in options: do_sanger_seq() # --------------------------------------------------------------- # Output protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze #print("\nProtocol {}\n\n{}".format(experiment_name, jprotocol)) open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Invent√°rio: IPTG / IPTG / IPTG / IPTG / IPTG / IPTG 832.0: microlitro {}</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
Invent√°rio: sfgfp_puc19_gibson_v3_clone / sfgfp_puc19_gibson_v3_clone / sfgfp_puc19_gibson_v3_clone / sfgfp_puc19_gibson_v3_clone / sfgfp_puc19_clib}</font></font><font></font>
<font></font>
  <font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
‚úì Protocolo analisado</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  40 instru√ß√µes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  8 contentores</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Custo total: $ 53.20</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Tempo de Workcell: $ 17,35</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Reagentes e Consum√≠veis: $ 35,86 </font></font></pre><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Cole√ß√£o col√¥nia </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Quando as col√¥nias crescem em uma placa de ampicilina, eu posso "coletar" col√¥nias individuais e plant√°-las em uma placa de 96 tubos. </font><font style="vertical-align: inherit;">Para isso, h√° uma equipe especial ( </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">autopick</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ) </font><font style="vertical-align: inherit;">no </font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;">protocolo</font></a><font style="vertical-align: inherit;"> autom√°tico </font><font style="vertical-align: inherit;">.</font></font><br><br><div class="spoiler">  <b class="spoiler_title">C√≥digo</b> <div class="spoiler_text"><pre> <code class="python hljs"><span class="hljs-string"><span class="hljs-string">"""Pick colonies from plates and grow in amp media and check for fluorescence. v2: try again with a new plate (no blue colonies) v3: repeat with different emission and excitation wavelengths"""</span></span> p = Protocol() options = {} <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> k, v <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> list(options.items()): <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> v <span class="hljs-keyword"><span class="hljs-keyword">is</span></span> <span class="hljs-keyword"><span class="hljs-keyword">False</span></span>: <span class="hljs-keyword"><span class="hljs-keyword">del</span></span> options[k] experiment_name = <span class="hljs-string"><span class="hljs-string">"sfgfp_puc19_gibson_pick_v3"</span></span> <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">plate_expid</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(val)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-string"><span class="hljs-string">"""refer to the previous plating experiment's outputs"""</span></span> plate_exp = <span class="hljs-string"><span class="hljs-string">"sfgfp_puc19_gibson_plates_v4"</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> <span class="hljs-string"><span class="hljs-string">"{}_{}"</span></span>.format(plate_exp, val) <span class="hljs-comment"><span class="hljs-comment"># ----------------------------------------------------------------------- # Inventory # inv = { # catalog "water" : "rs17gmh5wafm5p", # catalog; Autoclaved MilliQ H2O; ambient "LB Miller" : "rs17bafcbmyrmh", # catalog; LB Broth Miller; cold_4 "Amp 100mgml" : "rs17msfk8ujkca", # catalog; Ampicillin 100mg/ml; cold_20 "IPTG" : "ct18a2r5wn6tqz", # inventory; IPTG at 1M (conc semi-documented); cold_20 # plates from previous experiment, must be changed every new experiment plate_expid("amp_6_flat") : "ct18snmr9avvg9", # inventory; Ampicillin plates with blue-white screening of pUC19 plate_expid("noAB_6_flat") : "ct18snmr9dxfw2", # inventory; no AB plates with blue-white screening of pUC19 } # Tubes and plates lb_amp_tubes = [p.ref("lb_amp_{}".format(i+1), cont_type="micro-2.0", storage="ambient", discard=True).well(0) for i in range(4)] lb_xab_tube = p.ref("lb_xab", cont_type="micro-2.0", storage="ambient", discard=True).well(0) growth_plate = p.ref(expid("growth"), cont_type="96-flat", storage="cold_4", discard=False) # My inventory IPTG_tube = p.ref("IPTG", id=inv["IPTG"], cont_type="micro-1.5", storage="cold_20").well(0) # ampicillin plate amp_6_flat = Container(None, p.container_type('6-flat')) p.refs[plate_expid("amp_6_flat")] = Ref(plate_expid("amp_6_flat"), {"id":inv[plate_expid("amp_6_flat")], "store": {"where": 'cold_4'}}, amp_6_flat) # Use a total of 50 wells abs_wells = ["{}{}".format(row,col) for row in "BCDEF" for col in range(1,11)] abs_wells_T = ["{}{}".format(row,col) for col in range(1,11) for row in "BCDEF"] assert abs_wells[:3] == ["B1","B2","B3"] and abs_wells_T[:3] == ["B1","C1","D1"] def prepare_growth_wells(): # # To LB, add ampicillin at ~1/1000 concentration # Mix slowly in case of overflow # p.provision(inv["LB Miller"], lb_xab_tube, ¬µl(1913)) for lb_amp_tube in lb_amp_tubes: p.provision(inv["Amp 100mgml"], lb_amp_tube, ¬µl(2)) p.provision(inv["LB Miller"], lb_amp_tube, ¬µl(1911)) p.mix(lb_amp_tube, volume=¬µl(800), repetitions=10) # # Add IPTG but save on X-Gal # http://openwetware.org/images/f/f1/Dh5a_sub.pdf # "If you are concerned about obtaining maximal levels of expression, add IPTG to a final concentration of 1 mM." # 2ul of IPTG in 2000ul equals 1mM # p.transfer(IPTG_tube, [lb_xab_tube] + lb_amp_tubes, ¬µl(2), one_tip=True) # # Distribute LB among wells, row D is control (no ampicillin) # cols = range(1,11) row = "D" # control, no AB cwells = ["{}{}".format(row,col) for col in cols] assert set(cwells).issubset(set(abs_wells)) p.distribute(lb_xab_tube, growth_plate.wells(cwells), ¬µl(190), allow_carryover=True) rows = "BCEF" for row, lb_amp_tube in zip(rows, lb_amp_tubes): cwells = ["{}{}".format(row,col) for col in cols] assert set(cwells).issubset(set(abs_wells)) p.distribute(lb_amp_tube, growth_plate.wells(cwells), ¬µl(190), allow_carryover=True) assert all(lb_amp_tube.volume == lb_xab_tube.volume == dead_volume['micro-2.0'] for lb_amp_tube in lb_amp_tubes) return def measure_growth_wells(): # # Growth: absorbance and fluorescence over 24 hours # Absorbance at 600nm: cell growth # Absorbance at 615nm: X-gal, in theory # Fluorescence at 485nm/510nm: sfGFP # or 450nm/508nm (http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2695656/) # hr = 4 for t in range(0,24,hr): if t &gt; 0: p.cover(growth_plate) p.incubate(growth_plate, "warm_37", "{}:hour".format(hr), shaking=True) p.uncover(growth_plate) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="485:nanometer", emission="535:nanometer", dataref=expid("fl2_{}".format(t)), flashes=25) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="450:nanometer", emission="508:nanometer", dataref=expid("fl1_{}".format(t)), flashes=25) p.fluorescence(growth_plate, growth_plate.wells(abs_wells).indices(), excitation="395:nanometer", emission="508:nanometer", dataref=expid("fl0_{}".format(t)), flashes=25) p.absorbance(growth_plate, growth_plate.wells(abs_wells).indices(), wavelength="600:nanometer", dataref=expid("abs_{}".format(t)), flashes=25) return # --------------------------------------------------------------- # Protocol steps # prepare_growth_wells() batch = 10 for i in range(5): p.autopick(amp_6_flat.well(i), growth_plate.wells(abs_wells_T[i*batch:i*batch+batch]), dataref=expid("autopick_{}".format(i))) p.image_plate(amp_6_flat, mode="top", dataref=expid("autopicked_{}".format(i))) measure_growth_wells() # --------------------------------------------------------------- # Output protocol # jprotocol = json.dumps(p.as_dict(), indent=2) !echo '{jprotocol}' | transcriptic analyze open("protocol_{}.json".format(experiment_name),'w').write(jprotocol)</span></span></code> </pre> </div></div><br><pre><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ‚úì Protocolo analisado</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  62 instru√ß√µes</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  8 contentores</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Custo total: $ 66.38</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Tempo de Trabalho: $ 57.59</font></font><font></font><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">
  Reagentes e Consum√≠veis: $ 8,78 </font></font></pre><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: colheita de col√¥nias </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">A tela azul-branca mostrou perfeitamente, principalmente, col√¥nias brancas em placas com antibi√≥tico (1-4) e apenas azul em placas sem antibi√≥tico (5-6). Era exatamente o que eu esperava e fiquei feliz em v√™-lo, especialmente porque usei meu pr√≥prio IPTG e X-gal, que enviei para a Transcriptic. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/bc4/d2b/c10/bc4d2bc102d423993907a021598257ef.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Triagem por sele√ß√£o de placas azul-branco com ampicilina (1-4) e sem antibi√≥tico (5-6)</font></font></font></i> <br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> No entanto, o rob√¥ coletor de col√¥nias n√£o funcionou bem com essas col√¥nias azul-brancas. A imagem abaixo foi criada subtraindo fotos sucessivas das placas ap√≥s cada rodada de sele√ß√£o de placas e aumentando o contraste das diferen√ßas (no </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">GraphicsMagick</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ). Dessa maneira, posso visualizar quais col√¥nias foram coletadas (embora n√£o idealmente, pois as col√¥nias coletadas n√£o s√£o completamente removidas).</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Eu tamb√©m assinei a imagem com o n√∫mero de col√¥nias coletadas pelo rob√¥ Transcriptic. Supunha-se que ele coletaria no m√°ximo 10 col√¥nias das cinco primeiras placas. No entanto, em geral, v√°rias col√¥nias foram coletadas, e geralmente s√£o col√¥nias azuis. O rob√¥ s√≥ conseguiu encontrar dez col√¥nias em uma placa de controle com apenas col√¥nias azuis. Minha teoria de trabalho √© que um rob√¥ coletor de col√¥nias preferencialmente coleta col√¥nias azuis porque elas s√£o mais contrastantes. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/c8f/ee9/b71/c8fee9b710a214fdb7421dbc2d74e1b4.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Placas de triagem para sele√ß√£o azul e branco com ampicilina (1-4) e sem antibi√≥tico (5-6), indicando o n√∫mero de col√¥nias coletadas</font></font></font></i> <br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">A triagem em azul e branco serviu a um prop√≥sito espec√≠fico. </font><font style="vertical-align: inherit;">Ele mostrou que a maioria das col√¥nias se transforma corretamente. </font><font style="vertical-align: inherit;">Pelo menos h√° uma inser√ß√£o. </font><font style="vertical-align: inherit;">No entanto, para uma melhor coleta de col√¥nias, repeti o experimento sem o X-gal. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Somente com col√¥nias brancas o coletor de rob√¥s conseguiu reunir dez col√¥nias de cada uma das cinco primeiras placas. </font><font style="vertical-align: inherit;">Pode-se supor que na maioria das col√¥nias coletadas h√° inser√ß√µes bem-sucedidas. </font></font><br><br><img src="https://habrastorage.org/getpro/habr/post_images/2cb/6f1/835/2cb6f183502e484f66e1f95ba1b44ed7.jpg"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Col√¥nias que crescem em placas com ampicilina (1-4) e sem antibi√≥tico (5-6)</font></font></font></i> <br><br><h2><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Resultados: transforma√ß√£o com produto montado </font></font></h2><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Depois de cultivar 50 col√¥nias selecionadas em uma placa de 96 tubos por 20 horas, medi a fluoresc√™ncia para verificar a express√£o de sfGFP. A Transcriptic usa um leitor </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Tecan Infinite</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> para medir a fluoresc√™ncia e absor√ß√£o (e luminesc√™ncia, se voc√™ preferir) </font><font style="vertical-align: inherit;">.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Em teoria, em qualquer col√¥nia, um plasm√≠deo deve ser montado com crescimento, uma vez que precisa de resist√™ncia a antibi√≥ticos para crescer, e cada plasm√≠deo coletado expressa sfGFP. Na verdade, existem muitas raz√µes pelas quais isso pode n√£o ser o caso, principalmente porque voc√™ pode perder o gene sfGFP do plasm√≠deo sem perder a resist√™ncia √† ampicilina. Uma bact√©ria que perde o gene sfGFP tem uma vantagem na sele√ß√£o em rela√ß√£o a seus concorrentes, porque n√£o gasta energia extra e, considerando um n√∫mero suficiente de gera√ß√µes de crescimento, isso certamente acontecer√°. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Coleto dados de absor√ß√£o (OD600) e fluoresc√™ncia a cada quatro horas por 20 horas (cerca de 60 gera√ß√µes).</font></font><br><br><pre> <code class="python hljs"><span class="hljs-keyword"><span class="hljs-keyword">for</span></span> t <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> [<span class="hljs-number"><span class="hljs-number">0</span></span>,<span class="hljs-number"><span class="hljs-number">4</span></span>,<span class="hljs-number"><span class="hljs-number">8</span></span>,<span class="hljs-number"><span class="hljs-number">12</span></span>,<span class="hljs-number"><span class="hljs-number">16</span></span>,<span class="hljs-number"><span class="hljs-number">20</span></span>]: abs_data = pd.read_csv(<span class="hljs-string"><span class="hljs-string">"glow/sfgfp_puc19_gibson_pick_v3_abs_{}.csv"</span></span>.format(t), index_col=<span class="hljs-string"><span class="hljs-string">"Well"</span></span>) flr_data = pd.read_csv(<span class="hljs-string"><span class="hljs-string">"glow/sfgfp_puc19_gibson_pick_v3_fl2_{}.csv"</span></span>.format(t), index_col=<span class="hljs-string"><span class="hljs-string">"Well"</span></span>) <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> t == <span class="hljs-number"><span class="hljs-number">0</span></span>: new_data = abs_data.join(flr_data) <span class="hljs-keyword"><span class="hljs-keyword">else</span></span>: new_data = new_data.join(abs_data, rsuffix=<span class="hljs-string"><span class="hljs-string">'_{}'</span></span>.format(t)) new_data = new_data.join(flr_data, rsuffix=<span class="hljs-string"><span class="hljs-string">'_{}'</span></span>.format(t)) new_data.columns = [<span class="hljs-string"><span class="hljs-string">"OD 600:nanometer_0"</span></span>, <span class="hljs-string"><span class="hljs-string">"Fluorescence_0"</span></span>] + list(new_data.columns[<span class="hljs-number"><span class="hljs-number">2</span></span>:])</code> </pre> <br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Colocamos no gr√°fico os dados da vig√©sima hora e os tra√ßos das medi√ß√µes anteriores. De fato, estou interessado apenas nos dados mais recentes, pois √© ent√£o que um pico de fluoresc√™ncia deve ser observado.</font></font><br><br><pre> <code class="python hljs">svg = [] W, H = <span class="hljs-number"><span class="hljs-number">800</span></span>, <span class="hljs-number"><span class="hljs-number">500</span></span> min_x, max_x = <span class="hljs-number"><span class="hljs-number">0</span></span>, <span class="hljs-number"><span class="hljs-number">0.8</span></span> min_y, max_y = <span class="hljs-number"><span class="hljs-number">0</span></span>, <span class="hljs-number"><span class="hljs-number">50000</span></span> <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">_toxy</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(x, y)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> W*(x-min_x)/(max_x-min_x), HH*(y-min_y)/(max_y-min_y) <span class="hljs-function"><span class="hljs-keyword"><span class="hljs-function"><span class="hljs-keyword">def</span></span></span><span class="hljs-function"> </span><span class="hljs-title"><span class="hljs-function"><span class="hljs-title">_topt</span></span></span><span class="hljs-params"><span class="hljs-function"><span class="hljs-params">(x, y)</span></span></span><span class="hljs-function">:</span></span> <span class="hljs-keyword"><span class="hljs-keyword">return</span></span> <span class="hljs-string"><span class="hljs-string">','</span></span>.join(map(str,_toxy(x,y))) ab_fls = [[row[<span class="hljs-number"><span class="hljs-number">0</span></span>]] + [list(row[<span class="hljs-number"><span class="hljs-number">1</span></span>])] <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> row <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> new_data.iterrows()] <span class="hljs-comment"><span class="hljs-comment"># axes svg.append('&lt;g fill="#888" font-size="18" transform="translate(20,0),scale(.95)"&gt;') svg.append('&lt;text x="0" y="{}"&gt;OD600 ‚Üí&lt;/text&gt;'.format(H+20)) svg.append('&lt;text x="0" y="0" transform="rotate(-90),translate(-{},-8)"&gt;Fluorescence ‚Üí&lt;/text&gt;'.format(H)) svg.append('&lt;line x1="0" y1="{}" x2="{}" y2="{}" style="stroke:#888;stroke-width:2" /&gt;'.format(H,W,H)) svg.append('&lt;line x1="0" y1="0" x2="0" y2="{}" style="stroke:#888;stroke-width:2" /&gt;'.format(H)) # glow filter svg.append("""&lt;filter id="glow" x="-200%" y="-200%" height="400%" width="400%"&gt; &lt;feColorMatrix type="matrix" values="0 0 0 0 0 255 0 0 0 0 0 0 0 0 0 0 0 0 1 0"/&gt; &lt;feGaussianBlur stdDeviation="10" result="coloredBlur"/&gt; &lt;feMerge&gt;&lt;feMergeNode in="coloredBlur"/&gt;&lt;feMergeNode in="SourceGraphic"/&gt;&lt;/feMerge&gt; &lt;/filter&gt;""") for n, (well, vals) in enumerate(ab_fls): fill = "#444" if not well.startswith("D") else "#aaa" gfilter = 'filter="url(#glow)"' if well in ["C3", "D1", "D3"] else "" cx, cy = _toxy(*vals[-2:]) svg.append('''&lt;g id="point{n:d}"&gt;&lt;circle {gfilter:s} r="12" cx="{cx:f}" cy="{cy:f}" fill="{fill:s}" /&gt; &lt;text x="{cx:f}" y="{cy:f}" font-size="10" text-anchor="middle" fill="#fff" alignment-baseline="middle"&gt;{txt:s}&lt;/text&gt;&lt;/g&gt; '''.format(n=n, cx=cx, cy=cy, fill=fill, txt=well, gfilter=gfilter)) pathd = 'M{} '.format(_topt(*vals[:2])) pathd += ' '.join("L{}".format(_topt(*vals[i:i+2])) for i in range(2,len(vals),2)) svg.append('''&lt;path d="{pathd:}" stroke="#ccc" stroke-width=".2" fill="none" id="path{n:d}"/&gt;'''.format(pathd=pathd, n=n)) svg.append("&lt;/g&gt;") # entire chart group show_svg(''.join(svg), w=W, h=H)</span></span></code> </pre> <br><br><img src="https://habrastorage.org/webt/03/z3/kk/03z3kkczqokp7nhgplib_q1bhia.png"><br> <i><font color="gray"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Fluoresc√™ncia e OD600: col√¥nias com ampicilina s√£o pretas, col√¥nias de controle sem ampicilina s√£o cinzas. As col√¥nias foram destacadas em verde, onde eu confirmei que a sequ√™ncia da prote√≠na sfGFP estava correta:</font></font></font></i> <br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> execute o</font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> miniprep</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> para extrair o DNA do plasm√≠deo e, em seguida, o sequenciamento de Sanger usando os iniciadores M13. Infelizmente, por algum motivo, o miniprep atualmente est√° dispon√≠vel apenas pelo protocolo da web Transcriptic, e n√£o pelo protocolo autom√°tico. Vou sequenciar os tr√™s tubos com as leituras mais altas de fluoresc√™ncia (C1, D1, D3) e os outros tr√™s (B1, B3, E1), alinhando as seq√º√™ncias com sfGFP usando</font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> m√∫sculo</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> .</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Nos tubos C1, D3 e D3, a combina√ß√£o perfeita para minha sequ√™ncia sfGFP original, enquanto em B1, B3 e E1, muta√ß√µes grosseiras ou alinhamento simplesmente falham.</font></font><br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Tr√™s col√¥nias luminosas </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Os resultados s√£o bons, embora alguns aspectos sejam surpreendentes. </font><font style="vertical-align: inherit;">Por exemplo, um leitor de fluoresc√™ncia, sem motivo aparente, inicia no tempo 0 com valores muito altos (40.000 blocos). </font><font style="vertical-align: inherit;">Na vig√©sima hora, ele se acalmou para um padr√£o mais razo√°vel, com uma clara correla√ß√£o entre OD600 e fluoresc√™ncia (suponho que devido a uma pequena sobreposi√ß√£o nos espectros), al√©m de algumas emiss√µes com alta fluoresc√™ncia. </font><font style="vertical-align: inherit;">Surpreendentemente, pode ser uma, tr√™s ou, possivelmente, 11 a 15 emiss√µes. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Alguns dos tubos com altos valores de fluoresc√™ncia est√£o localizados nos tubos de controle (ou seja, sem ampicilina, s√£o coloridos em cinza), o que √© surpreendente, uma vez que n√£o h√° press√£o de sele√ß√£o nesses tubos, portanto, pode-se esperar perda de plasm√≠deo).</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Com base nos dados de fluoresc√™ncia e nos resultados de sequencia√ß√£o, parece que apenas tr√™s das 50 col√¥nias produzem sfGFP e fluorescem. </font><font style="vertical-align: inherit;">Isso n√£o √© tanto quanto eu esperava. </font><font style="vertical-align: inherit;">No entanto, desde que houve tr√™s est√°gios de crescimento separados (in vitro, para miniprep), cerca de 200 gera√ß√µes de crescimento foram submetidas a esse est√°gio da c√©lula; portanto, havia muitas oportunidades de ocorr√™ncia de muta√ß√µes. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Deve haver maneiras de tornar o processo mais eficiente, especialmente porque estou longe de ser um especialista nesses protocolos. </font><font style="vertical-align: inherit;">No entanto, produzimos com sucesso c√©lulas transformadas com express√£o do GFP manipulado usando apenas o c√≥digo Python!</font></font><br><br><h1><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Parte Tr√™s: Conclus√µes </font></font></h1><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Pre√ßo </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Dependendo de como medir, o custo desse experimento foi de cerca de US $ 360, sem incluir o dinheiro para depura√ß√£o: </font></font><br><br><ul><li> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">US $ 70</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> para a s√≠ntese de DNA</font></font><br></li><li> <b>$32</b>  PCR       <br></li><li> <b>$31</b>    <br></li><li> <b>$32</b>    <br></li><li> <b>$53</b>   <br></li><li> <b>$67</b>    <br></li><li> <b>$75</b>  3 miniprep'   </li></ul><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Eu acho que o custo pode ser reduzido para US $ 250-300 com algumas melhorias. Por exemplo, a cole√ß√£o rob√≥tica de 50 col√¥nias √© suspeita- mente cara e provavelmente pode ser abandonada. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Na minha experi√™ncia, esse pre√ßo parece alto para alguns (bi√≥logos moleculares) e baixo para outros (pessoas de TI). Como a Transcriptic basicamente cobra apenas os reagentes da lista de pre√ßos, a principal diferen√ßa nos custos √© m√£o-de-obra. O rob√¥ j√° √© bem barato em uma hora e ele n√£o se importaria de acordar no meio da noite para fotografar um prato. Uma vez aprovados os protocolos, √© dif√≠cil imaginar que mesmo um estudante de gradua√ß√£o seria mais barato, especialmente considerando os custos de oportunidade.</font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para maior clareza, estou apenas falando sobre a substitui√ß√£o de protocolos de rotina. </font><font style="vertical-align: inherit;">Certamente, o desenvolvimento de protocolos avan√ßados continuar√° sendo realizado por bi√≥logos moleculares qualificados, mas muitas √°reas interessantes da ci√™ncia usam protocolos rotineiros de rotina. </font><font style="vertical-align: inherit;">At√© recentemente, muitos laborat√≥rios produziam seus pr√≥prios oligonucleot√≠deos, mas agora poucas pessoas se preocupam com isso: isso n√£o custa a ningu√©m, nem mesmo estudantes de p√≥s-gradua√ß√£o, quando o IDT os envia a voc√™ dentro de alguns dias.</font></font><br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Laborat√≥rios rob√≥ticos: pr√≥s e contras </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Obviamente, eu realmente acredito no futuro dos laborat√≥rios rob√≥ticos. </font><font style="vertical-align: inherit;">Existem algumas coisas realmente engra√ßadas e √∫teis na experimenta√ß√£o de rob√¥s, especialmente se voc√™ estiver envolvido principalmente na computa√ß√£o e for al√©rgico a luvas de l√°tex e trabalho manual:</font></font><br><br><ul><li> <b>!</b> ,   .              autoprotocol,     . <br></li><li> <b></b> .      100    ,    . <br></li><li> <b>  </b> , ,  PCR.       ,          ,      ?    /       ?    ,   ,      ,      ¬´  2-3 ¬ª.  ? <br></li><li> <b> </b> .          .     ,   . <br></li><li> <b></b> .       . <br></li><li> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Expressividade</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Voc√™ pode usar a sintaxe de programa√ß√£o para codificar etapas repetidas ou l√≥gica de ramifica√ß√£o. </font><font style="vertical-align: inherit;">Por exemplo, se voc√™ deseja dispensar 1 a 96 Œºl de reagente e (96 - x) Œºl de √°gua em uma placa de 96 tubos, isso pode ser escrito brevemente.</font></font><br></li><li> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Dados leg√≠veis por m√°quina</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Os dados com resultados quase sempre s√£o retornados no formato csv ou outro formato adequado ao processamento da m√°quina.</font></font><br></li><li> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Abstra√ß√£o</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Idealmente, voc√™ pode executar o protocolo inteiro, independentemente dos reagentes utilizados ou do estilo de clonagem e substituir algo, se necess√°rio, se funcionar melhor.</font></font></li></ul><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Obviamente, existem algumas desvantagens, especialmente desde que as ferramentas come√ßaram a se desenvolver. </font><font style="vertical-align: inherit;">Se compararmos com a Internet, estaremos na √°rea de 1994:</font></font><br><br><ul><li>    Transcriptic   ‚Äî  .   ,    ,         ,    .         , ,     . <br></li><li>         ‚Äî        Transcriptic. <br></li><li>   ,      .     Transcriptic     ( ,   ,     ). <br></li><li><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para muitos laborat√≥rios, pode ser mais caro usar um laborat√≥rio em nuvem do que apenas levar um estudante de p√≥s-gradua√ß√£o para fazer o trabalho (custo marginal por hora: ~ $ 0). </font><font style="vertical-align: inherit;">Depende se o laborat√≥rio precisa das m√£os do aluno ou de sua intelig√™ncia.</font></font><br></li><li><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">A transcri√ß√£o ainda n√£o est√° fazendo experi√™ncias no fim de semana. </font><font style="vertical-align: inherit;">Voc√™ pode entend√™-los, mas isso pode ser inconveniente, mesmo se voc√™ tiver um projeto pequeno.</font></font></li></ul><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Software para fabrica√ß√£o de prote√≠nas </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Embora exista muito c√≥digo e muita depura√ß√£o, acho que √© poss√≠vel criar algum tipo de software que use uma sequ√™ncia de prote√≠nas como entrada e crie bact√©rias com a express√£o dessa prote√≠na na sa√≠da. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para que isso funcione, v√°rias coisas devem acontecer:</font></font><br><br><ul><li><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Verdadeira integra√ß√£o do Twist / IDT / Gen9 ao Transcriptic (provavelmente ser√° lento devido √† baixa demanda atual). </font></font><br></li><li>    ,    ,       , ,    . . <br></li><li>     (    NEB,    IDT)       (, <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">primer3</a> ). </li></ul><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Em muitas aplica√ß√µes, voc√™ tamb√©m deseja purificar sua prote√≠na (atrav√©s de uma </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">coluna</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ), ou talvez apenas fazer com que as bact√©rias a </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">secretem</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Suponha que em breve possamos fazer isso em um laborat√≥rio em nuvem, ou que possamos realizar experimentos in vivo (ou seja, dentro de uma c√©lula bacteriana). </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Existem muitas possibilidades para o protocolo realmente funcionar melhor que os seres humanos, por exemplo: o design de promotores e o RBS para otimizar a express√£o espec√≠fica da sua sequ√™ncia; </font><font style="vertical-align: inherit;">estat√≠sticas da probabilidade de sucesso de um experimento com base em experimentos compar√°veis; </font><font style="vertical-align: inherit;">an√°lise automatizada de gel.</font></font><br><br><h4><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Por que tudo isso? </font></font></h4><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Depois de tudo isso, pode n√£o estar completamente claro por que criar essa prote√≠na. </font><font style="vertical-align: inherit;">Aqui est√£o algumas id√©ias:</font></font><br><br><ul><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="></a>   - //,  . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="></a>   ,   ,  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="> </a> . <br></li><li>    <i>in vivo</i>   <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">split-GFP</a>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="> </a> . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">scFv</a>     .  scFvs   -  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">  </a> . <br></li><li>  <a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u=">BiTE</a>     ,      (   ,  ). <br></li><li><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Fa√ßa uma vacina local que entre no corpo </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">atrav√©s dos fol√≠culos capilares</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> (eu n√£o recomendo tentar isso em casa).</font></font><br></li><li><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Mutagenize sua prote√≠na de centenas de maneiras diferentes e veja o que acontece. </font><font style="vertical-align: inherit;">Ent√£o escalar para 1000 ou 10.000 muta√ß√µes? </font><font style="vertical-align: inherit;">Talvez caracterizar muta√ß√µes GFP?</font></font></li></ul><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Para novas id√©ias sobre o que √© poss√≠vel com o design de prote√≠nas, veja centenas </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">de projetos do iGEM</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">No final, quero agradecer √† </font><font style="vertical-align: inherit;">Transcriptic </font></font><a href="https://translate.googleusercontent.com/translate_c?depth=1&amp;pto=nl&amp;rurl=translate.google.com&amp;sl=ru&amp;sp=nmt4&amp;tl=pt&amp;u="><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Ben Miles</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> por sua ajuda na conclus√£o deste projeto.</font></font></div></div><p>Source: <a rel="nofollow" href="https://habr.com/ru/post/pt451124/">https://habr.com/ru/post/pt451124/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../pt451112/index.html">Jetpack compor</a></li>
<li><a href="../pt451114/index.html">Micr√≥bios ind√≠genas e pesquisa de produtos nas lojas</a></li>
<li><a href="../pt451116/index.html">Como lutei com a transi√ß√£o de elemento compartilhado e escrevi minha primeira biblioteca de c√≥digo-fonte aberto</a></li>
<li><a href="../pt451118/index.html">O teste n√£o √© para iniciantes</a></li>
<li><a href="../pt451120/index.html">Sobre os desafios de transportar c√©lulas mortas para plataformas m√≥veis</a></li>
<li><a href="../pt451126/index.html">Caixa de ferramentas para pesquisadores - Edi√ß√£o um: Auto-organiza√ß√£o e visualiza√ß√£o de dados</a></li>
<li><a href="../pt451130/index.html">Swift: ARC e gerenciamento de mem√≥ria</a></li>
<li><a href="../pt451132/index.html">Contratos orientados ao consumidor ou automa√ß√£o de teste de controle de qualidade com base no Gitlab CI</a></li>
<li><a href="../pt451138/index.html">Symfony CLI - Nova ferramenta de desenvolvimento local</a></li>
<li><a href="../pt451140/index.html">Quanto custa criar uma solu√ß√£o de IoT?</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter57283870 = new Ya.Metrika({
                  id:57283870,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/57283870" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134228602-6', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

<footer class="page-footer">
  <div class="page-footer-legal-info-container page-footer-element">
    <p>
      Weekly-Geekly ES | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
    </p>
  </div>
  <div class="page-footer-counters-container page-footer-element">
    <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=9uU9J9pq8z7k8xEBHYSfs6DenIBAHs3vLIHcPIJW9d0&co=3a3a3a&ct=ffffff'/></a>
  </div>
</footer>
  
</body>

</html>